// epic-designer/dist/core/src/components/designer/src/modules/rightSidebar/index.vue2.js
import { defineComponent as _, ref as d, computed as g, shallowRef as k, createElementBlock as s, createCommentVNode as b, openBlock as o, createElementVNode as n, createVNode as v, unref as C, normalizeClass as u, Fragment as y, renderList as w, toDisplayString as B, createBlock as x, resolveDynamicComponent as N } from "vue";
/* empty css                                                 */
import R from "../../../../../../../ui-kit/base-ui/src/icon/icon.vue.js";
/* empty css                                                             */
/* empty css                                                                  */
import { pluginManager as S } from "../../../../../../../manager/src/pluginManager.js";
import "../../../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import "@vueuse/core";
import "../../../../../../../hooks/src/store/index.js";
/* empty css                                                           */
import "vuedraggable";
import V from "./breadcrumb.vue.js";
const $ = {
  key: 0,
  class: "epic-right-sidebar-container relative"
}, z = { class: "epic-actions-container" }, D = ["title", "onClick"], E = { class: "epic-sidebar-content" }, T = /* @__PURE__ */ _({
  __name: "index",
  setup(M) {
    var m;
    const i = d(!1), p = g(() => S.panel.rightSidebars.value.filter((e) => e.visible).sort((e, t) => e.sort - t.sort)), a = d(0), r = k(null);
    r.value = (m = p.value[0]) == null ? void 0 : m.component;
    function f() {
      i.value = !i.value;
    }
    function h(e, t) {
      if (a.value === t)
        return !1;
      r.value = e.component, a.value = t;
    }
    return (e, t) => r.value ? (o(), s("div", $, [
      n("div", {
        class: "epic-right-sidebar-hide-btn z-9 absolute flex cursor-pointer items-center justify-center",
        onClick: f
      }, [
        v(C(R), {
          class: u(["transition-all", { "rotate-180": i.value }]),
          name: "icon--epic--arrow-forward-ios-rounded"
        }, null, 8, ["class"])
      ]),
      n("div", {
        class: u(["epic-right-sidebar w-308px", { hide: i.value }])
      }, [
        v(V),
        n("ul", z, [
          (o(!0), s(y, null, w(p.value, (l, c) => (o(), s("li", {
            key: c,
            class: u(["epic-action-item", { checked: a.value === c }]),
            title: l.title,
            onClick: (j) => h(l, c)
          }, B(l.title), 11, D))), 128))
        ]),
        n("div", E, [
          (o(), x(N(r.value)))
        ])
      ], 2)
    ])) : b("", !0);
  }
});
export {
  T as default
};
