// epic-designer/dist/core/src/components/designer/src/modules/editContainer/editScreenContainer.vue.js
import { defineComponent as L, ref as s, computed as P, shallowRef as U, watchEffect as q, onMounted as F, watch as _, nextTick as $, createElementBlock as G, openBlock as I, createVNode as J, createElementVNode as u, unref as r, normalizeClass as H, normalizeStyle as x, renderSlot as K } from "vue";
import { debounce as Q } from "../../../../../../../utils/src/common/common.js";
/* empty css                                                 */
/* empty css                                                             */
/* empty css                                                                  */
import "../../../../../../../manager/src/pluginManager.js";
/* empty css                                                           */
import "vuedraggable";
import "../../../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import { useElementDrag as V, useElementZoom as X } from "../../../../../../../hooks/src/common/element.js";
import { useElementSize as Y, watchOnce as ee, useResizeObserver as W } from "@vueuse/core";
import { useStore as te } from "../../../../../../../hooks/src/store/index.js";
import { useDesignerContext as ne } from "../../../../../../../hooks/src/designer/useDesignerContext.js";
import le from "./toolbar.vue.js";
const ae = { class: "relative flex h-full flex-col" }, oe = ["draggable"], be = /* @__PURE__ */ L({
  __name: "editScreenContainer",
  setup(ie) {
    const { pageSchema: w, props: b } = ne(), a = s(null), d = s(null), h = s(null), { disabledZoom: k, pressSpace: M } = te(), { handleElementDrag: T, handleElementDragEnd: y, handleElementDragStart: Z } = V(a), { height: C, width: m } = Y(a), { canvasScale: E, handleZoom: R } = X(d), c = P(() => M.value && b.draggable);
    let v = 0, f = 0;
    const B = s({}), z = s({}), p = s({}), l = U({ height: 0, width: 0 });
    q(N), F(() => {
      _(
        p,
        () => {
          $(A);
        },
        { immediate: !0 }
      );
    });
    function A() {
      h.value && (l.value = {
        height: h.value.clientHeight,
        width: h.value.clientWidth
      });
    }
    function N() {
      var t, e;
      p.value = {
        height: ((t = w.canvas) == null ? void 0 : t.height) ?? "0",
        width: ((e = w.canvas) == null ? void 0 : e.width) ?? "0"
      };
    }
    ee(m, S), _(l, (t, e) => {
      t.width === e.width && t.height === e.height || S();
    });
    function S() {
      const t = l.value.width || v, e = l.value.height || f;
      B.value = b.draggable ? {
        height: `${C.value + e}px`,
        // 滚动容器高度
        width: `${m.value + t}px`
        // 滚动容器宽度
      } : {
        height: `${C.value}px`,
        // 滚动容器高度
        width: `${m.value}px`
        // 滚动容器宽度
      }, z.value = {
        height: `${e}px`,
        // 画布高度
        width: `${t}px`
        // 画布宽度
      }, O();
    }
    function O() {
      $(() => {
        var g;
        const t = l.value.width || v, n = (l.value.height || f) / 2, o = t / 2;
        (g = a.value) == null || g.scrollTo(o, n);
      });
    }
    W(a, D), W(d, D);
    const j = Q(S, 10);
    function D() {
      var e;
      if (!a.value) return;
      const t = a.value.getBoundingClientRect();
      if (v = t.width - 60, f = t.height - 80, !k.value)
        if (l.value.width === 0)
          E.value = 1;
        else {
          const n = (v - 20) / l.value.width, o = f / (l.value.height || ((e = d.value) == null ? void 0 : e.clientHeight) || 1), g = Math.abs(n - 1) < Math.abs(o - 1);
          let i = 1;
          g ? i = n : i = o, (n < 1 || o < 1) && (i = Math.min(n, o)), i < 1.4 && i > 0.5 && (E.value = i);
        }
      j();
    }
    return (t, e) => (I(), G("div", ae, [
      J(le),
      u("div", {
        ref_key: "editScreenContainerRef",
        ref: a,
        class: H(["epic-edit-screen-container flex-1 overflow-auto overflow-y-hidden", { "cursor-grab": c.value }]),
        draggable: c.value,
        onWheelPassive: e[0] || (e[0] = //@ts-ignore
        (...n) => r(R) && r(R)(...n)),
        onDragstart: e[1] || (e[1] = (n) => r(Z)(n, c.value)),
        onDrag: e[2] || (e[2] = (n) => r(T)(n, c.value)),
        onDragend: e[3] || (e[3] = //@ts-ignore
        (...n) => r(y) && r(y)(...n))
      }, [
        u("div", {
          id: "canvasContainer",
          class: "flex items-center justify-center",
          style: x(B.value)
        }, [
          u("div", {
            ref_key: "draggableElRef",
            ref: d,
            class: "transition-all"
          }, [
            u("div", {
              class: H({ "pointer-events-none": c.value }),
              style: x(z.value)
            }, [
              K(t.$slots, "default")
            ], 6)
          ], 512)
        ], 4)
      ], 42, oe),
      u("div", {
        ref_key: "sizeBoxRef",
        ref: h,
        class: "pointer-events-none absolute top-0",
        style: x(p.value)
      }, null, 4)
    ]));
  }
});
export {
  be as default
};
