// epic-designer/dist/core/src/components/designer/src/modules/editContainer/previewWidgets.vue.js
import { defineComponent as de, ref as y, computed as k, watch as H, createElementBlock as Q, openBlock as W, Fragment as fe, withDirectives as F, createElementVNode as m, normalizeClass as pe, createCommentVNode as ve, toDisplayString as me, unref as a, createVNode as D, vShow as X } from "vue";
/* empty css                                                 */
import I from "../../../../../../../ui-kit/base-ui/src/icon/icon.vue.js";
/* empty css                                                             */
/* empty css                                                                  */
import { findSchemaInfoById as ge } from "../../../../../../../utils/src/common/data.js";
import "../../../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import { useTimedQuery as he } from "../../../../../../../hooks/src/common/element.js";
import { useResizeObserver as Y } from "@vueuse/core";
import { useStore as be } from "../../../../../../../hooks/src/store/index.js";
import { useDesignerContext as ye } from "../../../../../../../hooks/src/designer/useDesignerContext.js";
import { usePageManager as we } from "../../../../../../../hooks/src/designer/usePageManager.js";
import { pluginManager as A } from "../../../../../../../manager/src/pluginManager.js";
/* empty css                                                           */
import "vuedraggable";
const xe = { class: "epic-widget-action-item whitespace-nowrap" }, Se = {
  key: 0,
  class: "flex items-center"
}, He = /* @__PURE__ */ de({
  __name: "previewWidgets",
  setup(Ce, { expose: Z }) {
    const x = we(), l = ye(), j = l.props, B = l.pageSchema, g = y(null), h = y(null), f = y(null), R = y(!1), T = y(!0), S = y("top"), { canvasScale: _, disabledZoom: L } = be();
    let o = null;
    const q = k(() => {
      const e = l.state.selectedNode;
      return !(!(e != null && e.id) || j.lockDefaultSchemaEdit && x.defaultComponentIds.value.includes(e == null ? void 0 : e.id));
    }), M = (e) => {
      var u, i;
      const t = e.id;
      if ((u = e.props) != null && u.hidden)
        return null;
      const r = A.component.getConfigByType(e.type) ?? null;
      if (!t || !x.findInstance(t))
        return null;
      if (r != null && r.defaultSchema.input && (e == null ? void 0 : e.noFormItem) !== !0)
        return (i = x.findInstance(`${t}_formItem`)) == null ? void 0 : i.vnode.el;
      const n = x.findInstance(t), c = n == null ? void 0 : n.vnode.el;
      return !c || !c.getBoundingClientRect ? null : c;
    }, C = k(() => l.state.selectedNode ? M(l.state.selectedNode) : null), O = k(() => !l.state.hoverNode || l.state.hoverNode.id === B.schemas[0].id ? null : M(l.state.hoverNode)), { mutationObserver: G, observerConfig: J } = z(w), { startTimedQuery: K, stopTimedQuery: U } = he(w);
    H(
      () => C.value,
      (e) => {
        if (e) {
          R.value = !0, G.observe(e, J);
          const t = e.parentNode;
          t && (t.addEventListener("dragstart", () => {
            T.value = !1, K();
          }), t.addEventListener("dragend", () => {
            T.value = !0, U();
          })), w();
        } else
          R.value = !1;
      }
    );
    const {
      mutationObserver: ee,
      observerConfig: te
    } = z($);
    H(
      () => O.value,
      (e) => {
        e && (ee.observe(e, te), $());
      }
    );
    let V = 0, E = 0;
    function w() {
      var N;
      const e = C.value;
      if (!e || !o) return;
      const { left: t, top: r } = o.getBoundingClientRect();
      let n = (N = e.getBoundingClientRect) == null ? void 0 : N.call(e);
      !n && e.nextElementSibling && (n = e.nextElementSibling.getBoundingClientRect());
      const { height: c, left: u, top: i, width: d } = n, s = L.value ? 1 : _.value, p = i - r + ((o == null ? void 0 : o.scrollTop) ?? 0) * s, b = u - t + ((o == null ? void 0 : o.scrollLeft) ?? 0) * s, v = c / s;
      g.value && (g.value.style.width = `${d / s}px`, g.value.style.height = `${v}px`, g.value.style.top = `${p / s}px`, g.value.style.left = `${b / s}px`, oe(p, b)), f.value && (p < 45 && v < 800 ? (f.value.style.top = "", f.value.style.bottom = "-32px", f.value.style["border-radius"] = "4px", S.value = "bottom") : p < 45 ? (f.value.style.top = "0px", f.value.style["border-radius"] = "0px 0px 4px 0", S.value = "center") : (f.value.style.top = "-32px", f.value.style["border-radius"] = "4px", S.value = "top"));
    }
    function oe(e, t) {
      var P;
      if (B.schemas[0].id === ((P = l.state.selectedNode) == null ? void 0 : P.id)) return;
      const r = C.value;
      if (!o || !r) return;
      const n = o.getBoundingClientRect(), { height: c, width: u } = r.getBoundingClientRect(), i = L.value ? 1 : _.value, d = e / i - n.top;
      let s = t / i - n.left + u / i;
      s < n.width && (s = 0);
      const p = e + c, b = t + u;
      if (Math.abs(d - V) < 10 && Math.abs(s - E) < 10)
        return;
      V = d, E = s;
      const v = o.scrollTop, N = o.scrollTop + n.height, se = o.scrollLeft, re = o.scrollLeft + n.width, ie = e / i, ae = p / i, ce = t / i, ue = b / i;
      ae > v && // 组件底部在容器顶部之下
      ie < N && // 组件顶部在容器底部之上
      ue > se && // 组件右侧在容器左侧之右
      ce < re || (o.scrollTop = d, o.scrollLeft = s);
    }
    function $() {
      var v;
      const e = O.value;
      if (!e || !o) return;
      const { left: t, top: r } = o.getBoundingClientRect();
      let n = (v = e.getBoundingClientRect) == null ? void 0 : v.call(e);
      !n && e.nextElementSibling && (n = e.nextElementSibling.getBoundingClientRect());
      const { height: c, left: u, top: i, width: d } = n, s = L.value ? 1 : _.value, p = i - r + (o.scrollTop ?? 0) * s, b = u - t + (o.scrollLeft ?? 0) * s;
      h.value && (h.value.style.width = `${d / s}px`, h.value.style.height = `${c / s}px`, h.value.style.top = `${p / s}px`, h.value.style.left = `${b / s}px`);
    }
    function z(e) {
      const t = window.MutationObserver, r = {
        attributes: !0,
        childList: !0,
        subtree: !0
      };
      return {
        mutationObserver: new t(e),
        observerConfig: r
      };
    }
    function ne() {
      var r;
      const e = ge(
        B.schemas,
        ((r = l.state.selectedNode) == null ? void 0 : r.id) ?? "root"
      );
      if (!e)
        return !1;
      const { parentSchema: t } = e;
      l.setSelectedNode(t);
    }
    function le(e) {
      o = e, o == null || o.addEventListener("scroll", () => {
        w();
      }), Y(C, () => {
        setTimeout(w, 150);
      }), Y(O, $);
    }
    return Z({
      handleInit: le
    }), (e, t) => {
      var r, n, c, u, i;
      return W(), Q(fe, null, [
        F(m("div", {
          ref_key: "selectorRef",
          ref: g,
          class: pe(["epic-selected-widget z-999 pointer-events-none absolute", `${S.value} ${T.value ? "transition-all" : ""}`])
        }, [
          m("div", {
            ref_key: "actionBoxRef",
            ref: f,
            class: "epic-widget-action-box"
          }, [
            m("div", xe, me((n = a(A).component.getConfigByType(
              ((r = a(l).state.selectedNode) == null ? void 0 : r.type) ?? ""
            )) == null ? void 0 : n.defaultSchema.label), 1),
            q.value ? (W(), Q("div", Se, [
              m("div", {
                title: "选择父节点",
                class: "epic-widget-action-item pointer-events-auto",
                onClick: ne
              }, [
                D(a(I), { name: "icon--epic--upward" })
              ]),
              m("div", {
                title: "复制",
                class: "epic-widget-action-item pointer-events-auto",
                onClick: t[0] || (t[0] = //@ts-ignore
                (...d) => a(l).handleDuplicate && a(l).handleDuplicate(...d))
              }, [
                D(a(I), { name: "icon--epic--copy-all-outline-rounded" })
              ]),
              m("div", {
                title: "删除",
                class: "epic-widget-action-item pointer-events-auto",
                onClick: t[1] || (t[1] = //@ts-ignore
                (...d) => a(l).handleDelete && a(l).handleDelete(...d))
              }, [
                D(a(I), { name: "icon--epic--delete-outline-rounded" })
              ])
            ])) : ve("", !0)
          ], 512)
        ], 2), [
          [X, R.value && ((c = a(l).state.selectedNode) == null ? void 0 : c.id) !== "root"]
        ]),
        F(m("div", {
          ref_key: "hoverWidgetRef",
          ref: h,
          class: "epic-hover-widget z-998 pointer-events-none absolute"
        }, null, 512), [
          [
            X,
            a(l).state.hoverNode && ((u = a(l).state.selectedNode) == null ? void 0 : u.id) !== ((i = a(l).state.hoverNode) == null ? void 0 : i.id)
          ]
        ])
      ], 64);
    };
  }
});
export {
  He as default
};
