// epic-designer/dist/core/src/components/designer/src/modules/editContainer/toolbar.vue.js
import { defineComponent as B, ref as w, computed as v, createElementBlock as l, openBlock as r, Fragment as f, createElementVNode as c, createVNode as s, withDirectives as H, renderList as C, createCommentVNode as h, unref as a, withCtx as I, normalizeClass as S, vShow as K } from "vue";
/* empty css                                                 */
import M from "../../../../../../../ui-kit/base-ui/src/icon/icon.vue.js";
/* empty css                                                             */
/* empty css                                                                  */
import { convertKFormData as P, migrateComponentProps as Z, deepCompareAndModify as q } from "../../../../../../../utils/src/common/data.js";
import "../../../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import "@vueuse/core";
import { useStore as G } from "../../../../../../../hooks/src/store/index.js";
import { useDesignerContext as Q } from "../../../../../../../hooks/src/designer/useDesignerContext.js";
import { pluginManager as W } from "../../../../../../../manager/src/pluginManager.js";
import X from "../../../../../../../ui-kit/base-ui/src/tooltip/tooltip.vue.js";
import "vuedraggable";
import Y from "./previewJson.vue.js";
const ee = { class: "epic-edit-toolbar flex items-center justify-between px-2" }, te = { class: "flex h-full flex-1 items-center" }, oe = {
  key: 0,
  class: "epic-divider"
}, ne = ["onClick"], ie = { class: "flex h-full flex-1 items-center justify-end" }, le = {
  key: 0,
  class: "ml-12px flex items-center"
}, re = { class: "pr-8px w-82px cursor-pointer" }, ce = { class: "epic-device h-28px p-2px flex items-center gap-1 rounded-md border" }, se = ["title", "onClick"], Ce = /* @__PURE__ */ B({
  __name: "toolbar",
  setup(ae) {
    const F = W.component.get("select"), { canvasScale: x, disabledZoom: N } = G(), n = Q(), d = n.pageSchema, p = n.revoke, b = w(null), O = [
      {
        icon: "icon--epic--computer-outline-rounded",
        key: "desktop",
        title: "pc"
      },
      {
        icon: "icon--epic--tablet-android-outline-rounded",
        key: "tablet",
        title: "平板"
      },
      {
        icon: "icon--epic--phone-iphone-outline-sharp",
        key: "mobile",
        title: "手机"
      }
    ], D = v(() => [
      {
        icon: "icon--epic--code",
        on: R,
        title: "查看数据"
      },
      {
        icon: "icon--epic--exit-to-app-rounded",
        on: T,
        title: "导入数据"
      },
      {
        icon: "icon--epic--trash",
        on: n.reset,
        title: "清空"
      },
      {
        disabled: p.recordList.value.length === 0,
        divider: !0,
        icon: "icon--epic--undo",
        on: J,
        title: "撤销"
      },
      {
        disabled: p.undoList.value.length === 0,
        icon: "icon--epic--redo",
        on: L,
        title: "重做"
      },
      {
        divider: !0,
        icon: "icon--epic--eye",
        on: n.preview,
        show: () => n.props.hiddenHeader,
        title: "预览"
      },
      {
        icon: "icon--epic--save-outline-rounded",
        on: n.save,
        show: () => n.props.hiddenHeader,
        title: "保存"
      }
    ]), u = {
      mobile: "mobile",
      pad: "tablet",
      pc: "desktop"
    }, V = {
      desktop: {},
      mobile: {
        mode: "mobile",
        width: "390px"
      },
      // 保留旧配置以兼容旧数据
      pad: {
        mode: "tablet",
        width: "780px"
      },
      pc: {
        mode: "desktop"
      },
      tablet: {
        mode: "tablet",
        width: "780px"
      }
    };
    function $(t) {
      return t ? t in u ? u[t] : ["desktop", "mobile", "tablet"].includes(t) ? t : "desktop" : "desktop";
    }
    const k = v({
      get() {
        var o;
        const t = (o = d.canvas) == null ? void 0 : o.mode;
        return $(t);
      },
      set(t) {
        n.handleToggleDeviceMode(t), d.canvas = V[t];
      }
    }), _ = w(null), m = v({
      get() {
        return `${(x.value * 100).toFixed(0)}%`;
      },
      set(t) {
        const o = Number.parseFloat(t);
        x.value = o / 100;
      }
    }), E = [
      {
        label: "60%",
        value: "60%"
      },
      {
        label: "80%",
        value: "80%"
      },
      {
        label: "100%",
        value: "100%"
      },
      {
        label: "120%",
        value: "120%"
      },
      {
        label: "140%",
        value: "140%"
      }
    ];
    function g(t) {
      return typeof t == "function" ? t() : !0;
    }
    function J() {
      p.undo();
    }
    function L() {
      p.redo();
    }
    function R() {
      b.value.handleOpen();
    }
    function T() {
      var t;
      (t = _.value) == null || t.click();
    }
    function j(t) {
      var i;
      const o = (i = t.target.files) == null ? void 0 : i[0];
      if (!o) return;
      t.target.value = null;
      const e = new FileReader();
      e.readAsText(o), e.addEventListener("load", (A) => {
        var y;
        z((y = A.target) == null ? void 0 : y.result);
      });
    }
    function z(t) {
      var o;
      try {
        let e = JSON.parse(t ?? "");
        if (e.schemas || (e = P(e)), (o = e.canvas) != null && o.mode) {
          const i = e.canvas.mode;
          i in u && (e.canvas.mode = u[i]);
        }
        Z(e), q(d, e), n.handleImported(e), n.setSelectedNode(d.schemas[0]);
      } catch (e) {
        console.error(e);
      }
    }
    function U(t) {
      n.handleToggleDeviceMode(t), k.value = t;
    }
    return (t, o) => (r(), l(f, null, [
      c("div", ee, [
        c("div", te, [
          (r(!0), l(f, null, C(D.value, (e, i) => (r(), l(f, { key: i }, [
            e.divider && g(e.show) ? (r(), l("div", oe)) : h("", !0),
            s(a(X), {
              content: e.title
            }, {
              default: I(() => [
                g(e.show) ? (r(), l("div", {
                  key: 0,
                  class: S(["epic-action-item h-90% px-10px flex cursor-pointer items-center text-base", { disabled: e.disabled }]),
                  onClick: e.on
                }, [
                  s(a(M), {
                    name: e.icon
                  }, null, 8, ["name"])
                ], 10, ne)) : h("", !0)
              ]),
              _: 2
            }, 1032, ["content"])
          ], 64))), 128))
        ]),
        H(c("input", {
          ref_key: "fileRef",
          ref: _,
          type: "file",
          accept: ".json,.txt",
          onChange: j
        }, null, 544), [
          [K, !1]
        ]),
        c("div", ie, [
          a(N) ? h("", !0) : (r(), l("div", le, [
            c("div", re, [
              s(a(F), {
                value: m.value,
                "onUpdate:value": o[0] || (o[0] = (e) => m.value = e),
                modelValue: m.value,
                "onUpdate:modelValue": o[1] || (o[1] = (e) => m.value = e),
                options: E,
                size: "small"
              }, null, 8, ["value", "modelValue"])
            ])
          ])),
          c("div", ce, [
            (r(), l(f, null, C(O, (e) => c("div", {
              key: e.key,
              title: e.title,
              class: S(["epic-device-item rounded-$ep-radius flex h-full cursor-pointer items-center px-1 text-base transition-colors", { checked: e.key === k.value }]),
              onClick: (i) => U(e.key)
            }, [
              s(a(M), {
                name: e.icon
              }, null, 8, ["name"])
            ], 10, se)), 64))
          ])
        ])
      ]),
      s(Y, {
        ref_key: "previewJson",
        ref: b
      }, null, 512)
    ], 64));
  }
});
export {
  Ce as default
};
