// epic-designer/dist/core/src/components/builder/src/builder.vue.js
import { defineComponent as L, ref as Y, watch as m, computed as g, provide as h, useSlots as $, getCurrentInstance as z, createElementBlock as _, createBlock as p, unref as r, openBlock as a, createVNode as v, Suspense as O, withCtx as y, createElementVNode as B, createCommentVNode as T, Fragment as U, renderList as j, resolveDynamicComponent as q } from "vue";
/* empty css                                           */
import C from "../../../../../ui-kit/base-ui/src/asyncLoader/baseLoader.vue.js";
/* empty css                                                       */
/* empty css                                                            */
import G from "../../../../../ui-kit/base-ui/src/node/node.vue.js";
/* empty css                                                     */
import { migrateComponentProps as H, deepCompareAndModify as J, findSchemas as Q } from "../../../../../utils/src/common/data.js";
import "../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import { createEventBus as W } from "../../../../../hooks/src/common/createEventBus.js";
import "@vueuse/core";
import "../../../../../hooks/src/store/index.js";
import { provideBuilderDisabled as X } from "../../../../../hooks/src/designer/useBuiderDisabled.js";
import { BUILDER_KEY as Z } from "../../../../../hooks/src/designer/useBuilderContext.js";
import { providePageManager as w } from "../../../../../hooks/src/designer/usePageManager.js";
import { FORM_INSTANCES_KEY as ee } from "../../../../../hooks/src/logic/useForm.js";
import { pluginManager as c } from "../../../../../manager/src/pluginManager.js";
import "vuedraggable";
import { setupPage as te } from "../../../../../ui-kit/panel-ui/src/components/index.js";
/* empty css                                            */
import { useBuilder as oe } from "../hooks/useBuilder.js";
const re = {
  key: 0,
  class: "epic-loading-box"
}, ae = { class: "epic-builder-main ep-scoped" }, ie = { class: "epic-loading-box" }, De = /* @__PURE__ */ L({
  __name: "builder",
  props: {
    disabled: { type: Boolean },
    fieldStates: {},
    formData: {},
    pageSchema: {}
  },
  emits: ["change", "ready"],
  setup(M, { expose: S, emit: b }) {
    const i = M, d = b;
    te(c);
    const l = c.component.get("epBuilderSlot"), {
      formInstances: k,
      getData: E,
      getFormInstance: D,
      getForms: R,
      pageManager: t,
      ready: s,
      resetData: x,
      setData: u,
      setForms: I,
      validate: A,
      validateAll: F
    } = oe(), f = Y(0);
    m(
      () => i.pageSchema,
      (e) => {
        var o;
        (o = e == null ? void 0 : e.schemas) != null && o.length && (H(e, !0), J(t.pageSchema, e), t.mountMonitor.reset(), f.value++, s.value = !1);
      },
      {
        deep: !0,
        immediate: !0
      }
    ), m(
      () => i.formData,
      (e) => {
        e && u(e);
      },
      {
        deep: !0,
        immediate: !0
      }
    ), W(), X(g(() => i.disabled)), h(Z, {
      fieldStateMap: g(() => {
        var o;
        const e = {};
        return (o = i.fieldStates) == null || o.forEach((n) => {
          e[n.field] = n;
        }), e;
      }),
      slots: $()
    }), w(t), h(ee, k);
    function N() {
      const e = m(
        () => t.mountMonitor.isAllMounted.value,
        (o) => {
          o && (e && e(), K());
        }
      );
    }
    function K() {
      s.value = !0, d("ready", t), Q(t.pageSchema.schemas, (e) => {
        var o;
        return (o = e.on) != null && o.epicReady && t.doActions(e.on.epicReady), !1;
      });
    }
    const P = z();
    return t.addComponentInstance("builder", P), t.hook.register("formChange", (e) => {
      d("change", e);
    }), S({
      getData: E,
      getFormInstance: D,
      getForms: R,
      pageManager: t,
      ready: s,
      resetData: x,
      setData: u,
      setForms: I,
      validate: A,
      validateAll: F
    }), (e, o) => !r(c).designer.initialized.value || r(t).pageSchema.schemas.length === 0 ? (a(), _("div", re, [
      v(r(C))
    ])) : (a(), p(O, {
      key: f.value,
      onResolve: N
    }, {
      default: y(() => [
        B("div", ae, [
          (a(!0), _(U, null, j(r(t).pageSchema.schemas, (n, V) => (a(), p(r(G), {
            key: V,
            "component-schema": n
          }, null, 8, ["component-schema"]))), 128)),
          r(l) ? (a(), p(q(r(l)), { key: 0 })) : T("", !0)
        ])
      ]),
      fallback: y(() => [
        B("div", ie, [
          v(r(C))
        ])
      ]),
      _: 1
    }));
  }
});
export {
  De as default
};
