// @epic-designer/element-plus/dist/chunks/card.mjs
import { defineComponent, renderSlot, h } from 'vue';
import { ElCard } from 'element-plus';

const card = defineComponent({
  props: {
    componentSchema: {
      default: () => ({}),
      required: true,
      type: Object
    }
  },
  setup(props, { slots }) {
    return () => {
      var _a, _b, _c;
      const componentSchema = {
        ...props.componentSchema,
        header: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : ""
      };
      const children = (_c = componentSchema.children) != null ? _c : [];
      delete componentSchema.children;
      let vNodeClildren = null;
      vNodeClildren = children.length > 0 ? () => children.map(
        (node) => renderSlot(slots, "node", { componentSchema: node })
      ) : () => [renderSlot(slots, "default")];
      return h(ElCard, componentSchema, {
        default: () => renderSlot(slots, "edit-node", {}, vNodeClildren),
        header: () => renderSlot(slots, "header")
      });
    };
  }
});

export { card as default };
