// @epic-designer/element-plus/dist/chunks/form.mjs
import { defineComponent, ref, onMounted, computed, createBlock, openBlock, unref, mergeProps, withCtx, renderSlot, createElementBlock, Fragment, renderList } from 'vue';
import { useForm } from 'epic-designer';
import { ElForm } from 'element-plus';

const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "form",
  props: {
    componentSchema: {
      default: () => ({}),
      require: true,
      type: Object
    }
  },
  setup(__props, { expose: __expose }) {
    var _a, _b, _c;
    const props = __props;
    const form = ref(null);
    const { formData, formInstances } = useForm(
      (_c = (_b = (_a = props.componentSchema) == null ? void 0 : _a.props) == null ? void 0 : _b.name) != null ? _c : "default"
    );
    function getData() {
      return formData;
    }
    function setData(data) {
      Object.assign(formData, data);
    }
    function resetData() {
      var _a2;
      (_a2 = form.value) == null ? void 0 : _a2.resetFields();
    }
    function validate() {
      var _a2;
      return (_a2 = form.value) == null ? void 0 : _a2.validate();
    }
    onMounted(async () => {
      var _a2, _b2, _c2, _d, _e, _f;
      if (((_a2 = props.componentSchema) == null ? void 0 : _a2.type) === "form" && formInstances.value && form.value) {
        const name = (_f = (_e = (_c2 = (_b2 = props.componentSchema) == null ? void 0 : _b2.props) == null ? void 0 : _c2.name) != null ? _e : (_d = props.componentSchema) == null ? void 0 : _d.name) != null ? _f : "default";
        formInstances.value[name] = form.value;
        form.value.getData = getData;
        form.value.setData = setData;
        form.value.resetData = resetData;
        return false;
      }
    });
    const formProps = computed(() => {
      const recordProps = props.componentSchema.props;
      let labelCol = recordProps.labelCol;
      let wrapperCol = recordProps.wrapperCol;
      if (recordProps.labelLayout === "fixed") {
        labelCol = {
          style: `width:${typeof recordProps.labelWidth === "number" ? `${recordProps.labelWidth}px` : recordProps.labelWidth}`
        };
        wrapperCol = { style: "width:auto;flex:1" };
      }
      return {
        ...recordProps,
        labelCol,
        wrapperCol
      };
    });
    const children = computed(() => {
      var _a2;
      return (_a2 = props.componentSchema.children) != null ? _a2 : [];
    });
    __expose({
      form,
      getData,
      resetData,
      setData,
      validate
    });
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(ElForm), mergeProps({
        ref_key: "form",
        ref: form,
        model: unref(formData)
      }, formProps.value), {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "edit-node", {}, () => [
            (openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList(children.value, (item) => {
                return renderSlot(_ctx.$slots, "node", { componentSchema: item });
              }),
              256
              /* UNKEYED_FRAGMENT */
            ))
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 16, ["model"]);
    };
  }
});

export { _sfc_main as default };
