// @epic-designer/element-plus/dist/chunks/modal.mjs
import { defineComponent, useAttrs, computed, createBlock, openBlock, unref, mergeProps, withCtx, renderSlot, createElementBlock, createCommentVNode, Fragment, renderList, createElementVNode, createVNode, createTextVNode, toDisplayString } from 'vue';
import { ElDialog, ElButton } from 'element-plus';

const _hoisted_1 = { class: "ep-el-modal-footer" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "modal",
  props: {
    cancelText: { type: String, required: false, default: "\u5173\u95ED" },
    componentSchema: { type: Object, required: false, default: () => ({
      type: "modal"
    }) },
    hideConfirm: { type: Boolean, required: false },
    okText: { type: String, required: false, default: "\u786E\u5B9A" }
  },
  emits: ["ok", "close", "update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const attrs = useAttrs();
    const getProps = computed(() => {
      var _a, _b;
      return {
        ...props.componentSchema,
        ...attrs,
        class: "ep-el-modal ep-scoped",
        "destroy-on-close": true,
        title: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : ""
      };
    });
    const children = computed(() => {
      var _a, _b;
      return (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    });
    function handleOk() {
      emits("ok");
    }
    function updateModelValue(value = false) {
      emits("update:modelValue", value);
    }
    function handleClose() {
      emits("close");
    }
    return (_ctx, _cache) => {
      var _a;
      return openBlock(), createBlock(unref(ElDialog), mergeProps(getProps.value, {
        title: (_a = getProps.value.label) != null ? _a : "",
        "destroy-on-close": "",
        "onUpdate:modelValue": updateModelValue
      }), {
        footer: withCtx(() => [
          createElementVNode("div", _hoisted_1, [
            createVNode(unref(ElButton), { onClick: handleClose }, {
              default: withCtx(() => [
                createTextVNode(
                  toDisplayString(props.cancelText),
                  1
                  /* TEXT */
                )
              ]),
              _: 1
              /* STABLE */
            }),
            !props.hideConfirm ? (openBlock(), createBlock(unref(ElButton), {
              key: 0,
              type: "primary",
              onClick: handleOk
            }, {
              default: withCtx(() => [
                createTextVNode(
                  toDisplayString(props.okText),
                  1
                  /* TEXT */
                )
              ]),
              _: 1
              /* STABLE */
            })) : createCommentVNode("v-if", true)
          ])
        ]),
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "default", {}, () => [
            renderSlot(_ctx.$slots, "edit-node", {}, () => [
              children.value.length > 0 ? (openBlock(true), createElementBlock(
                Fragment,
                { key: 0 },
                renderList(children.value, (node) => {
                  return renderSlot(_ctx.$slots, "node", { componentSchema: node });
                }),
                256
                /* UNKEYED_FRAGMENT */
              )) : createCommentVNode("v-if", true)
            ])
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 16, ["title"]);
    };
  }
});

export { _sfc_main as default };
