// @epic-designer/element-plus/dist/chunks/modal.cjs
'use strict';

const vue = require('vue');
const elementPlus = require('element-plus');

const _hoisted_1 = { class: "ep-el-modal-footer" };
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "modal",
  props: {
    cancelText: { type: String, required: false, default: "\u5173\u95ED" },
    componentSchema: { type: Object, required: false, default: () => ({
      type: "modal"
    }) },
    hideConfirm: { type: Boolean, required: false },
    okText: { type: String, required: false, default: "\u786E\u5B9A" }
  },
  emits: ["ok", "close", "update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const attrs = vue.useAttrs();
    const getProps = vue.computed(() => {
      var _a, _b;
      return {
        ...props.componentSchema,
        ...attrs,
        class: "ep-el-modal ep-scoped",
        "destroy-on-close": true,
        title: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : ""
      };
    });
    const children = vue.computed(() => {
      var _a, _b;
      return (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    });
    function handleOk() {
      emits("ok");
    }
    function updateModelValue(value = false) {
      emits("update:modelValue", value);
    }
    function handleClose() {
      emits("close");
    }
    return (_ctx, _cache) => {
      var _a;
      return vue.openBlock(), vue.createBlock(vue.unref(elementPlus.ElDialog), vue.mergeProps(getProps.value, {
        title: (_a = getProps.value.label) != null ? _a : "",
        "destroy-on-close": "",
        "onUpdate:modelValue": updateModelValue
      }), {
        footer: vue.withCtx(() => [
          vue.createElementVNode("div", _hoisted_1, [
            vue.createVNode(vue.unref(elementPlus.ElButton), { onClick: handleClose }, {
              default: vue.withCtx(() => [
                vue.createTextVNode(
                  vue.toDisplayString(props.cancelText),
                  1
                  /* TEXT */
                )
              ]),
              _: 1
              /* STABLE */
            }),
            !props.hideConfirm ? (vue.openBlock(), vue.createBlock(vue.unref(elementPlus.ElButton), {
              key: 0,
              type: "primary",
              onClick: handleOk
            }, {
              default: vue.withCtx(() => [
                vue.createTextVNode(
                  vue.toDisplayString(props.okText),
                  1
                  /* TEXT */
                )
              ]),
              _: 1
              /* STABLE */
            })) : vue.createCommentVNode("v-if", true)
          ])
        ]),
        default: vue.withCtx(() => [
          vue.renderSlot(_ctx.$slots, "default", {}, () => [
            vue.renderSlot(_ctx.$slots, "edit-node", {}, () => [
              children.value.length > 0 ? (vue.openBlock(true), vue.createElementBlock(
                vue.Fragment,
                { key: 0 },
                vue.renderList(children.value, (node) => {
                  return vue.renderSlot(_ctx.$slots, "node", { componentSchema: node });
                }),
                256
                /* UNKEYED_FRAGMENT */
              )) : vue.createCommentVNode("v-if", true)
            ])
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 16, ["title"]);
    };
  }
});

exports.default = _sfc_main;
