// @epic-designer/element-plus/dist/chunks/uploadImage.mjs
import { defineComponent, ref, watch, computed, h } from 'vue';
import { getUUID, pluginManager } from 'epic-designer';
import { ElUpload, ElImageViewer, ElMessage } from 'element-plus';

const uploadImage = defineComponent({
  emits: ["update:modelValue", "change"],
  props: {
    modelValue: {
      default: "",
      type: String
    }
  },
  setup(props, { attrs, emit }) {
    const fileList = ref([]);
    let urlString = "";
    const imgUrl = ref("");
    const visible = ref(false);
    const setVisible = (value) => {
      visible.value = value;
    };
    watch(
      () => fileList.value,
      (list) => {
        urlString = list.filter((file) => file.status === "success").map((file) => file.url).join(",");
        emit("update:modelValue", urlString);
        emit("change", urlString);
      }
    );
    watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue && fileList.value) {
          fileList.value = modelValue.split(",").map((url) => ({
            name: url,
            status: "success",
            uid: Number(getUUID(10, "number")),
            url
          }));
        }
      },
      { deep: true, immediate: true }
    );
    const handleRemove = (_file, uploadFiles) => {
      fileList.value = uploadFiles;
    };
    const handlePreview = (uploadFile) => {
      if (!uploadFile.url) return;
      imgUrl.value = uploadFile.url;
      setVisible(true);
    };
    const handleSuccess = (response, uploadFile, uploadFiles) => {
      fileList.value = uploadFiles.map((file) => {
        var _a, _b;
        return {
          ...file,
          url: (_b = (_a = file.response.data) == null ? void 0 : _a.url) != null ? _b : file.response.data
        };
      });
    };
    const handleError = (error) => {
      ElMessage.error("\u4E0A\u4F20\u5931\u8D25");
      console.error(error);
    };
    const beforeUpload = () => {
    };
    const getUploadProps = computed(() => {
      var _a, _b;
      return {
        ...attrs,
        accept: "image/gif,image/jpeg,image/jpg,image/png,image/svg",
        action: (_a = pluginManager.global.uploadImage) != null ? _a : attrs.action,
        headers: (_b = pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
        onBeforeUpload: beforeUpload,
        onError: handleError,
        onPreview: handlePreview,
        onRemove: handleRemove,
        onSuccess: handleSuccess,
        "file-list": fileList.value,
        "list-type": "picture-card"
      };
    });
    return () => {
      return h(
        "div",
        {
          class: "epic-upload-image"
        },
        {
          default: () => [
            h(ElUpload, getUploadProps.value, {
              default: () => [
                h(
                  "div",
                  { style: { "text-align": "center" } },
                  {
                    default: () => [
                      h("span", {
                        class: "icon--epic icon--epic--cloud-upload-outlined mr-2px text-lg"
                      }),
                      h(
                        "div",
                        { class: "ant-upload-text" },
                        { default: () => "\u70B9\u51FB\u4E0A\u4F20" }
                      )
                    ]
                  }
                )
              ]
            }),
            (() => {
              if (!visible.value) {
                return;
              }
              return h(ElImageViewer, {
                onClose: () => {
                  setVisible(false);
                },
                urlList: [imgUrl.value]
              });
            })()
          ]
        }
      );
    };
  }
});

export { uploadImage as default };
