// @epic-designer/.element-plus-59EcLDdc/dist/chunks/card.cjs
'use strict';

const vue = require('vue');
const elementPlus = require('element-plus');

const card = vue.defineComponent({
  props: {
    componentSchema: {
      default: () => ({}),
      required: true,
      type: Object
    }
  },
  setup(props, { slots }) {
    return () => {
      var _a, _b, _c;
      const componentSchema = {
        ...props.componentSchema,
        header: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : ""
      };
      const children = (_c = componentSchema.children) != null ? _c : [];
      delete componentSchema.children;
      let vNodeClildren = null;
      vNodeClildren = children.length > 0 ? () => children.map(
        (node) => vue.renderSlot(slots, "node", { componentSchema: node })
      ) : () => [vue.renderSlot(slots, "default")];
      return vue.h(elementPlus.ElCard, componentSchema, {
        default: () => vue.renderSlot(slots, "edit-node", {}, vNodeClildren),
        header: () => vue.renderSlot(slots, "header")
      });
    };
  }
});

exports.default = card;
