// @epic-designer/.element-plus-59EcLDdc/dist/chunks/uploadFile.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const elementPlus = require('element-plus');

const uploadFile = vue.defineComponent({
  emits: ["update:modelValue", "change"],
  props: {
    modelValue: {
      default: "",
      type: String
    }
  },
  setup(props, { attrs, emit }) {
    const fileList = vue.ref([]);
    let urlString = "";
    vue.watch(
      () => fileList.value,
      (list) => {
        console.log("list", list);
        urlString = list.filter((file) => file.status === "success").map((file) => file.url).join(",");
        emit("update:modelValue", urlString);
        emit("change", urlString);
      }
    );
    vue.watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue !== null && fileList.value !== null) {
          fileList.value = modelValue.split(",").map((url) => ({
            id: epicDesigner.getUUID(),
            name: epicDesigner.getFileNameByUrl(url),
            status: "success",
            url
          }));
        }
      },
      { immediate: true }
    );
    const handleRemove = (_file, uploadFiles) => {
      fileList.value = uploadFiles;
    };
    const beforeUpload = () => {
    };
    const handleSuccess = (response, uploadFile, uploadFiles) => {
      fileList.value = uploadFiles.map((file) => {
        var _a, _b;
        return {
          ...file,
          url: (_b = (_a = file.response.data) == null ? void 0 : _a.url) != null ? _b : file.response.data
        };
      });
    };
    const handleError = (error) => {
      elementPlus.ElMessage.error("\u4E0A\u4F20\u5931\u8D25");
      console.error(error);
    };
    const getUploadProps = vue.computed(() => {
      var _a, _b;
      return {
        ...attrs,
        action: (_a = epicDesigner.pluginManager.global.uploadFile) != null ? _a : attrs.action,
        headers: (_b = epicDesigner.pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
        onBeforeUpload: beforeUpload,
        onError: handleError,
        onRemove: handleRemove,
        onSuccess: handleSuccess,
        "file-list": fileList.value
      };
    });
    return () => {
      return vue.h("div", null, {
        default: () => [
          vue.h(elementPlus.ElUpload, getUploadProps.value, {
            default: () => [
              vue.h(
                elementPlus.ElButton,
                {},
                {
                  default: () => [
                    vue.h("span", {
                      class: "icon--epic icon--epic--cloud-upload-outlined text-lg",
                      style: { "margin-right": "2px" }
                    }),
                    vue.h("span", null, { default: () => "\u4E0A\u4F20\u6587\u4EF6" })
                  ]
                }
              )
            ]
          })
        ]
      });
    };
  }
});

exports.default = uploadFile;
