// @epic-designer/.element-plus-59EcLDdc/dist/chunks/uploadFile.mjs
import { defineComponent, ref, watch, computed, h } from 'vue';
import { getFileNameByUrl, getUUID, pluginManager } from 'epic-designer';
import { ElUpload, ElButton, ElMessage } from 'element-plus';

const uploadFile = defineComponent({
  emits: ["update:modelValue", "change"],
  props: {
    modelValue: {
      default: "",
      type: String
    }
  },
  setup(props, { attrs, emit }) {
    const fileList = ref([]);
    let urlString = "";
    watch(
      () => fileList.value,
      (list) => {
        console.log("list", list);
        urlString = list.filter((file) => file.status === "success").map((file) => file.url).join(",");
        emit("update:modelValue", urlString);
        emit("change", urlString);
      }
    );
    watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue !== null && fileList.value !== null) {
          fileList.value = modelValue.split(",").map((url) => ({
            id: getUUID(),
            name: getFileNameByUrl(url),
            status: "success",
            url
          }));
        }
      },
      { immediate: true }
    );
    const handleRemove = (_file, uploadFiles) => {
      fileList.value = uploadFiles;
    };
    const beforeUpload = () => {
    };
    const handleSuccess = (response, uploadFile, uploadFiles) => {
      fileList.value = uploadFiles.map((file) => {
        var _a, _b;
        return {
          ...file,
          url: (_b = (_a = file.response.data) == null ? void 0 : _a.url) != null ? _b : file.response.data
        };
      });
    };
    const handleError = (error) => {
      ElMessage.error("\u4E0A\u4F20\u5931\u8D25");
      console.error(error);
    };
    const getUploadProps = computed(() => {
      var _a, _b;
      return {
        ...attrs,
        action: (_a = pluginManager.global.uploadFile) != null ? _a : attrs.action,
        headers: (_b = pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
        onBeforeUpload: beforeUpload,
        onError: handleError,
        onRemove: handleRemove,
        onSuccess: handleSuccess,
        "file-list": fileList.value
      };
    });
    return () => {
      return h("div", null, {
        default: () => [
          h(ElUpload, getUploadProps.value, {
            default: () => [
              h(
                ElButton,
                {},
                {
                  default: () => [
                    h("span", {
                      class: "icon--epic icon--epic--cloud-upload-outlined text-lg",
                      style: { "margin-right": "2px" }
                    }),
                    h("span", null, { default: () => "\u4E0A\u4F20\u6587\u4EF6" })
                  ]
                }
              )
            ]
          })
        ]
      });
    };
  }
});

export { uploadFile as default };
