// @epic-designer/.element-plus-59EcLDdc/dist/index.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const elementPlus = require('element-plus');

const Button = {
  component: () => import('./chunks/button.cjs'),
  config: {
    attribute: [
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.type",
        label: "\u7C7B\u578B",
        props: {
          clearable: true,
          options: [
            {
              label: "primary",
              value: "primary"
            },
            {
              label: "success",
              value: "success"
            },
            {
              label: "info",
              value: "info"
            },
            {
              label: "warning",
              value: "warning"
            },
            {
              label: "danger",
              value: "danger"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.native-type",
        label: "\u64CD\u4F5C\u7C7B\u578B",
        props: {
          clearable: true,
          options: [
            {
              label: "button",
              value: "button"
            },
            {
              label: "submit",
              value: "submit"
            },
            {
              label: "reset",
              value: "reset"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.color",
        label: "\u81EA\u5B9A\u4E49\u989C\u8272",
        props: {
          type: "color"
        },
        type: "color-picker"
      },
      {
        field: "props.plain",
        label: "\u6734\u7D20\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.round",
        label: "\u5706\u89D2\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.circle",
        label: "\u5706\u5F62\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.text",
        label: "\u6587\u5B57\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.loading",
        label: "\u52A0\u8F7D\u72B6\u6001",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ],
    event: [
      {
        description: "\u70B9\u51FB\u6309\u94AE\u65F6",
        type: "click"
      },
      {
        description: "\u53CC\u51FB\u6309\u94AE\u65F6",
        type: "dblclick"
      }
    ]
  },
  defaultSchema: {
    field: "input",
    input: false,
    label: "\u6309\u94AE",
    type: "button"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--fit-screen-rounded",
  sort: 1200
};

const Card = {
  component: () => import('./chunks/card.cjs'),
  config: {
    attribute: [
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.shadow",
        label: "\u9634\u5F71\u65F6\u673A",
        props: {
          clearable: true,
          options: [
            {
              label: "always",
              value: "always"
            },
            {
              label: "hover",
              value: "hover"
            },
            {
              label: "never",
              value: "never"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u5361\u7247\u5E03\u5C40",
    props: {},
    type: "card",
    children: []
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--wysiwyg-rounded",
  sort: 700
};

const Cascader = {
  component: async () => (await import('element-plus')).ElCascader,
  config: {
    action: [
      {
        description: "\u83B7\u53D6\u5F53\u524D\u9009\u4E2D\u8282\u70B9",
        type: "getSelectedNodes"
      },
      {
        description: "\u5207\u6362 popper \u53EF\u89C1\u72B6\u6001",
        type: "togglePopperVisible"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "cascader"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.separator",
        label: "\u5206\u5272\u7B26",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.props.expandTrigger",
        label: "\u6B21\u7EA7\u83DC\u5355\u5C55\u5F00\u65B9\u5F0F",
        props: {
          clearable: true,
          options: [
            {
              label: "click",
              value: "click"
            },
            {
              label: "hover",
              value: "hover"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.filterable",
        label: "\u53EF\u641C\u7D22",
        type: "switch"
      },
      {
        field: "props.props.multiple",
        label: "\u53EF\u591A\u9009",
        onChange: ({ value, values }) => {
          values.props.defaultValue = value ? [] : null;
        },
        type: "switch"
      },
      {
        field: "props.collapseTags",
        label: "\u6298\u53E0Tag",
        show: ({ values }) => values.props.props.multiple,
        type: "switch"
      },
      {
        field: "props.collapseTagsTooltip",
        label: "\u663E\u793A\u88AB\u6298\u53E0\u6807\u7B7E",
        show: ({ values }) => values.props.props.multiple && values.props.collapseTags,
        type: "switch"
      },
      {
        field: "props.showAllLevels",
        label: "\u4E0D\u663E\u793A\u8DEF\u5F84",
        props: {
          activeValue: false,
          inactiveValue: true
        },
        type: "switch"
      },
      {
        description: "\u7236\u5B50\u8282\u70B9\u4E0D\u4E92\u76F8\u5173\u8054",
        field: "props.props.checkStrictly",
        label: "\u7236\u7EA7\u53EF\u9009",
        type: "switch"
      },
      {
        field: "props.props.emitPath",
        label: "\u53EA\u83B7\u53D6\u9009\u4E2D\u8282\u70B9",
        props: {
          activeValue: false,
          inactiveValue: true
        },
        type: "switch"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u914D\u7F6E\u9009\u9879",
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        props: {
          tree: true
        },
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "array"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u9009\u4E2D\u8282\u70B9\u53D8\u5316\u65F6",
        type: "change"
      },
      {
        description: "	\u5F53\u5931\u53BB\u7126\u70B9\u65F6\u89E6\u53D1",
        type: "blur"
      },
      {
        description: "\u5F53\u83B7\u5F97\u7126\u70B9\u65F6\u89E6\u53D1",
        type: "focus"
      }
    ]
  },
  defaultSchema: {
    field: "cascader",
    input: true,
    label: "\u7EA7\u8054\u9009\u62E9\u5668",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ],
      placeholder: "\u8BF7\u9009\u62E9",
      props: {
        expandTrigger: "click"
      }
    },
    type: "cascader"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--full-coverage-outline",
  sort: 900
};

const Checkbox = {
  component: () => import('./chunks/checkbox.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "checkbox"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u9009\u4E2D\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u9009\u4E2D\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.radioButton",
        label: "\u6309\u94AE\u6A21\u5F0F",
        type: "switch"
      },
      {
        description: "\u6309\u94AE\u6A21\u5F0F\u4E0B\u751F\u6548",
        field: "props.textColor",
        label: "\u9009\u9879\u6587\u672C\u989C\u8272",
        layout: "horizontal",
        show: ({ values }) => {
          return values.props.radioButton;
        },
        type: "color-picker"
      },
      {
        description: "\u6309\u94AE\u6A21\u5F0F\u4E0B\u751F\u6548",
        field: "props.fill",
        label: "\u9009\u9879\u6309\u94AE\u989C\u8272",
        layout: "horizontal",
        show: ({ values }) => {
          return values.props.radioButton;
        },
        type: "color-picker"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "array"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "checkbox",
    input: true,
    label: "\u590D\u9009\u6846",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ]
    },
    type: "checkbox"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--dialogs-outline-rounded",
  sort: 860
};

const Col = {
  component: () => import('./chunks/col.cjs'),
  config: {
    attribute: [
      {
        field: "props.span",
        label: "\u5360\u4F4D\u683C\u6570",
        type: "number"
      }
    ]
  },
  defaultSchema: {
    label: "\u6805\u683C\u5E03\u5C40-\u5217",
    props: {
      span: 6
    },
    type: "col",
    children: []
  },
  editConstraints: {
    locked: true
  }
};

const Collapse = {
  component: () => import('./chunks/collapse.cjs'),
  config: {
    attribute: [
      {
        field: "children",
        label: "\u6298\u53E0\u9879\u7BA1\u7406",
        type: "EColEditor"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u6298\u53E0\u9762\u677F",
    type: "collapse",
    children: [
      {
        type: "collapse-item",
        children: [],
        props: {
          span: 12
        },
        id: "g062zikd2jk001"
      },
      {
        type: "collapse-item",
        children: [],
        props: {
          span: 12
        },
        id: "gy5z9jtfb3s001"
      }
    ]
  },
  icon: "epic-icon-xiala"
};

const CollapseItem = {
  component: () => import('./chunks/collapseItem.cjs'),
  config: {
    attribute: []
  },
  defaultSchema: {
    label: "\u6298\u53E0\u9879",
    type: "collapse-item",
    children: []
  }
};

const ColorPicker = {
  component: async () => (await import('element-plus')).ElColorPicker,
  config: {
    action: [],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "color-picker"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.colorFormat",
        label: "\u683C\u5F0F",
        props: {
          clearable: true,
          options: [
            {
              label: "hsl",
              value: "hsl"
            },
            {
              label: "hsv",
              value: "hsv"
            },
            {
              label: "hex",
              value: "hex"
            },
            {
              label: "rgb",
              value: "rgb"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.showAlpha",
        label: "\u900F\u660E\u5EA6\u9009\u62E9",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "color-picker",
    input: true,
    label: "\u989C\u8272\u9009\u62E9\u5668",
    type: "color-picker"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--palette-outline",
  sort: 950
};

const DatePicker = {
  component: async () => (await import('element-plus')).ElDatePicker,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "date"
      },
      {
        field: "props.type",
        label: "\u65E5\u671F\u7C7B\u578B",
        onChange({ value, values }) {
          values.props.defaultValue = null;
          if (["date", "daterange", "dates"].includes(value)) {
            values.props.format = "YYYY-MM-DD";
            values.props.valueFormat = "YYYY-MM-DD";
          } else if (["week"].includes(value)) {
            values.props.format = "ww [\u5468]";
            values.props.valueFormat = "YYYY-MM-DD";
          } else if (["month", "monthrange"].includes(value)) {
            values.props.format = "YYYY-MM";
            values.props.valueFormat = "YYYY-MM";
          } else if (["datetime", "datetimerange"].includes(value)) {
            values.props.format = "YYYY-MM-DD HH:mm:ss";
            values.props.valueFormat = "YYYY-MM-DD HH:mm:ss";
          } else if (["year"].includes(value)) {
            values.props.format = "YYYY";
            values.props.valueFormat = "YYYY";
          }
        },
        props: {
          options: [
            {
              label: "\u65E5\u671F",
              value: "date"
            },
            {
              label: "\u5468",
              value: "week"
            },
            {
              label: "\u6708\u4EFD",
              value: "month"
            },
            {
              label: "\u5E74\u4EFD",
              value: "year"
            },
            {
              label: "\u65F6\u95F4",
              value: "datetime"
            },
            {
              label: "\u65F6\u95F4\u8303\u56F4",
              value: "datetimerange"
            },
            {
              label: "\u65E5\u671F\u8303\u56F4",
              value: "daterange"
            },
            {
              label: "\u6708\u4EFD\u8303\u56F4",
              value: "monthrange"
            },
            {
              label: "\u591A\u4E2A\u65E5\u671F",
              value: "dates"
            }
          ]
        },
        type: "select"
      },
      {
        field: "props.startPlaceholder",
        label: "\u5F00\u59CB\u5360\u4F4D\u7B26",
        show({ values }) {
          return ["daterange", "datetimerange", "monthrange"].includes(
            values.props.type
          );
        },
        type: "input"
      },
      {
        field: "props.endPlaceholder",
        label: "\u7ED3\u675F\u5360\u4F4D\u7B26",
        show({ values }) {
          return ["daterange", "datetimerange", "monthrange"].includes(
            values.props.type
          );
        },
        type: "input"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        show({ values }) {
          return !["daterange", "datetimerange", "monthrange"].includes(
            values.props.type
          );
        },
        type: "input"
      },
      {
        defaultValue: "default",
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.editable",
        label: "\u53EF\u8F93\u5165",
        type: "switch"
      },
      {
        field: "props.format",
        label: "\u663E\u793A\u683C\u5F0F",
        type: "input"
      },
      {
        field: "props.valueFormat",
        label: "\u6570\u636E\u683C\u5F0F",
        type: "input"
      },
      {
        field: "props.rangeSeparator",
        label: "\u5206\u5272\u7B26",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show({ values }) {
          return ["daterange", "datetimerange", "monthrange"].includes(
            values.props.type
          );
        },
        type: "input"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.readonly",
        label: "\u53EA\u8BFB",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "date",
    input: true,
    label: "\u65E5\u671F\u9009\u62E9\u5668",
    props: {
      endPlaceholder: "\u8BF7\u9009\u62E9",
      format: "YYYY-MM-DD",
      placeholder: "\u8BF7\u9009\u62E9",
      startPlaceholder: "\u8BF7\u9009\u62E9",
      type: "date",
      valueFormat: "YYYY-MM-DD"
    },
    type: "date"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--calendar-month-outline-rounded",
  sort: 910
};

const Form = {
  component: () => import('./chunks/form.cjs'),
  config: {
    action: [
      {
        description: "\u83B7\u53D6\u8868\u5355\u6570\u636E",
        type: "getData"
      },
      {
        description: "\u8BBE\u7F6E\u8868\u5355\u6570\u636E",
        type: "setData"
      },
      {
        description: "\u6821\u9A8C\u8868\u5355",
        type: "validate"
      }
    ],
    attribute: [
      {
        field: "props.name",
        label: "Name",
        type: "input"
      },
      {
        field: "props.label-position",
        label: "\u6807\u7B7E\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5DE6\u8FB9",
              value: "left"
            },
            {
              label: "\u53F3\u8FB9",
              value: "right"
            },
            {
              label: "\u9876\u90E8",
              value: "top"
            }
          ],
          radioButton: true
        },
        type: "radio"
      },
      {
        field: "props.labelWidth",
        label: "\u6807\u7B7E\u5BBD\u5EA6",
        type: "EInputSize"
      },
      {
        field: "props.labelSuffix",
        label: "\u6807\u7B7E\u540E\u7F00",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.size",
        label: "\u8868\u5355\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.require-asterisk-position",
        label: "\u661F\u53F7\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5DE6\u8FB9",
              value: "left"
            },
            {
              label: "\u53F3\u8FB9",
              value: "right"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.inline-message",
        label: "\u884C\u5185\u5C55\u793A\u6821\u9A8C\u4FE1\u606F",
        type: "switch"
      },
      {
        field: "props.status-icon",
        label: "\u6821\u9A8C\u53CD\u9988\u56FE\u6807",
        type: "switch"
      },
      {
        field: "props.scroll-to-error",
        label: "\u6EDA\u52A8\u6821\u9A8C\u9519\u8BEF\u5904",
        type: "switch"
      },
      {
        field: "props.inline",
        label: "\u884C\u5185\u6A21\u5F0F",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u8868\u5355",
    props: {
      "label-position": "left",
      labelWidth: "100px",
      name: "default"
    },
    type: "form",
    children: []
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--list-alt-outline-rounded",
  sort: 600
};

const FormItem = {
  component: () => import('./chunks/formItem.cjs'),
  config: {},
  defaultSchema: {
    label: "\u8868\u5355\u9879",
    type: "form-item"
  }
};

const Input = {
  component: async () => (await import('element-plus')).ElInput,
  config: {
    action: [
      {
        description: "\u4F7F input \u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F input \u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u6E05\u9664 input \u503C",
        type: "clear"
      },
      {
        description: "\u9009\u4E2D input \u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "input"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        defaultValue: "default",
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.maxlength",
        label: "\u6700\u5927\u8F93\u5165\u957F\u5EA6",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showWordLimit",
        label: "\u7EDF\u8BA1\u5B57\u6570",
        show: ({ values }) => {
          return values.props.maxlength;
        },
        type: "switch"
      },
      {
        defaultValue: "text",
        field: "props.type",
        label: "\u8F93\u5165\u7C7B\u578B",
        props: {
          clearable: true,
          options: [
            {
              label: "text",
              value: "text"
            },
            {
              label: "number",
              value: "number"
            },
            {
              label: "password",
              value: "password"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.showPassword",
        label: "\u663E\u793A\u5207\u6362\u56FE\u6807",
        show: ({ values }) => {
          return values.props.type === "password";
        },
        type: "switch"
      },
      {
        field: "props.readonly",
        label: "\u53EA\u8BFB",
        type: "switch"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u8F93\u5165\u503C\u65F6",
        type: "input"
      },
      {
        description: "\u503C\u4FEE\u6539\u65F6",
        type: "change"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9\u65F6",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9\u65F6",
        type: "blur"
      },
      {
        description: "\u70B9\u51FB\u6E05\u7A7A\u6309\u94AE\u65F6",
        type: "clear"
      }
    ]
  },
  defaultSchema: {
    field: "input",
    input: true,
    label: "\u8F93\u5165\u6846",
    props: {
      placeholder: "\u8BF7\u8F93\u5165"
    },
    type: "input"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--border-color-outline-rounded",
  sort: 700
};

const InputNumber = {
  component: async () => (await import('element-plus')).ElInputNumber,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "number"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.step",
        label: "\u6B65\u957F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        defaultValue: "",
        field: "props.controlsPosition",
        label: "\u63A7\u5236\u6309\u94AE\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "default",
              value: ""
            },
            {
              label: "right",
              value: "right"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.precision",
        label: "\u7CBE\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.stepStrictly",
        label: "\u8F93\u5165\u63A7\u5236\u4E3A\u6B65\u957F\u7684\u500D\u6570",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "number"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "number",
    input: true,
    label: "\u6570\u5B57\u8F93\u5165\u6846",
    props: {
      placeholder: "\u8BF7\u8F93\u5165"
    },
    type: "number"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--123-rounded",
  sort: 710
};

const Modal = {
  component: () => import('./chunks/modal.cjs'),
  config: {
    attribute: [
      {
        field: "title",
        label: "\u6807\u9898",
        type: "input"
      }
    ]
  },
  defaultSchema: {
    label: "\u6A21\u6001\u6846",
    type: "modal",
    children: []
  }
};

const Radio = {
  component: () => import('./chunks/radio.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "radio"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.radioButton",
        label: "\u6309\u94AE\u6A21\u5F0F",
        type: "switch"
      },
      {
        description: "\u6309\u94AE\u6A21\u5F0F\u4E0B\u751F\u6548",
        field: "props.textColor",
        label: "\u9009\u9879\u6587\u672C\u989C\u8272",
        layout: "horizontal",
        show: ({ values }) => {
          return values.props.radioButton;
        },
        type: "color-picker"
      },
      {
        description: "\u6309\u94AE\u6A21\u5F0F\u4E0B\u751F\u6548",
        field: "props.fill",
        label: "\u9009\u9879\u6309\u94AE\u989C\u8272",
        layout: "horizontal",
        show: ({ values }) => {
          return values.props.radioButton;
        },
        type: "color-picker"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "radio",
    input: true,
    label: "\u5355\u9009\u6846",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ],
      size: "default"
    },
    type: "radio"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--radio-button-checked-outline",
  sort: 850
};

const Row = {
  component: () => import('./chunks/row.cjs'),
  config: {
    attribute: [
      {
        field: "props.align",
        label: "\u5782\u76F4\u5BF9\u9F50\u65B9\u5F0F",
        props: {
          options: [
            {
              label: "top",
              value: "top"
            },
            {
              label: "middle",
              value: "middle"
            },
            {
              label: "bottom",
              value: "bottom"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9",
          style: { width: "100%" }
        },
        type: "select"
      },
      {
        field: "props.justify",
        label: "\u6C34\u5E73\u6392\u5217\u65B9\u5F0F",
        props: {
          options: [
            {
              label: "start",
              value: "start"
            },
            {
              label: "end",
              value: "end"
            },
            {
              label: "center",
              value: "center"
            },
            {
              label: "space-around",
              value: "space-around"
            },
            {
              label: "space-between",
              value: "space-between"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9",
          style: { width: "100%" }
        },
        type: "select"
      },
      {
        field: "props.gutter",
        label: "\u6805\u683C\u95F4\u8DDD",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "children",
        label: "\u5217\u7F16\u8F91",
        type: "EColEditor"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u6805\u683C\u5E03\u5C40",
    props: {
      gutter: 12
    },
    type: "row",
    children: [
      {
        type: "col",
        children: [],
        props: {
          span: 12
        }
      },
      {
        type: "col",
        children: [],
        props: {
          span: 12
        }
      }
    ]
  },
  editConstraints: {
    childImmovable: true
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--width-normal-outline",
  sort: 800
};

const Select = {
  component: () => import('./chunks/select.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "select"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.multiple",
        label: "\u53EF\u591A\u9009",
        onChange: ({ value, values }) => {
          values.props.defaultValue = value ? [] : null;
        },
        type: "switch"
      },
      {
        field: "props.collapseTags",
        label: "\u591A\u9009\u9690\u85CF",
        show: ({ values }) => {
          return values.props.multiple;
        },
        type: "switch"
      },
      {
        field: "props.collapseTagsTooltip",
        label: "\u9690\u85CF\u63D0\u793A",
        show: ({ values }) => {
          return values.props.multiple && values.props.collapseTags;
        },
        type: "switch"
      },
      {
        field: "props.reserveKeyword",
        label: "\u4FDD\u7559\u641C\u7D22\u5173\u952E\u5B57",
        show: ({ values }) => {
          return values.props.multiple;
        },
        type: "switch"
      },
      {
        defaultValue: "info",
        field: "props.tagType",
        label: "\u6807\u7B7E\u7C7B\u578B",
        props: {
          options: [
            {
              label: "success",
              value: "success"
            },
            {
              label: "info",
              value: "info"
            },
            {
              label: "warning",
              value: "warning"
            },
            {
              label: "danger",
              value: "danger"
            }
          ]
        },
        show: ({ values }) => {
          return values.props.multiple;
        },
        type: "select"
      },
      {
        field: "props.multipleLimit",
        label: "\u591A\u9009\u9650\u5236",
        show: ({ values }) => {
          return values.props.multiple;
        },
        type: "number"
      },
      {
        field: "props.effect",
        label: "\u63D0\u793A\u4E3B\u9898",
        props: {
          options: [
            {
              label: "light",
              value: "light"
            },
            {
              label: "dark",
              value: "dark"
            }
          ]
        },
        type: "select"
      },
      {
        field: "props.filterable",
        label: "\u53EF\u7B5B\u9009",
        type: "switch"
      },
      {
        field: "props.allowCreate",
        label: "\u5141\u8BB8\u521B\u5EFA\u6761\u76EE",
        type: "switch"
      },
      {
        defaultValue: "bottom-start",
        field: "props.placement",
        label: "\u4E0B\u62C9\u6846\u4F4D\u7F6E",
        props: {
          options: [
            {
              label: "top",
              value: "top"
            },
            {
              label: "top-start",
              value: "top-start"
            },
            {
              label: "top-end",
              value: "top-end"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "bottom-start",
              value: "bottom-start"
            },
            {
              label: "bottom-end",
              value: "bottom-end"
            },
            {
              label: "left",
              value: "left"
            },
            {
              label: "left-start",
              value: "left-start"
            },
            {
              label: "left-end",
              value: "left-end"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "right-start",
              value: "right-start"
            },
            {
              label: "right-end",
              value: "right-end"
            }
          ]
        },
        type: "select"
      },
      {
        field: "props.fitInputWidth",
        label: "\u4E0B\u62C9\u6846\u5BBD\u5EA6\u4E0E\u8F93\u5165\u6846\u76F8\u540C",
        type: "switch"
      },
      {
        field: "props.noDataText",
        label: "\u65E0\u6570\u636E\u65F6\u6587\u672C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u914D\u7F6E\u9009\u9879",
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "select",
    input: true,
    label: "\u9009\u62E9\u6846",
    props: {
      effect: "light",
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ],
      placeholder: "\u8BF7\u9009\u62E9",
      placement: "bottom-start",
      size: "default"
    },
    type: "select"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--select",
  sort: 900
};

const Slider = {
  component: async () => (await import('element-plus')).ElSlider,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "slider"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u503C",
        onChange: ({ componentAttributes, value, values }) => {
          var _a;
          const componentIndex = componentAttributes.value.findIndex(
            (item) => item.field === "props.max"
          );
          const componentConfig = componentAttributes.value[componentIndex];
          componentConfig.props = {
            min: value
          };
          if (((_a = values.props.max) != null ? _a : 0) < value) {
            values.props.max = value;
          }
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u503C",
        onChange: ({ componentAttributes, value, values }) => {
          var _a;
          const componentIndex = componentAttributes.value.findIndex(
            (item) => item.field === "props.min"
          );
          const componentConfig = componentAttributes.value[componentIndex];
          componentConfig.props = {
            max: value
          };
          if (((_a = values.props.min) != null ? _a : 0) > value) {
            values.props.min = value;
          }
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.step",
        label: "\u6B65\u957F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showStops",
        label: "\u663E\u793A\u95F4\u65AD\u70B9",
        type: "switch"
      },
      {
        field: "props.showTooltip",
        label: "\u63D0\u793A\u4FE1\u606F",
        type: "switch"
      },
      {
        field: "props.range",
        label: "\u5F00\u542F\u9009\u62E9\u8303\u56F4",
        type: "switch"
      },
      {
        field: "props.vertical",
        label: "\u5782\u76F4\u6A21\u5F0F",
        type: "switch"
      },
      {
        field: "props.height",
        label: "\u9AD8\u5EA6",
        show: ({ values }) => {
          return values.props.vertical;
        },
        type: "EInputSize"
      },
      {
        field: "props.showInput",
        label: "\u663E\u793A\u6570\u5B57\u6846",
        onChange: ({ value, values }) => {
          value ? values.props.showInputControls = true : null;
        },
        type: "switch"
      },
      {
        field: "props.showInputControls",
        label: "\u63A7\u5236\u6309\u94AE",
        show: ({ values }) => values.props.showInput,
        type: "switch"
      },
      {
        defaultValue: "bottom-start",
        field: "props.placement",
        label: "\u63D0\u793A\u6846\u4F4D\u7F6E",
        props: {
          options: [
            {
              label: "top",
              value: "top"
            },
            {
              label: "top-start",
              value: "top-start"
            },
            {
              label: "top-end",
              value: "top-end"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "bottom-start",
              value: "bottom-start"
            },
            {
              label: "bottom-end",
              value: "bottom-end"
            },
            {
              label: "left",
              value: "left"
            },
            {
              label: "left-start",
              value: "left-start"
            },
            {
              label: "left-end",
              value: "left-end"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "right-start",
              value: "right-start"
            },
            {
              label: "right-end",
              value: "right-end"
            }
          ]
        },
        type: "select"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "number"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u62D6\u62FD\u5B9E\u65F6\u89E6\u53D1",
        type: "input"
      },
      {
        description: "\u62D6\u62FD\u7ED3\u675F\u6539\u53D8\u503C\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "slider",
    input: true,
    label: "\u6ED1\u5757",
    props: {
      placement: "top-start",
      showTooltip: true
    },
    type: "slider"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--switches",
  sort: 930
};

const Switch = {
  component: async () => (await import('element-plus')).ElSwitch,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "switch"
      },
      {
        field: "props.activeValue",
        label: "ON\u72B6\u6001\u503C",
        onChange(e) {
          setTimeout(() => setDefaultValue(e));
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.inactiveValue",
        label: "OFF\u72B6\u6001\u503C",
        onChange(e) {
          setTimeout(() => setDefaultValue(e));
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.activeText",
        label: "ON\u72B6\u6001\u63CF\u8FF0",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.inactiveText",
        label: "OFF\u72B6\u6001\u63CF\u8FF0",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.width",
        label: "\u5BBD\u5EA6",
        props: {
          min: 50,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.inlinePrompt",
        label: "\u6587\u672C\u70B9\u5185\u663E\u793A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "boolean"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u72B6\u6001\u53D1\u751F\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "switch",
    input: true,
    label: "\u5F00\u5173",
    props: {
      defaultValue: false
    },
    type: "switch"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--toggle-off-outline",
  sort: 930
};
function setDefaultValue(e) {
  var _a;
  const defaultValue = ((_a = e.values.props) == null ? void 0 : _a.inactiveValue) || false;
  if (e.values.props.unCheckedValue === "") {
    delete e.values.props.unCheckedValue;
  }
  if (e.values.props.activeValue === "") {
    delete e.values.props.activeValue;
  }
  if (e.values.props) {
    e.values.props.defaultValue = defaultValue;
  } else {
    e.values.props = {
      defaultValue
    };
  }
}

const TabsPane = {
  component: () => import('./chunks/tabPane.cjs'),
  config: {
    attribute: [
      {
        field: "props.tab",
        label: "\u5782\u76F4\u5BF9\u9F50\u65B9\u5F0F",
        type: "input"
      }
    ]
  },
  defaultSchema: {
    label: "\u6807\u7B7E\u5185\u5BB9",
    props: {
      tab: "\u6807\u7B7E"
    },
    type: "tab-pane",
    children: []
  },
  editConstraints: {
    locked: true
  }
};

const Tabs = {
  component: () => import('./chunks/tabs.cjs'),
  config: {
    action: [],
    attribute: [
      {
        field: "props.tabPosition",
        label: "\u9875\u7B7E\u4F4D\u7F6E",
        props: {
          allowClear: true,
          options: [
            {
              label: "top",
              value: "top"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "left",
              value: "left"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.type",
        label: "\u7C7B\u578B",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u9ED8\u8BA4",
              value: ""
            },
            {
              label: "\u5361\u7247",
              value: "card"
            },
            {
              label: "\u8FB9\u6846",
              value: "border-card"
            }
          ],
          radioButton: true
        },
        type: "radio"
      },
      {
        field: "children",
        label: "\u9875\u7B7E\u7BA1\u7406",
        layout: "vertical",
        type: "ETabPaneEditor"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ],
    event: [
      {
        description: "tab \u88AB\u9009\u4E2D\u65F6\u89E6\u53D1",
        type: "tabClick"
      },
      {
        description: "activeName \u6539\u53D8\u65F6\u89E6\u53D1",
        type: "tabChange"
      },
      {
        description: "\u70B9\u51FB tab \u65B0\u589E\u6309\u94AE\u65F6\u89E6\u53D1",
        type: "tabRemove"
      },
      {
        description: "\u70B9\u51FB tab \u65B0\u589E\u6309\u94AE\u65F6\u89E6\u53D1",
        type: "tabAdd"
      }
    ]
  },
  defaultSchema: {
    label: "\u6807\u7B7E\u9875",
    props: {
      defaultValue: "\u6807\u7B7E1",
      type: ""
    },
    type: "tabs",
    children: [
      {
        label: "\u6807\u7B7E1",
        type: "tab-pane",
        children: []
      },
      {
        label: "\u6807\u7B7E2",
        type: "tab-pane",
        children: []
      }
    ]
  },
  editConstraints: {
    childImmovable: true
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--layout-tabs"
};

const Textarea = {
  component: async () => (await import('element-plus')).ElInput,
  config: {
    action: [
      {
        description: "\u4F7F input \u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F input \u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u6E05\u9664 input \u503C",
        type: "clear"
      },
      {
        description: "\u9009\u4E2D input \u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "textarea"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "default"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.maxlength",
        label: "\u6700\u5927\u8F93\u5165\u957F\u5EA6",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showWordLimit",
        label: "\u7EDF\u8BA1\u5B57\u6570",
        show: ({ values }) => {
          return values.props.maxlength;
        },
        type: "switch"
      },
      {
        field: "props.autosize.minRows",
        label: "\u6700\u5C0F\u884C\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.autosize.maxRows",
        label: "\u6700\u5927\u884C\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u8F93\u5165\u503C",
        type: "input"
      },
      {
        description: "\u503C\u4FEE\u6539",
        type: "change"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "textarea",
    input: true,
    label: "\u6587\u672C\u57DF",
    props: {
      placeholder: "\u8BF7\u8F93\u5165",
      type: "textarea"
    },
    type: "textarea"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--edit-square-outline-rounded",
  sort: 705
};

const UploadFile = {
  bindModel: "modelValue",
  component: () => import('./chunks/uploadFile.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.action",
        label: "\u8BF7\u6C42\u5730\u5740",
        type: "input"
      },
      {
        field: "props.name",
        label: "\u4E0A\u4F20\u6587\u4EF6\u5B57\u6BB5",
        type: "input"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.showFileList",
        label: "\u5C55\u793A\u6587\u4EF6\u5217\u8868",
        type: "switch"
      },
      {
        field: "props.drag",
        label: "\u62D6\u62FD\u4E0A\u4F20",
        type: "switch"
      },
      {
        field: "props.limit",
        label: "\u5141\u8BB8\u4E0A\u4F20\u6700\u5927\u6570\u91CF",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "string"
        },
        type: "ERuleEditor"
      }
    ]
  },
  defaultSchema: {
    field: "uploadFile",
    input: true,
    label: "\u4E0A\u4F20\u6587\u4EF6",
    props: {
      action: "https://examples.epicjs.cn/epic-mock/common/upload",
      name: "file",
      showFileList: true
    },
    type: "upload-file"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--upload-rounded",
  sort: 920
};

const UploadImage = {
  bindModel: "modelValue",
  component: () => import('./chunks/uploadImage.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.action",
        label: "\u8BF7\u6C42\u5730\u5740",
        type: "input"
      },
      {
        field: "props.name",
        label: "\u4E0A\u4F20\u6587\u4EF6\u5B57\u6BB5",
        type: "input"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.showFileList",
        label: "\u5C55\u793A\u6587\u4EF6\u5217\u8868",
        type: "switch"
      },
      {
        field: "props.drag",
        label: "\u62D6\u62FD\u4E0A\u4F20",
        type: "switch"
      },
      {
        field: "props.limit",
        label: "\u5141\u8BB8\u4E0A\u4F20\u6700\u5927\u6570\u91CF",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "string"
        },
        type: "ERuleEditor"
      }
    ]
  },
  defaultSchema: {
    field: "uploadImage",
    input: true,
    label: "\u4E0A\u4F20\u56FE\u7247",
    props: {
      action: "https://examples.epicjs.cn/epic-mock/common/upload",
      name: "file",
      showFileList: true
    },
    type: "upload-image"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--imagesmode-outline-rounded",
  sort: 920
};

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = ".epic-designer-main .el-input {\n  width: 100% !important;\n}\n.epic-designer-main .el-form--inline [name='epic-draggable-range'] {\n  display: flex;\n  flex-wrap: wrap;\n}\n.epic-designer-main .el-form--inline .el-form-item__content {\n  width: 175px;\n}\n.epic-designer-main .el-collapse {\n  border: 1px solid var(--ep-border);\n  border-top: 0;\n}\n.epic-designer-main .el-collapse .el-collapse-item__header {\n  width: 100%;\n  height: 42px;\n  background: var(--ep-canvas-container-background);\n  padding-left: 12px;\n  border: 1px solid var(--ep-border);\n  border-left: 0;\n  border-right: 0;\n  border-bottom: 1px solid transparent;\n}\n.epic-designer-main .el-collapse .el-collapse-item__header.is-active {\n  border-bottom: 1px solid var(--ep-border);\n}\n.epic-designer-main .el-collapse .el-collapse-item:last-child .el-collapse-item__header {\n  border-bottom: 1px solid var(--ep-border);\n}\n.epic-designer-main .el-collapse .el-collapse-item__content {\n  padding: 8px;\n}\n.ep-input-size > .el-input__wrapper {\n  padding-right: 1px;\n}\n.ep-input-size .el-input__suffix {\n  margin-left: 0px;\n}\n.ep-input-size .el-input__suffix .el-select__wrapper {\n  height: 32px;\n  line-height: 30px;\n  padding: 0 8px 0 4px;\n  border: 0;\n  box-shadow: none !important;\n  border-left: 1;\n  position: relative;\n  line-height: auto;\n}\n.ep-input-size .el-select-selector {\n  border: none !important;\n  box-shadow: none !important;\n  padding: 0 4px !important;\n}\n.ep-el-modal {\n  margin: 20px auto 0 !important;\n  height: calc(100vh - 50px);\n}\n.ep-el-modal .el-dialog__body {\n  padding: 0;\n  height: calc(100% - 89px) !important;\n  overflow: auto;\n}\n.ep-el-modal-footer {\n  display: flex;\n  justify-content: flex-end;\n}\n";
styleInject(css_248z);

function setupElementPlus(pluginManager = epicDesigner.pluginManager, config = {}) {
  pluginManager.component.add("FormItem", elementPlus.ElFormItem);
  pluginManager.component.add("Collapse", elementPlus.ElCollapse);
  pluginManager.component.add("CollapseItem", elementPlus.ElCollapseItem);
  const componentArray = [
    Form,
    FormItem,
    Input,
    Textarea,
    InputNumber,
    Radio,
    Checkbox,
    DatePicker,
    Select,
    Switch,
    ColorPicker,
    Cascader,
    Slider,
    UploadFile,
    UploadImage,
    Button,
    Card,
    Row,
    Col,
    Collapse,
    CollapseItem,
    Modal,
    Tabs,
    TabsPane
  ];
  const { uploadFile, uploadImage } = config;
  if (uploadImage) UploadImage.defaultSchema.props.action = uploadImage;
  if (uploadFile) UploadFile.defaultSchema.props.action = uploadFile;
  vue.watchEffect(() => {
    var _a, _b;
    if (pluginManager.global.uploadImage) {
      UploadImage.config.attribute = (_a = UploadImage.config.attribute) == null ? void 0 : _a.filter(
        (item) => item.field !== "props.action"
      );
    }
    if (pluginManager.global.uploadFile) {
      UploadFile.config.attribute = (_b = UploadFile.config.attribute) == null ? void 0 : _b.filter(
        (item) => item.field !== "props.action"
      );
    }
  });
  componentArray.forEach((item) => {
    pluginManager.component.register(item);
    pluginManager.component.addBaseComponentType(item.defaultSchema.type);
  });
  pluginManager.global.$message = {
    error: elementPlus.ElMessage.error,
    info: elementPlus.ElMessage.info,
    success: elementPlus.ElMessage.success,
    warning: elementPlus.ElMessage.warning
  };
  pluginManager.designer.setInitialized(true);
}
function provideGlobalConfig(args) {
  import('element-plus').then(({ provideGlobalConfig: provideGlobalConfig2 }) => {
    provideGlobalConfig2(args);
  });
}

exports.provideGlobalConfig = provideGlobalConfig;
exports.setupElementPlus = setupElementPlus;
