// @epic-designer/naive-ui/dist/index.mjs
import { watchEffect } from 'vue';
import { pluginManager } from 'epic-designer';
import { createDiscreteApi } from 'naive-ui';

const Button = {
  component: () => import('./chunks/button.mjs'),
  config: {
    attribute: [
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.type",
        label: "\u7C7B\u578B",
        props: {
          options: [
            {
              label: "default",
              value: "default"
            },
            {
              label: "tertiary",
              value: "tertiary"
            },
            {
              label: "primary",
              value: "primary"
            },
            {
              label: "success",
              value: "success"
            },
            {
              label: "info",
              value: "info"
            },
            {
              label: "warning",
              value: "warning"
            },
            {
              label: "error",
              value: "error"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        defaultValue: "default",
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "tiny",
              value: "tiny"
            },
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.block",
        label: "\u662F\u5426\u5757\u7EA7",
        type: "switch"
      },
      {
        field: "props.bordered",
        label: "\u663E\u793A\u8FB9\u6846",
        type: "switch"
      },
      {
        field: "props.dashed",
        label: "\u865A\u7EBF\u6846",
        type: "switch"
      },
      {
        field: "props.round",
        label: "\u5706\u89D2\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.circle",
        label: "\u5706\u5F62\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.color",
        label: "\u6309\u94AE\u989C\u8272",
        type: "color-picker"
      },
      {
        field: "props.ghost",
        label: "\u662F\u5426\u900F\u660E",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ],
    event: [
      {
        description: "\u70B9\u51FB\u6309\u94AE\u65F6",
        type: "click"
      },
      {
        description: "\u53CC\u51FB\u6309\u94AE\u65F6",
        type: "dblclick"
      }
    ]
  },
  defaultSchema: {
    field: "input",
    input: false,
    label: "\u6309\u94AE",
    props: {
      bordered: true,
      size: "medium",
      type: "default"
    },
    type: "button"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--fit-screen-rounded",
  sort: 1200
};

const Card = {
  component: () => import('./chunks/card.mjs'),
  config: {
    attribute: [
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u5361\u7247\u5E03\u5C40",
    type: "card",
    children: []
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--wysiwyg-rounded",
  sort: 900
};

const Cascader = {
  bindModel: "value",
  component: async () => (await import('naive-ui')).NCascader,
  config: {
    action: [
      {
        description: "\u83B7\u53D6\u5F53\u524D\u9009\u4E2D\u8282\u70B9",
        type: "getSelectedNodes"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "cascader"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u6781\u5C0F",
              value: "tiny"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5927\u53F7",
              value: "large"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.cascade",
        label: "\u591A\u9009\u65F6\u5173\u8054\u9009\u9879",
        show: ({ values }) => values.props.multiple,
        type: "switch"
      },
      {
        field: "props.separator",
        label: "\u5206\u5272\u7B26",
        type: "switch"
      },
      {
        field: "props.showPath",
        label: "\u663E\u793A\u9009\u9879\u8DEF\u5F84",
        type: "switch"
      },
      {
        field: "props.filterable",
        label: "\u53EF\u641C\u7D22",
        type: "switch"
      },
      {
        field: "props.placement",
        label: "\u83DC\u5355\u5F39\u51FA\u7684\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "top-start",
              value: "top-start"
            },
            {
              label: "top",
              value: "top"
            },
            {
              label: "top-end",
              value: "top-end"
            },
            {
              label: "right-start",
              value: "right-start"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "right-end",
              value: "right-end"
            },
            {
              label: "bottom-start",
              value: "bottom-start"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "bottom-end",
              value: "bottom-end"
            },
            {
              label: "left-start",
              value: "left-start"
            },
            {
              label: "left",
              value: "left"
            },
            {
              label: "left-end",
              value: "left-end"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.maxTagCount",
        label: "\u6700\u5927tag\u663E\u793A\u6570",
        show: ({ values }) => values.props.multiple,
        type: "number"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u914D\u7F6E\u9009\u9879",
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        props: {
          tree: true
        },
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "array"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u9009\u4E2D\u8282\u70B9\u53D8\u5316\u65F6",
        type: "change"
      },
      {
        description: "\u9762\u677F\u7684\u5173\u95ED\u4E8B\u4EF6",
        type: "close"
      }
    ]
  },
  defaultSchema: {
    field: "cascader",
    input: true,
    label: "\u7EA7\u8054\u9009\u62E9\u5668",
    props: {
      cascade: false,
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ],
      placeholder: "\u8BF7\u9009\u62E9",
      showPath: true
    },
    type: "cascader"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--full-coverage-outline",
  sort: 880
};

const Checkbox = {
  component: () => import('./chunks/checkbox.mjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "checkbox"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u52FE\u9009\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u52FE\u9009\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u914D\u7F6E\u9009\u9879",
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "array"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "checkbox",
    input: true,
    label: "\u590D\u9009\u6846",
    props: {
      // defaultValue: [],
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ]
    },
    type: "checkbox"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--dialogs-outline-rounded",
  sort: 860
};

const Col = {
  component: () => import('./chunks/col.mjs'),
  config: {
    attribute: [
      {
        field: "props.span",
        label: "\u5360\u4F4D\u683C\u6570",
        type: "number"
      }
    ]
  },
  defaultSchema: {
    label: "\u6805\u683C\u5E03\u5C40-\u5217",
    props: {
      span: 6
    },
    type: "col",
    children: []
  }
};

const Collapse = {
  component: () => import('./chunks/collapse.mjs'),
  config: {
    attribute: [
      {
        field: "children",
        label: "\u6298\u53E0\u9879\u7BA1\u7406",
        type: "EColEditor"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u6298\u53E0\u9762\u677F",
    type: "collapse",
    children: [
      {
        type: "collapse-item",
        children: [],
        props: {
          span: 12
        },
        id: "g062zikd2jk001"
      },
      {
        type: "collapse-item",
        children: [],
        props: {
          span: 12
        },
        id: "gy5z9jtfb3s001"
      }
    ]
  }
};

const CollapseItem = {
  component: () => import('./chunks/collapseItem.mjs'),
  config: {
    attribute: []
  },
  defaultSchema: {
    label: "\u6298\u53E0\u9879",
    type: "collapse-item",
    children: []
  }
};

const ColorPicker = {
  bindModel: "value",
  component: async () => (await import('naive-ui')).NColorPicker,
  config: {
    action: [],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "color-picker"
      },
      {
        defaultValue: "default",
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.placement",
        label: "\u83DC\u5355\u5F39\u51FA\u7684\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "top-start",
              value: "top-start"
            },
            {
              label: "top",
              value: "top"
            },
            {
              label: "top-end",
              value: "top-end"
            },
            {
              label: "right-start",
              value: "right-start"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "right-end",
              value: "right-end"
            },
            {
              label: "bottom-start",
              value: "bottom-start"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "bottom-end",
              value: "bottom-end"
            },
            {
              label: "left-start",
              value: "left-start"
            },
            {
              label: "left",
              value: "left"
            },
            {
              label: "left-end",
              value: "left-end"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.modes",
        label: "\u683C\u5F0F",
        props: {
          max: 1,
          options: [
            {
              label: "rgb",
              value: "rgb"
            },
            {
              label: "hex",
              value: "hex"
            },
            {
              label: "hsl",
              value: "hsl"
            },
            {
              label: "hsv",
              value: "hsv"
            }
          ]
        },
        type: "checkbox"
      },
      {
        field: "props.showPreview",
        label: "\u5C55\u793A\u9884\u89C8\u5757",
        type: "switch"
      },
      {
        field: "props.showAlpha",
        label: "\u53EF\u8C03\u8282\u900F\u660E\u5EA6",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "color-picker",
    input: true,
    label: "\u989C\u8272\u9009\u62E9\u5668",
    props: {
      modes: ["hex"],
      showAlpha: true,
      type: "color"
    },
    type: "color-picker"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--palette-outline",
  sort: 950
};

const DatePicker = {
  bindModel: "formatted-value",
  component: async () => (await import('naive-ui')).NDatePicker,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "date"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.type",
        label: "\u65E5\u671F\u7C7B\u578B",
        onChange({ value, values }) {
          values.props.defaultValue = null;
          if (["date", "daterange", "dates"].includes(value)) {
            values.props.format = "yyyy-MM-dd";
            values.props.valueFormat = "yyyy-MM-dd";
          } else if (["week"].includes(value)) {
            values.props.format = null;
            values.props.valueFormat = "yyyy-MM-dd";
          } else if (["quarter"].includes(value)) {
            values.props.format = null;
            values.props.valueFormat = "yyyy-MM-dd";
          } else if (["month", "monthrange"].includes(value)) {
            values.props.format = "yyyy-MM";
            values.props.valueFormat = "yyyy-MM";
          } else if (["datetime", "datetimerange"].includes(value)) {
            values.props.format = "yyyy-MM-dd HH:mm:ss";
            values.props.valueFormat = "yyyy-MM-dd HH:mm:ss";
          } else if (["year"].includes(value)) {
            values.props.format = "yyyy";
            values.props.valueFormat = "yyyy";
          }
        },
        props: {
          options: [
            {
              label: "\u65E5\u671F",
              value: "date"
            },
            {
              label: "\u65E5\u671F\u65F6\u95F4",
              value: "datetime"
            },
            {
              label: "\u65E5\u671F\u8303\u56F4",
              value: "daterange"
            },
            {
              label: "\u65F6\u95F4\u8303\u56F4",
              value: "datetimerange"
            },
            {
              label: "\u6708\u4EFD",
              value: "month"
            },
            {
              label: "\u6708\u4EFD\u8303\u56F4",
              value: "monthrange"
            },
            {
              label: "\u5E74\u4EFD",
              value: "year"
            },
            {
              label: "\u5B63\u5EA6",
              value: "quarter"
            },
            {
              label: "\u5468",
              value: "week"
            }
          ]
        },
        type: "select"
      },
      {
        field: "props.format",
        label: "\u663E\u793A\u683C\u5F0F",
        type: "input"
      },
      {
        field: "props.valueFormat",
        label: "\u6570\u636E\u683C\u5F0F",
        type: "input"
      },
      {
        field: "props.separator",
        label: "\u5206\u5272\u7B26",
        props: {
          clearable: true,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => ["daterange", "datetimerange", "monthrange"].includes(
          values.props.type
        ),
        type: "input"
      },
      {
        field: "props.startPlaceholder",
        label: "start\u6846\u5360\u4F4D\u7B26",
        props: {
          clearable: true,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => ["daterange", "datetimerange", "monthrange"].includes(
          values.props.type
        ),
        type: "input"
      },
      {
        field: "props.endPlaceholder",
        label: "end\u6846\u5360\u4F4D\u7B26",
        props: {
          clearable: true,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => ["daterange", "datetimerange", "monthrange"].includes(
          values.props.type
        ),
        type: "input"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "date",
    input: true,
    label: "\u65E5\u671F\u9009\u62E9\u5668",
    props: {
      format: "yyyy-MM-dd",
      placeholder: "\u8BF7\u9009\u62E9",
      type: "date",
      valueFormat: "yyyy-MM-dd"
    },
    type: "date"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--calendar-month-outline-rounded",
  sort: 910
};

const Form = {
  component: () => import('./chunks/form.mjs'),
  config: {
    action: [
      {
        description: "\u83B7\u53D6\u8868\u5355\u6570\u636E",
        type: "getData"
      },
      {
        description: "\u8BBE\u7F6E\u8868\u5355\u6570\u636E",
        type: "setData"
      },
      {
        description: "\u6821\u9A8C\u8868\u5355",
        type: "validate"
      }
    ],
    attribute: [
      {
        field: "props.name",
        label: "name",
        type: "input"
      },
      {
        field: "props.labelWidth",
        label: "\u6807\u7B7E\u5BBD\u5EA6",
        type: "EInputSize"
      },
      {
        field: "props.labelPlacement",
        label: "\u6807\u7B7E\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5DE6\u8FB9",
              value: "left"
            },
            {
              label: "\u9876\u90E8",
              value: "top"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.labelAlign",
        label: "\u6807\u7B7E\u5BF9\u9F50\u65B9\u5F0F",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5DE6\u8FB9",
              value: "left"
            },
            {
              label: "\u53F3\u8FB9",
              value: "right"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.requireMarkPlacement",
        label: "\u5FC5\u586B\u661F\u53F7\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5DE6\u8FB9",
              value: "left"
            },
            {
              label: "\u53F3\u8FB9",
              value: "right"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.size",
        label: "\u8868\u5355\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.inline",
        label: "\u884C\u5185\u6A21\u5F0F",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u8868\u5355",
    props: {
      hideRequiredMark: false,
      labelCol: {
        span: 5
      },
      labelWidth: 100,
      name: "default",
      wrapperCol: {
        span: 19
      }
    },
    type: "form",
    children: []
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--list-alt-outline-rounded",
  sort: 600
};

const FormItem = {
  component: () => import('./chunks/formItem.mjs'),
  config: {},
  defaultSchema: {
    label: "\u8868\u5355\u9879",
    type: "form-item"
  }
};

const Input = {
  bindModel: "value",
  component: async () => (await import('naive-ui')).NInput,
  config: {
    action: [
      {
        description: "\u4F7F input \u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F input \u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u6E05\u9664 input \u503C",
        type: "clear"
      },
      {
        description: "\u9009\u4E2D input \u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "input"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u6781\u5C0F",
              value: "tiny"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5927\u53F7",
              value: "large"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        defaultValue: "text",
        field: "props.type",
        label: "\u8F93\u5165\u7C7B\u578B",
        onChange: ({ value, values }) => {
          if (value !== "text") values.props.pair = false;
        },
        props: {
          clearable: true,
          options: [
            {
              label: "text",
              value: "text"
            },
            {
              label: "textarea",
              value: "textarea"
            },
            {
              label: "password",
              value: "password"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.maxlength",
        label: "\u6700\u5927\u8F93\u5165\u957F\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showPasswordOn",
        label: "\u663E\u793A\u5BC6\u7801\u7684\u65F6\u673A",
        props: {
          clearable: true,
          options: [
            {
              label: "click",
              value: "click"
            },
            {
              label: "mousedown",
              value: "mousedown"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        show: ({ values }) => values.props.type === "password",
        type: "select"
      },
      {
        field: "props.rows",
        label: "\u884C\u6570",
        show: ({ values }) => values.props.type === "textarea",
        type: "number"
      },
      {
        field: "props.separator",
        label: "\u5206\u5272\u7B26",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => values.props.type === "text" && values.props.pair,
        type: "input"
      },
      {
        field: "props.round",
        label: "\u662F\u5426\u5706\u89D2",
        type: "switch"
      },
      {
        field: "props.showCount",
        label: "\u662F\u5426\u7EDF\u8BA1\u5B57\u6570",
        type: "switch"
      },
      {
        field: "props.autosize",
        label: "\u81EA\u9002\u5E94\u5185\u5BB9\u9AD8\u5EA6",
        show: ({ values }) => values.props.type === "textarea",
        type: "switch"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u8F93\u5165\u503C",
        type: "input"
      },
      {
        description: "\u503C\u4FEE\u6539",
        type: "change"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "input",
    input: true,
    label: "\u8F93\u5165\u6846",
    props: {
      defaultValue: "",
      placeholder: "\u8BF7\u8F93\u5165",
      type: "text"
    },
    type: "input"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--border-color-outline-rounded",
  sort: 700
};

const InputNumber = {
  bindModel: "value",
  component: async () => (await import('naive-ui')).NInputNumber,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "number"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u6781\u5C0F",
              value: "tiny"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5927\u53F7",
              value: "large"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.buttonPlacement",
        label: "\u63A7\u5236\u6309\u94AE\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "both",
              value: "both"
            },
            {
              label: "right",
              value: "right"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.step",
        label: "\u6B65\u957F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.precision",
        label: "\u7CBE\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "number"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u6539\u53D8\u65F6\u89E6\u53D1",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "number",
    input: true,
    label: "\u6570\u5B57\u8F93\u5165\u6846",
    props: {
      placeholder: "\u8BF7\u8F93\u5165"
    },
    type: "number"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--123-rounded",
  sort: 710
};

const Modal = {
  component: () => import('./chunks/modal.mjs'),
  config: {
    attribute: [
      {
        field: "title",
        label: "\u6807\u9898",
        type: "input"
      }
    ]
  },
  defaultSchema: {
    label: "\u6A21\u6001\u6846",
    type: "modal",
    children: []
  }
};

const Radio = {
  component: () => import('./chunks/radio.mjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "radio"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u914D\u7F6E\u9009\u9879",
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "radio",
    input: true,
    label: "\u5355\u9009\u6846",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ]
    },
    type: "radio"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--radio-button-checked-outline",
  sort: 850
};

const Row = {
  component: () => import('./chunks/row.mjs'),
  config: {
    attribute: [
      // {
      //   label: "垂直对齐方式",
      //   type: "select",
      //   props: {
      //     style: { width: "100%" },
      //     options: [
      //       {
      //         label: "top",
      //         value: "top",
      //       },
      //       {
      //         label: "middle",
      //         value: "middle",
      //       },
      //       {
      //         label: "bottom",
      //         value: "bottom",
      //       },
      //     ],
      //     placeholder: "请选择",
      //   },
      //   field: "props.align",
      // },
      // {
      //   label: "水平排列方式",
      //   type: "select",
      //   props: {
      //     style: { width: "100%" },
      //     options: [
      //       {
      //         label: "start",
      //         value: "start",
      //       },
      //       {
      //         label: "end",
      //         value: "end",
      //       },
      //       {
      //         label: "center",
      //         value: "center",
      //       },
      //       {
      //         label: "space-around",
      //         value: "space-around",
      //       },
      //       {
      //         label: "space-between",
      //         value: "space-between",
      //       },
      //     ],
      //     placeholder: "请选择",
      //   },
      //   field: "props.justify",
      // },
      {
        field: "props.gutter",
        label: "\u6805\u683C\u95F4\u8DDD",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "children",
        label: "\u5217\u7F16\u8F91",
        type: "EColEditor"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u6805\u683C\u5E03\u5C40",
    type: "row",
    children: [
      {
        type: "col",
        children: [],
        props: {
          span: 12
        }
      },
      {
        type: "col",
        children: [],
        props: {
          span: 12
        }
      }
    ]
  },
  editConstraints: {
    childImmovable: true
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--width-normal-outline",
  sort: 800
};

const Select = {
  bindModel: "value",
  component: async () => (await import('naive-ui')).NSelect,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "select"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u6781\u5C0F",
              value: "tiny"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5927\u53F7",
              value: "large"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.placement",
        label: "\u83DC\u5355\u5F39\u51FA\u7684\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "top-start",
              value: "top-start"
            },
            {
              label: "top",
              value: "top"
            },
            {
              label: "top-end",
              value: "top-end"
            },
            {
              label: "right-start",
              value: "right-start"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "right-end",
              value: "right-end"
            },
            {
              label: "bottom-start",
              value: "bottom-start"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "bottom-end",
              value: "bottom-end"
            },
            {
              label: "left-start",
              value: "left-start"
            },
            {
              label: "left",
              value: "left"
            },
            {
              label: "left-end",
              value: "left-end"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.filterable",
        label: "\u53EF\u8FC7\u6EE4",
        type: "switch"
      },
      {
        field: "props.tag",
        label: "\u5141\u8BB8\u521B\u5EFA\u65B0\u9009\u9879",
        show: ({ values }) => values.props.filterable,
        type: "switch"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.maxTagCount",
        label: "\u6700\u5927tag\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => values.props.multiple,
        type: "number"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      },
      {
        description: "\u914D\u7F6E\u9009\u9879",
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u4FEE\u6539",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "select",
    input: true,
    label: "\u9009\u62E9\u6846",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ],
      placeholder: "\u8BF7\u9009\u62E9"
    },
    type: "select"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--select",
  sort: 900
};

const Slider = {
  bindModel: "value",
  component: async () => (await import('naive-ui')).NSlider,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "slider"
      },
      {
        field: "props.keyboard",
        label: "\u952E\u76D8\u53EF\u63A7",
        type: "switch"
      },
      {
        changeSync: true,
        field: "props.range",
        label: "\u8303\u56F4\u9009\u62E9",
        onChange: ({ value, values }) => {
          values.props.defaultValue = value ? [0, 100] : 0;
        },
        type: "switch"
      },
      {
        field: "props.reverse",
        label: "\u5012\u8F6C\u8F68\u9053",
        type: "switch"
      },
      {
        field: "props.step",
        label: "\u6B65\u957F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showTooltip",
        label: "\u4E00\u76F4\u663E\u793A\u63D0\u793A",
        type: "switch"
      },
      {
        field: "props.vertical",
        label: "\u5782\u76F4\u6A21\u5F0F",
        type: "switch"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.placement",
        label: "\u5F39\u51FA\u4F4D\u7F6E",
        props: {
          clearable: true,
          options: [
            {
              label: "top-start",
              value: "top-start"
            },
            {
              label: "top",
              value: "top"
            },
            {
              label: "top-end",
              value: "top-end"
            },
            {
              label: "right-start",
              value: "right-start"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "right-end",
              value: "right-end"
            },
            {
              label: "bottom-start",
              value: "bottom-start"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "bottom-end",
              value: "bottom-end"
            },
            {
              label: "left-start",
              value: "left-start"
            },
            {
              label: "left",
              value: "left"
            },
            {
              label: "left-end",
              value: "left-end"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "number"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u62D6\u62FD\u5B9E\u65F6\u89E6\u53D1",
        type: "input"
      },
      {
        description: "\u62D6\u62FD\u7ED3\u675F\u6539\u53D8\u503C\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "slider",
    input: true,
    label: "\u6ED1\u5757",
    props: {
      placement: "top-start"
    },
    type: "slider"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--switches",
  sort: 930
};

const Switch = {
  bindModel: "value",
  component: async () => (await import('naive-ui')).NSwitch,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "switch"
      },
      {
        field: "props.checkedValue",
        label: "ON\u72B6\u6001\u503C",
        onChange(e) {
          setTimeout(() => setDefaultValue(e));
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.uncheckedValue",
        label: "OFF\u72B6\u6001\u503C",
        onChange(e) {
          setTimeout(() => setDefaultValue(e));
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.round",
        label: "\u5706\u578B\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "boolean"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u72B6\u6001\u53D1\u751F\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "switch",
    input: true,
    label: "\u5F00\u5173",
    props: {
      defaultValue: false,
      round: true
    },
    type: "switch"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--toggle-off-outline",
  sort: 930
};
function setDefaultValue(e) {
  var _a;
  const defaultValue = ((_a = e.values.props) == null ? void 0 : _a.uncheckedValue) || false;
  if (e.values.props.uncheckedValue === "") {
    delete e.values.props.uncheckedValue;
  }
  if (e.values.props.checkedValue === "") {
    delete e.values.props.checkedValue;
  }
  if (e.values.props) {
    e.values.props.defaultValue = defaultValue;
  } else {
    e.values.props = {
      defaultValue
    };
  }
}

const TabsPane = {
  component: () => import('./chunks/tabPane.mjs'),
  config: {
    attribute: [
      {
        field: "props.tab",
        label: "\u5782\u76F4\u5BF9\u9F50\u65B9\u5F0F",
        type: "input"
      }
    ]
  },
  defaultSchema: {
    label: "\u6807\u7B7E\u5185\u5BB9",
    props: {
      tab: "\u6807\u7B7E"
    },
    type: "tab-pane",
    children: []
  },
  editConstraints: {
    locked: true
  }
};

const Tabs = {
  bindModel: "value",
  component: () => import('./chunks/tabs.mjs'),
  config: {
    action: [],
    attribute: [
      {
        field: "children",
        label: "\u9875\u7B7E\u7BA1\u7406",
        layout: "vertical",
        type: "ETabPaneEditor"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ],
    event: [
      {
        description: "tab \u88AB\u9009\u4E2D\u65F6\u89E6\u53D1",
        type: "tabClick"
      },
      {
        description: "activeName \u6539\u53D8\u65F6\u89E6\u53D1",
        type: "tabChange"
      },
      {
        description: "\u70B9\u51FB tab \u65B0\u589E\u6309\u94AE\u65F6\u89E6\u53D1",
        type: "tabRemove"
      },
      {
        description: "\u70B9\u51FB tab \u65B0\u589E\u6309\u94AE\u65F6\u89E6\u53D1",
        type: "tabAdd"
      }
    ]
  },
  defaultSchema: {
    field: "tabs",
    label: "\u6807\u7B7E\u9875",
    props: {
      defaultValue: "\u6807\u7B7E1"
    },
    type: "tabs",
    children: [
      {
        label: "\u6807\u7B7E1",
        type: "tab-pane",
        children: []
      },
      {
        label: "\u6807\u7B7E2",
        type: "tab-pane",
        children: []
      }
    ]
  },
  editConstraints: {
    childImmovable: true
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--layout-tabs"
};

const Textarea = {
  bindModel: "value",
  component: () => import('./chunks/textarea.mjs'),
  config: {
    action: [
      {
        description: "\u4F7F input \u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F input \u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u6E05\u9664 input \u503C",
        type: "clear"
      },
      {
        description: "\u9009\u4E2D input \u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "textarea"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          clearable: true,
          options: [
            {
              label: "\u6781\u5C0F",
              value: "tiny"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            },
            {
              label: "\u4E2D\u7B49",
              value: "medium"
            },
            {
              label: "\u5927\u53F7",
              value: "large"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.maxlength",
        label: "\u6700\u5927\u8F93\u5165\u957F\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.rows",
        label: "\u884C\u6570",
        type: "number"
      },
      {
        field: "props.round",
        label: "\u662F\u5426\u5706\u89D2",
        type: "switch"
      },
      {
        field: "props.showCount",
        label: "\u662F\u5426\u7EDF\u8BA1\u5B57\u6570",
        type: "switch"
      },
      {
        field: "props.autosize",
        label: "\u81EA\u9002\u5E94\u5185\u5BB9\u9AD8\u5EA6",
        type: "switch"
      },
      {
        field: "props.clearable",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u8F93\u5165\u503C",
        type: "input"
      },
      {
        description: "\u503C\u4FEE\u6539",
        type: "change"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "textarea",
    input: true,
    label: "\u6587\u672C\u57DF",
    props: {
      defaultValue: "",
      placeholder: "\u8BF7\u8F93\u5165"
    },
    type: "textarea"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--edit-square-outline-rounded",
  sort: 705
};

const UploadFile = {
  bindModel: "modelValue",
  component: () => import('./chunks/uploadFile.mjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.action",
        label: "\u8BF7\u6C42\u5730\u5740",
        type: "input"
      },
      {
        field: "props.name",
        label: "\u4E0A\u4F20\u6587\u4EF6\u5B57\u6BB5",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.max",
        label: "\u5141\u8BB8\u4E0A\u4F20\u6700\u5927\u6570\u91CF",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showFileList",
        label: "\u5C55\u793A\u6587\u4EF6\u5217\u8868",
        type: "switch"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "string"
        },
        type: "ERuleEditor"
      }
    ]
  },
  defaultSchema: {
    field: "uploadFile",
    input: true,
    label: "\u4E0A\u4F20\u6587\u4EF6",
    props: {
      action: "https://examples.epicjs.cn/epic-mock/common/upload"
    },
    type: "upload-file"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--upload-rounded",
  sort: 920
};

const UploadImage = {
  bindModel: "modelValue",
  component: () => import('./chunks/uploadImage.mjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.action",
        label: "\u8BF7\u6C42\u5730\u5740",
        type: "input"
      },
      {
        field: "props.name",
        label: "\u4E0A\u4F20\u6587\u4EF6\u5B57\u6BB5",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.max",
        label: "\u5141\u8BB8\u4E0A\u4F20\u6700\u5927\u6570\u91CF",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showFileList",
        label: "\u5C55\u793A\u6587\u4EF6\u5217\u8868",
        type: "switch"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "string"
        },
        type: "ERuleEditor"
      }
    ]
  },
  defaultSchema: {
    field: "uploadImage",
    input: true,
    label: "\u4E0A\u4F20\u56FE\u7247",
    props: {
      action: "https://examples.epicjs.cn/epic-mock/common/upload"
    },
    type: "upload-image"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--imagesmode-outline-rounded",
  sort: 920
};

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = ".n-collapse .n-collapse-item:first-child > .n-collapse-item__header,\n.n-collapse .n-collapse-item > .n-collapse-item__header {\n  padding-top: 16px;\n}\n.n-collapse .n-collapse-item:first-child > .n-collapse-item__header .n-collapse-item__header-main,\n.n-collapse .n-collapse-item > .n-collapse-item__header .n-collapse-item__header-main {\n  padding-left: 8px;\n}\n.ep-input-size > .n-input-wrapper {\n  padding-right: 0;\n}\n.ep-input-size .n-base-selection-input {\n  padding: 0 4px !important;\n}\n.ep-input-size .n-base-selection__state-border,\n.ep-input-size .n-base-selection__border {\n  border: 0 !important;\n  box-shadow: none !important;\n}\n.epic-modal-n {\n  margin: 0 auto 0 !important;\n  height: calc(100vh - 50px);\n}\n.epic-modal-n .n-card-header {\n  padding: 16px 16px 0;\n}\n.epic-modal-n .n-card__content {\n  padding: 0;\n  height: calc(100% - 56px);\n}\n.epic-modal-n .n-card__content .epic-modal-main {\n  height: calc(100% - 56px);\n  overflow: auto;\n  padding: 12px;\n}\n";
styleInject(css_248z);

function setupNaiveUi(pluginManager$1 = pluginManager, config = {}) {
  pluginManager$1.component.add(
    "FormItem",
    async () => (await import('naive-ui')).NFormItem
  );
  pluginManager$1.component.add(
    "Collapse",
    async () => (await import('naive-ui')).NCollapse
  );
  pluginManager$1.component.add(
    "CollapseItem",
    async () => (await import('naive-ui')).NCollapseItem
  );
  const componentArray = [
    Form,
    FormItem,
    Input,
    InputNumber,
    Radio,
    Checkbox,
    DatePicker,
    Select,
    Switch,
    ColorPicker,
    Cascader,
    Slider,
    UploadFile,
    UploadImage,
    Button,
    Card,
    Row,
    Col,
    Collapse,
    CollapseItem,
    Modal,
    Tabs,
    TabsPane,
    Textarea
  ];
  const { uploadFile, uploadImage } = config;
  if (uploadImage) UploadImage.defaultSchema.props.action = uploadImage;
  if (uploadFile) UploadFile.defaultSchema.props.action = uploadFile;
  watchEffect(() => {
    var _a, _b;
    if (pluginManager$1.global.uploadImage) {
      UploadImage.config.attribute = (_a = UploadImage.config.attribute) == null ? void 0 : _a.filter(
        (item) => item.field !== "props.action"
      );
    }
    if (pluginManager$1.global.uploadFile) {
      UploadFile.config.attribute = (_b = UploadFile.config.attribute) == null ? void 0 : _b.filter(
        (item) => item.field !== "props.action"
      );
    }
  });
  componentArray.forEach((item) => {
    pluginManager$1.component.register(item);
    pluginManager$1.component.addBaseComponentType(item.defaultSchema.type);
  });
  const { message } = createDiscreteApi(["message"]);
  pluginManager$1.global.$message = {
    error: message.error,
    info: message.info,
    success: message.success,
    warning: message.warning
  };
  pluginManager$1.designer.setInitialized(true);
}

export { setupNaiveUi };
