// @epic-designer/naive-ui/dist/chunks/form.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const form = require('naive-ui/lib/form');

const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "form",
  props: {
    componentSchema: {
      default: () => ({}),
      require: true,
      type: Object
    }
  },
  setup(__props, { expose: __expose }) {
    var _a, _b, _c;
    const props = __props;
    const form$1 = vue.ref(null);
    const { formData, formInstances } = epicDesigner.useForm(
      (_c = (_b = (_a = props.componentSchema) == null ? void 0 : _a.props) == null ? void 0 : _b.name) != null ? _c : "default"
    );
    function getData() {
      return formData;
    }
    function setData(data) {
      Object.assign(formData, data);
    }
    function resetData() {
      var _a2;
      (_a2 = form$1.value) == null ? void 0 : _a2.restoreValidation();
      const inputSchemas = epicDesigner.findSchemas(
        props.componentSchema.children,
        (schema) => Boolean(schema.input)
      );
      if (Array.isArray(inputSchemas)) {
        inputSchemas.forEach((schema) => {
          var _a3;
          const defaultValue = (_a3 = schema.props) == null ? void 0 : _a3.defaultValue;
          if (defaultValue === void 0) {
            formData[schema.field] = null;
          } else {
            formData[schema.field] = defaultValue;
          }
        });
      }
    }
    function validate() {
      var _a2;
      return (_a2 = form$1.value) == null ? void 0 : _a2.validate();
    }
    function clearValidate() {
      var _a2;
      return (_a2 = form$1.value) == null ? void 0 : _a2.restoreValidation();
    }
    vue.onMounted(async () => {
      var _a2, _b2, _c2, _d, _e, _f;
      if (((_a2 = props.componentSchema) == null ? void 0 : _a2.type) === "form" && formInstances.value && form$1.value) {
        const name = (_f = (_e = (_c2 = (_b2 = props.componentSchema) == null ? void 0 : _b2.props) == null ? void 0 : _c2.name) != null ? _e : (_d = props.componentSchema) == null ? void 0 : _d.name) != null ? _f : "default";
        formInstances.value[name] = form$1.value;
        form$1.value.getData = getData;
        form$1.value.setData = setData;
        form$1.value.resetData = resetData;
      }
    });
    const formProps = vue.computed(() => {
      const recordProps = props.componentSchema.props;
      return recordProps;
    });
    const children = vue.computed(() => {
      var _a2;
      return (_a2 = props.componentSchema.children) != null ? _a2 : [];
    });
    __expose({
      clearValidate,
      form: form$1,
      getData,
      resetData,
      setData,
      validate
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createBlock(vue.unref(form.NForm), vue.mergeProps({
        ref_key: "form",
        ref: form$1,
        model: vue.unref(formData)
      }, formProps.value), {
        default: vue.withCtx(() => [
          vue.renderSlot(_ctx.$slots, "edit-node", {}, () => [
            (vue.openBlock(true), vue.createElementBlock(
              vue.Fragment,
              null,
              vue.renderList(children.value, (item) => {
                return vue.renderSlot(_ctx.$slots, "node", { componentSchema: item });
              }),
              256
              /* UNKEYED_FRAGMENT */
            ))
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 16, ["model"]);
    };
  }
});

exports.default = _sfc_main;
