// @epic-designer/naive-ui/dist/chunks/modal.mjs
import { defineComponent, useAttrs, computed, createBlock, openBlock, unref, mergeProps, withCtx, createElementVNode, renderSlot, createElementBlock, createCommentVNode, Fragment, renderList, createVNode, createTextVNode, toDisplayString } from 'vue';
import { NModal, NSpace, NButton } from 'naive-ui';

const _hoisted_1 = { class: "epic-modal-main" };
const _hoisted_2 = { class: "epic-modal-footer" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "modal",
  props: {
    cancelText: { type: String, required: false, default: "\u5173\u95ED" },
    componentSchema: { type: Object, required: false, default: () => ({
      type: "modal"
    }) },
    hideConfirm: { type: Boolean, required: false },
    okText: { type: String, required: false, default: "\u786E\u5B9A" },
    width: { type: String, required: false, default: "900px" }
  },
  emits: ["ok", "close", "update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const attrs = useAttrs();
    const getProps = computed(() => {
      var _a, _b;
      return {
        ...props.componentSchema,
        class: "epic-modal-n",
        preset: "card",
        show: attrs.modelValue,
        title: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : ""
      };
    });
    const children = computed(() => {
      var _a, _b;
      return (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    });
    const handleOk = () => {
      emit("ok");
    };
    function updateModelValue(value = false) {
      emit("update:modelValue", value);
    }
    function handleClose() {
      emit("close");
    }
    return (_ctx, _cache) => {
      var _a;
      return openBlock(), createBlock(unref(NModal), mergeProps(getProps.value, {
        title: (_a = getProps.value.label) != null ? _a : "",
        class: "epic-modal-n ep-scoped",
        preset: "card",
        style: { width: __props.width },
        "onUpdate:show": updateModelValue
      }), {
        default: withCtx(() => [
          createElementVNode("div", _hoisted_1, [
            renderSlot(_ctx.$slots, "default", {}, () => [
              renderSlot(_ctx.$slots, "edit-node", {}, () => [
                children.value.length > 0 ? (openBlock(true), createElementBlock(
                  Fragment,
                  { key: 0 },
                  renderList(children.value, (node) => {
                    return renderSlot(_ctx.$slots, "node", { componentSchema: node });
                  }),
                  256
                  /* UNKEYED_FRAGMENT */
                )) : createCommentVNode("v-if", true)
              ])
            ])
          ]),
          createElementVNode("div", _hoisted_2, [
            createVNode(unref(NSpace), { justify: "end" }, {
              default: withCtx(() => [
                createVNode(unref(NButton), { onClick: handleClose }, {
                  default: withCtx(() => [
                    createTextVNode(
                      toDisplayString(props.cancelText),
                      1
                      /* TEXT */
                    )
                  ]),
                  _: 1
                  /* STABLE */
                }),
                !props.hideConfirm ? (openBlock(), createBlock(unref(NButton), {
                  key: 0,
                  type: "primary",
                  onClick: handleOk
                }, {
                  default: withCtx(() => [
                    createTextVNode(
                      toDisplayString(props.okText),
                      1
                      /* TEXT */
                    )
                  ]),
                  _: 1
                  /* STABLE */
                })) : createCommentVNode("v-if", true)
              ]),
              _: 1
              /* STABLE */
            })
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 16, ["title", "style"]);
    };
  }
});

export { _sfc_main as default };
