// @epic-designer/naive-ui/dist/chunks/modal.cjs
'use strict';

const vue = require('vue');
const naiveUi = require('naive-ui');

const _hoisted_1 = { class: "epic-modal-main" };
const _hoisted_2 = { class: "epic-modal-footer" };
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "modal",
  props: {
    cancelText: { type: String, required: false, default: "\u5173\u95ED" },
    componentSchema: { type: Object, required: false, default: () => ({
      type: "modal"
    }) },
    hideConfirm: { type: Boolean, required: false },
    okText: { type: String, required: false, default: "\u786E\u5B9A" },
    width: { type: String, required: false, default: "900px" }
  },
  emits: ["ok", "close", "update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const attrs = vue.useAttrs();
    const getProps = vue.computed(() => {
      var _a, _b;
      return {
        ...props.componentSchema,
        class: "epic-modal-n",
        preset: "card",
        show: attrs.modelValue,
        title: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : ""
      };
    });
    const children = vue.computed(() => {
      var _a, _b;
      return (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    });
    const handleOk = () => {
      emit("ok");
    };
    function updateModelValue(value = false) {
      emit("update:modelValue", value);
    }
    function handleClose() {
      emit("close");
    }
    return (_ctx, _cache) => {
      var _a;
      return vue.openBlock(), vue.createBlock(vue.unref(naiveUi.NModal), vue.mergeProps(getProps.value, {
        title: (_a = getProps.value.label) != null ? _a : "",
        class: "epic-modal-n ep-scoped",
        preset: "card",
        style: { width: __props.width },
        "onUpdate:show": updateModelValue
      }), {
        default: vue.withCtx(() => [
          vue.createElementVNode("div", _hoisted_1, [
            vue.renderSlot(_ctx.$slots, "default", {}, () => [
              vue.renderSlot(_ctx.$slots, "edit-node", {}, () => [
                children.value.length > 0 ? (vue.openBlock(true), vue.createElementBlock(
                  vue.Fragment,
                  { key: 0 },
                  vue.renderList(children.value, (node) => {
                    return vue.renderSlot(_ctx.$slots, "node", { componentSchema: node });
                  }),
                  256
                  /* UNKEYED_FRAGMENT */
                )) : vue.createCommentVNode("v-if", true)
              ])
            ])
          ]),
          vue.createElementVNode("div", _hoisted_2, [
            vue.createVNode(vue.unref(naiveUi.NSpace), { justify: "end" }, {
              default: vue.withCtx(() => [
                vue.createVNode(vue.unref(naiveUi.NButton), { onClick: handleClose }, {
                  default: vue.withCtx(() => [
                    vue.createTextVNode(
                      vue.toDisplayString(props.cancelText),
                      1
                      /* TEXT */
                    )
                  ]),
                  _: 1
                  /* STABLE */
                }),
                !props.hideConfirm ? (vue.openBlock(), vue.createBlock(vue.unref(naiveUi.NButton), {
                  key: 0,
                  type: "primary",
                  onClick: handleOk
                }, {
                  default: vue.withCtx(() => [
                    vue.createTextVNode(
                      vue.toDisplayString(props.okText),
                      1
                      /* TEXT */
                    )
                  ]),
                  _: 1
                  /* STABLE */
                })) : vue.createCommentVNode("v-if", true)
              ]),
              _: 1
              /* STABLE */
            })
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 16, ["title", "style"]);
    };
  }
});

exports.default = _sfc_main;
