// @epic-designer/naive-ui/dist/chunks/tabs.mjs
import { defineComponent, ref, watch, computed, h } from 'vue';
import { NTabs } from 'naive-ui';

const tabs = defineComponent({
  inheritAttrs: false,
  props: {
    componentSchema: {
      default: () => ({}),
      require: true,
      type: Object
    }
  },
  setup(props, { slots }) {
    var _a, _b;
    const modelValue = ref();
    watch(
      () => props.componentSchema.props.defaultValue,
      (newValue) => {
        var _a2;
        const tabPaneLabels = (_a2 = props.componentSchema.children) == null ? void 0 : _a2.map(
          (node) => node.label
        );
        modelValue.value = (tabPaneLabels == null ? void 0 : tabPaneLabels.includes(newValue)) ? newValue : tabPaneLabels == null ? void 0 : tabPaneLabels[0];
      },
      {
        immediate: true
      }
    );
    const componentSchema = computed(() => {
      var _a2;
      return {
        ...(_a2 = props.componentSchema) == null ? void 0 : _a2.props,
        "onUpdate:value": (e) => {
          modelValue.value = e;
        },
        value: modelValue.value
      };
    });
    const children = (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    return () => h(NTabs, componentSchema.value, {
      default: () => {
        var _a2, _b2;
        const defaultSlotList = (_a2 = slots["edit-node"]) == null ? void 0 : _a2.call(slots)[0].children;
        if (defaultSlotList == null ? void 0 : defaultSlotList.length) {
          const list = (_b2 = defaultSlotList[0].children) == null ? void 0 : _b2.map((item) => {
            item.props.name = item.props.schema.label;
            item.props.tab = item.props.schema.label;
            item.type.__TAB_PANE__ = true;
            return item;
          });
          return list;
        }
        const defaultSlot = children.map((node) => {
          var _a3;
          const TabPane = (_a3 = slots.node) == null ? void 0 : _a3.call(slots, {
            componentSchema: node,
            name: node.label
          })[0];
          TabPane.type.__TAB_PANE__ = true;
          return TabPane;
        });
        return [defaultSlot];
      }
    });
  }
});

export { tabs as default };
