// @epic-designer/naive-ui/dist/chunks/uploadImage.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const naiveUi = require('naive-ui');

const uploadImage = vue.defineComponent({
  emits: ["update:modelValue", "change"],
  props: {
    modelValue: {
      default: "",
      type: String
    }
  },
  setup(props, { attrs, emit }) {
    const fileList = vue.ref([]);
    let urlString = "";
    vue.watch(
      () => fileList.value,
      (list) => {
        urlString = list.filter((file) => file.status === "finished").map((file) => file.url).join(",");
        emit("update:modelValue", urlString);
        emit("change", urlString);
      }
    );
    vue.watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue !== null && fileList.value !== null) {
          fileList.value = modelValue.split(",").map((url) => ({
            id: epicDesigner.getUUID(),
            name: url,
            status: "finished",
            url
          }));
        }
      },
      { immediate: true }
    );
    function handleUpdate(e) {
      vue.nextTick(() => {
        fileList.value = e;
      });
    }
    const handleSuccess = ({ event, file }) => {
      var _a, _b, _c;
      const resInfo = event == null ? void 0 : event.target;
      const resData = JSON.parse((_a = resInfo.response) != null ? _a : "{}");
      file.url = (_c = (_b = resData.data) == null ? void 0 : _b.url) != null ? _c : resData.data;
    };
    const handleError = ({ event, file }) => {
      console.log("OnError called->", file, event);
    };
    return () => {
      return vue.h(
        "div",
        {
          class: "epic-upload-image"
        },
        {
          default: () => {
            var _a, _b;
            return [
              vue.h(
                naiveUi.NUpload,
                {
                  ...attrs,
                  accept: "image/gif,image/jpeg,image/jpg,image/png,image/svg",
                  action: (_a = epicDesigner.pluginManager.global.uploadImage) != null ? _a : attrs.action,
                  headers: (_b = epicDesigner.pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
                  onError: handleError,
                  onFinish: handleSuccess,
                  "file-list": fileList.value,
                  "list-type": "image-card",
                  "onUpdate:file-list": handleUpdate
                },
                {
                  default: () => [
                    vue.h(
                      "div",
                      { style: { "text-align": "center" } },
                      {
                        default: () => [
                          vue.h("span", {
                            class: "icon--epic icon--epic--cloud-upload-outlined text-md mr-2px text-lg"
                          }),
                          vue.h(
                            "div",
                            { class: "ant-upload-text" },
                            { default: () => "\u70B9\u51FB\u4E0A\u4F20" }
                          )
                        ]
                      }
                    )
                  ]
                }
              )
            ];
          }
        }
      );
    };
  }
});

exports.default = uploadImage;
