// @epic-designer/naive-ui/dist/chunks/uploadFile.mjs
import { defineComponent, ref, watch, h, nextTick } from 'vue';
import { getFileNameByUrl, getUUID, pluginManager } from 'epic-designer';
import { NUpload, NButton } from 'naive-ui';

const uploadFile = defineComponent({
  emits: ["update:modelValue", "change"],
  props: {
    modelValue: {
      default: "",
      type: String
    }
  },
  setup(props, { attrs, emit }) {
    const fileList = ref([]);
    let urlString = "";
    watch(
      () => fileList.value,
      (list) => {
        urlString = list.filter((file) => file.status === "finished").map((file) => file.url).join(",");
        emit("update:modelValue", urlString);
        emit("change", urlString);
      }
    );
    watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue !== null && fileList.value !== null) {
          fileList.value = modelValue.split(",").map((url) => ({
            id: getUUID(),
            name: getFileNameByUrl(url),
            status: "finished",
            url
          }));
        }
      },
      { immediate: true }
    );
    function handleUpdate(e) {
      nextTick(() => {
        fileList.value = e;
      });
    }
    const handleError = () => {
    };
    const handleSuccess = ({ event, file }) => {
      var _a, _b, _c;
      const resInfo = event == null ? void 0 : event.target;
      const resData = JSON.parse((_a = resInfo == null ? void 0 : resInfo.response) != null ? _a : "{}");
      file.url = (_c = (_b = resData.data) == null ? void 0 : _b.url) != null ? _c : resData.data;
    };
    return () => {
      return h("div", null, {
        default: () => {
          var _a, _b;
          return [
            h(
              NUpload,
              {
                ...attrs,
                action: (_a = pluginManager.global.uploadFile) != null ? _a : attrs.action,
                "default-upload": true,
                headers: (_b = pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
                onError: handleError,
                onFinish: handleSuccess,
                "file-list": fileList.value,
                "onUpdate:file-list": handleUpdate
              },
              {
                default: () => [
                  h(
                    NButton,
                    {},
                    {
                      default: () => [
                        h("span", {
                          class: "icon--epic icon--epic--cloud-upload-outlined text-lg",
                          style: { "margin-right": "2px" }
                        }),
                        h("span", null, { default: () => "\u4E0A\u4F20\u6587\u4EF6" })
                      ]
                    }
                  )
                ]
              }
            )
          ];
        }
      });
    };
  }
});

export { uploadFile as default };
