// @epic-designer/antd/dist/chunks/input.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const antDesignVue = require('ant-design-vue');

const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "input",
  props: {
    componentSchema: { type: Object, required: false, default: () => ({
      field: "",
      type: "input"
    }) },
    dataSelector: { type: Boolean, required: false, default: false },
    options: { type: Array, required: false, default: () => [] }
  },
  setup(__props) {
    const props = __props;
    const formData = vue.inject("formData", {});
    const eventBus = epicDesigner.useEventBus();
    const handleDataSelector = () => {
      eventBus == null ? void 0 : eventBus.emit("openDataSelector", props.componentSchema, props.options);
    };
    eventBus == null ? void 0 : eventBus.on("dataSelector:selected", (data) => {
      var _a, _b;
      if (data.nodeId !== props.componentSchema.id) {
        return;
      }
      const dataFieldMap = (_b = (_a = props.componentSchema.props) == null ? void 0 : _a.dataFieldMap) != null ? _b : [];
      dataFieldMap.forEach((item) => {
        formData[item.formField] = data.record[item.dataField];
      });
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createBlock(vue.unref(antDesignVue.Input), null, {
        suffix: vue.withCtx(() => [
          props.dataSelector ? (vue.openBlock(), vue.createElementBlock("span", {
            key: 0,
            onClick: handleDataSelector,
            class: "iconfont icon--epic icon--epic--search-rounded"
          })) : vue.createCommentVNode("v-if", true),
          vue.renderSlot(_ctx.$slots, "suffix")
        ]),
        _: 3
        /* FORWARDED */
      });
    };
  }
});

exports.default = _sfc_main;
