// @epic-designer/antd/dist/chunks/modal.mjs
import { defineComponent, useAttrs, computed, createBlock, openBlock, unref, normalizeProps, guardReactiveProps, withCtx, createElementVNode, createElementBlock, createCommentVNode, renderSlot, Fragment, renderList, createVNode, createTextVNode, toDisplayString } from 'vue';
import { version, Modal, Space, Button } from 'ant-design-vue';

const _hoisted_1 = { class: "epic-modal-main ep-scoped" };
const _hoisted_2 = {
  key: 0,
  class: "epic-modal-footer"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "modal",
  props: {
    cancelText: { type: String, required: false, default: "\u5173\u95ED" },
    componentSchema: { type: Object, required: false, default: () => ({
      type: "modal"
    }) },
    fixedHeight: { type: Boolean, required: false, default: true },
    footer: { type: Boolean, required: false, default: true },
    hideConfirm: { type: Boolean, required: false },
    okText: { type: String, required: false, default: "\u786E\u5B9A" }
  },
  emits: ["ok", "close", "update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const versionArray = version.split(".");
    const firstNumber = Number.parseInt(versionArray[0]);
    const attrs = useAttrs();
    const dialogStyle = {
      bottom: "30px",
      left: "150px",
      minHeight: "500px",
      minWidth: "700px",
      position: "absolute",
      right: "150px",
      top: "56px",
      width: "auto"
    };
    const getProps = computed(() => {
      var _a, _b;
      return {
        ...props.componentSchema,
        ...attrs,
        bodyStyle: {
          height: props.fixedHeight ? "calc(100vh - 108px)" : "auto",
          padding: 0
        },
        dialogStyle,
        footer: null,
        "onUpdate:open": updateModelValue,
        "onUpdate:visible": updateModelValue,
        open: attrs.modelValue,
        style: props.fixedHeight ? "top:20px" : "",
        title: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : "",
        visible: firstNumber > 3 ? void 0 : attrs.modelValue,
        wrapClassName: "epic-modal-ant",
        children: null
      };
    });
    const children = computed(() => {
      var _a, _b;
      return (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    });
    function handleOk() {
      emits("ok");
    }
    function updateModelValue(value = false) {
      emits("update:modelValue", value);
    }
    function handleClose() {
      emits("close");
    }
    return (_ctx, _cache) => {
      return openBlock(), createBlock(
        unref(Modal),
        normalizeProps(guardReactiveProps(getProps.value)),
        {
          default: withCtx(() => [
            createElementVNode("div", _hoisted_1, [
              renderSlot(_ctx.$slots, "default", {}, () => [
                renderSlot(_ctx.$slots, "edit-node", {}, () => [
                  (openBlock(true), createElementBlock(
                    Fragment,
                    null,
                    renderList(children.value, (item) => {
                      return renderSlot(_ctx.$slots, "node", { componentSchema: item });
                    }),
                    256
                    /* UNKEYED_FRAGMENT */
                  ))
                ])
              ])
            ]),
            props.footer ? (openBlock(), createElementBlock("div", _hoisted_2, [
              createVNode(unref(Space), { align: "end" }, {
                default: withCtx(() => [
                  createVNode(unref(Button), { onClick: handleClose }, {
                    default: withCtx(() => [
                      createTextVNode(
                        toDisplayString(props.cancelText),
                        1
                        /* TEXT */
                      )
                    ]),
                    _: 1
                    /* STABLE */
                  }),
                  !props.hideConfirm ? (openBlock(), createBlock(unref(Button), {
                    key: 0,
                    type: "primary",
                    onClick: handleOk
                  }, {
                    default: withCtx(() => [
                      createTextVNode(
                        toDisplayString(props.okText),
                        1
                        /* TEXT */
                      )
                    ]),
                    _: 1
                    /* STABLE */
                  })) : createCommentVNode("v-if", true)
                ]),
                _: 1
                /* STABLE */
              })
            ])) : createCommentVNode("v-if", true)
          ]),
          _: 3
          /* FORWARDED */
        },
        16
        /* FULL_PROPS */
      );
    };
  }
});

export { _sfc_main as default };
