// @epic-designer/antd/dist/chunks/modal.cjs
'use strict';

const vue = require('vue');
const antDesignVue = require('ant-design-vue');

const _hoisted_1 = { class: "epic-modal-main ep-scoped" };
const _hoisted_2 = {
  key: 0,
  class: "epic-modal-footer"
};
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "modal",
  props: {
    cancelText: { type: String, required: false, default: "\u5173\u95ED" },
    componentSchema: { type: Object, required: false, default: () => ({
      type: "modal"
    }) },
    fixedHeight: { type: Boolean, required: false, default: true },
    footer: { type: Boolean, required: false, default: true },
    hideConfirm: { type: Boolean, required: false },
    okText: { type: String, required: false, default: "\u786E\u5B9A" }
  },
  emits: ["ok", "close", "update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const versionArray = antDesignVue.version.split(".");
    const firstNumber = Number.parseInt(versionArray[0]);
    const attrs = vue.useAttrs();
    const dialogStyle = {
      bottom: "30px",
      left: "150px",
      minHeight: "500px",
      minWidth: "700px",
      position: "absolute",
      right: "150px",
      top: "56px",
      width: "auto"
    };
    const getProps = vue.computed(() => {
      var _a, _b;
      return {
        ...props.componentSchema,
        ...attrs,
        bodyStyle: {
          height: props.fixedHeight ? "calc(100vh - 108px)" : "auto",
          padding: 0
        },
        dialogStyle,
        footer: null,
        "onUpdate:open": updateModelValue,
        "onUpdate:visible": updateModelValue,
        open: attrs.modelValue,
        style: props.fixedHeight ? "top:20px" : "",
        title: (_b = (_a = props.componentSchema) == null ? void 0 : _a.label) != null ? _b : "",
        visible: firstNumber > 3 ? void 0 : attrs.modelValue,
        wrapClassName: "epic-modal-ant",
        children: null
      };
    });
    const children = vue.computed(() => {
      var _a, _b;
      return (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    });
    function handleOk() {
      emits("ok");
    }
    function updateModelValue(value = false) {
      emits("update:modelValue", value);
    }
    function handleClose() {
      emits("close");
    }
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createBlock(
        vue.unref(antDesignVue.Modal),
        vue.normalizeProps(vue.guardReactiveProps(getProps.value)),
        {
          default: vue.withCtx(() => [
            vue.createElementVNode("div", _hoisted_1, [
              vue.renderSlot(_ctx.$slots, "default", {}, () => [
                vue.renderSlot(_ctx.$slots, "edit-node", {}, () => [
                  (vue.openBlock(true), vue.createElementBlock(
                    vue.Fragment,
                    null,
                    vue.renderList(children.value, (item) => {
                      return vue.renderSlot(_ctx.$slots, "node", { componentSchema: item });
                    }),
                    256
                    /* UNKEYED_FRAGMENT */
                  ))
                ])
              ])
            ]),
            props.footer ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_2, [
              vue.createVNode(vue.unref(antDesignVue.Space), { align: "end" }, {
                default: vue.withCtx(() => [
                  vue.createVNode(vue.unref(antDesignVue.Button), { onClick: handleClose }, {
                    default: vue.withCtx(() => [
                      vue.createTextVNode(
                        vue.toDisplayString(props.cancelText),
                        1
                        /* TEXT */
                      )
                    ]),
                    _: 1
                    /* STABLE */
                  }),
                  !props.hideConfirm ? (vue.openBlock(), vue.createBlock(vue.unref(antDesignVue.Button), {
                    key: 0,
                    type: "primary",
                    onClick: handleOk
                  }, {
                    default: vue.withCtx(() => [
                      vue.createTextVNode(
                        vue.toDisplayString(props.okText),
                        1
                        /* TEXT */
                      )
                    ]),
                    _: 1
                    /* STABLE */
                  })) : vue.createCommentVNode("v-if", true)
                ]),
                _: 1
                /* STABLE */
              })
            ])) : vue.createCommentVNode("v-if", true)
          ]),
          _: 3
          /* FORWARDED */
        },
        16
        /* FULL_PROPS */
      );
    };
  }
});

exports.default = _sfc_main;
