// epic-designer/dist/core/src/components/designer/src/modules/preview/index.vue.js
import { defineComponent as E, ref as a, computed as O, createBlock as h, openBlock as g, unref as n, withCtx as k, createElementVNode as x, createCommentVNode as S, createVNode as V, nextTick as B } from "vue";
/* empty css                                                 */
/* empty css                                                             */
/* empty css                                                                  */
import { pluginManager as m } from "../../../../../../../manager/src/pluginManager.js";
/* empty css                                                           */
import "vuedraggable";
import { findSchemas as M } from "../../../../../../../utils/src/common/data.js";
import "../../../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import "@vueuse/core";
import "../../../../../../../hooks/src/store/index.js";
import { useDesignerContext as $ } from "../../../../../../../hooks/src/designer/useDesignerContext.js";
import D from "../../../../builder/src/builder.vue.js";
const R = { class: "min-w-750px translate-y-0px h-full rounded" }, U = { class: "h-full rounded" }, Q = /* @__PURE__ */ E({
  __name: "index",
  props: {
    hideConfirm: { type: Boolean },
    width: { default: "900px" }
  },
  setup(s, { expose: y }) {
    const C = s, _ = m.component.get("monacoEditor"), d = m.component.get("modal"), f = a(null), l = a(!1), r = a(!1), i = a({}), { pageSchema: c } = $(), p = a(null), u = O(() => {
      const o = M(
        c.schemas,
        (e) => {
          var t;
          return e.type.includes("form") && ((t = e.props) == null ? void 0 : t.name);
        }
      ).map((e) => {
        var t;
        return (t = e.props) == null ? void 0 : t.name;
      });
      return [...new Set(o)];
    });
    function v() {
      r.value = !1;
    }
    function w() {
      l.value = !1;
    }
    function N() {
      l.value = !0;
    }
    async function b() {
      if (!u.value.length) {
        m.global.$message.error("缺少表单组件!");
        return;
      }
      try {
        let o = {};
        for (const e of u.value)
          o[`${e}`] = await p.value.validate(e);
        u.value.length === 1 && u.value[0] === "default" && (o = o.default), i.value = JSON.stringify(o, null, 2), B(() => {
          var e;
          (e = f.value) == null || e.setValue(i.value);
        }), r.value = !0;
      } catch (o) {
        console.error(o);
      }
    }
    return y({
      handleOpen: N
    }), (o, e) => (g(), h(n(d), {
      modelValue: l.value,
      "onUpdate:modelValue": e[1] || (e[1] = (t) => l.value = t),
      title: "预览",
      width: s.width,
      "hide-confirm": C.hideConfirm,
      "ok-text": "表单数据",
      onClose: w,
      onOk: b
    }, {
      default: k(() => [
        x("div", R, [
          l.value ? (g(), h(n(D), {
            key: 0,
            ref_key: "kb",
            ref: p,
            "page-schema": n(c)
          }, null, 8, ["page-schema"])) : S("", !0),
          V(n(d), {
            modelValue: r.value,
            "onUpdate:modelValue": e[0] || (e[0] = (t) => r.value = t),
            title: "表单数据",
            width: "860px",
            onClose: v,
            onOk: v
          }, {
            default: k(() => [
              x("div", U, [
                V(n(_), {
                  ref_key: "monacoEditorRef",
                  ref: f,
                  "auto-toggle-theme": "",
                  "read-only": "",
                  class: "editor h-full",
                  "model-value": i.value
                }, null, 8, ["model-value"])
              ])
            ]),
            _: 1
          }, 8, ["modelValue"])
        ])
      ]),
      _: 1
    }, 8, ["modelValue", "width", "hide-confirm"]));
  }
});
export {
  Q as default
};
