// @epic-designer/.antd-y9OB2uUN/dist/chunks/datePicker.cjs
'use strict';

const vue = require('vue');
const antDesignVue = require('ant-design-vue');

const datePicker = vue.defineComponent({
  emits: ["update:modelValue", "change", "blur"],
  name: "EDatePicker",
  props: {
    modelValue: {
      default: null,
      type: [String, Object, Array]
    },
    type: {
      default: "date",
      type: String
    }
  },
  setup(props, { emit }) {
    vue.watch(
      () => props.type,
      () => {
        handleUpdate();
      }
    );
    function handleUpdate(e = null) {
      emit("update:modelValue", e);
      emit("change", e);
      emit("blur", e);
    }
    return () => {
      let cmp = antDesignVue.DatePicker;
      const compProps = {
        "onUpdate:value": handleUpdate,
        picker: props.type.replace(/range$/, ""),
        showTime: props.type.includes("time"),
        value: props.modelValue
      };
      if (props.type.includes("range")) {
        cmp = antDesignVue.DatePicker.RangePicker;
      }
      return vue.h(cmp, compProps);
    };
  }
});

exports.default = datePicker;
