// @epic-designer/.antd-y9OB2uUN/dist/chunks/datePicker.mjs
import { defineComponent, watch, h } from 'vue';
import { DatePicker } from 'ant-design-vue';

const datePicker = defineComponent({
  emits: ["update:modelValue", "change", "blur"],
  name: "EDatePicker",
  props: {
    modelValue: {
      default: null,
      type: [String, Object, Array]
    },
    type: {
      default: "date",
      type: String
    }
  },
  setup(props, { emit }) {
    watch(
      () => props.type,
      () => {
        handleUpdate();
      }
    );
    function handleUpdate(e = null) {
      emit("update:modelValue", e);
      emit("change", e);
      emit("blur", e);
    }
    return () => {
      let cmp = DatePicker;
      const compProps = {
        "onUpdate:value": handleUpdate,
        picker: props.type.replace(/range$/, ""),
        showTime: props.type.includes("time"),
        value: props.modelValue
      };
      if (props.type.includes("range")) {
        cmp = DatePicker.RangePicker;
      }
      return h(cmp, compProps);
    };
  }
});

export { datePicker as default };
