// @epic-designer/.antd-y9OB2uUN/dist/chunks/form.mjs
import { defineComponent, ref, computed, createBlock, openBlock, unref, mergeProps, withCtx, renderSlot, createElementBlock, Fragment, renderList } from 'vue';
import { useForm } from 'epic-designer';
import { Form } from 'ant-design-vue';

const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "form",
  props: {
    componentSchema: { type: Object, required: true, default: () => ({ type: "" }) },
    scrollToFirstError: { type: Boolean, required: false, default: false }
  },
  setup(__props, { expose: __expose }) {
    var _a, _b, _c;
    const props = __props;
    const form = ref(null);
    const { formData, formInstances } = useForm(
      (_c = (_b = (_a = props.componentSchema) == null ? void 0 : _a.props) == null ? void 0 : _b.name) != null ? _c : "default"
    );
    function getData() {
      return formData;
    }
    async function validate() {
      var _a2, _b2;
      try {
        return await ((_a2 = form.value) == null ? void 0 : _a2.validateFields());
      } catch (error) {
        if (props.scrollToFirstError) {
          (_b2 = form.value) == null ? void 0 : _b2.scrollToField(error.errorFields[0].name.toString());
        }
        throw error;
      }
    }
    function setData(data) {
      Object.assign(formData, data);
    }
    function resetData() {
      var _a2;
      (_a2 = form.value) == null ? void 0 : _a2.resetFields();
    }
    const mountedForm = (vNode) => {
      var _a2, _b2, _c2, _d, _e, _f, _g;
      form.value = (_a2 = vNode.component) == null ? void 0 : _a2.exposed;
      if (((_b2 = props.componentSchema) == null ? void 0 : _b2.type) === "form" && formInstances.value && form.value) {
        const name = (_g = (_f = (_d = (_c2 = props.componentSchema) == null ? void 0 : _c2.props) == null ? void 0 : _d.name) != null ? _f : (_e = props.componentSchema) == null ? void 0 : _e.name) != null ? _g : "default";
        form.value.validate = validate;
        formInstances.value[name] = form.value;
        form.value.getData = getData;
        form.value.setData = setData;
        form.value.resetData = resetData;
        return false;
      }
    };
    const formProps = computed(() => {
      const recordProps = props.componentSchema.props;
      let labelCol = recordProps.labelCol;
      let wrapperCol = recordProps.wrapperCol;
      if (recordProps.layout === "vertical") {
        labelCol = wrapperCol = { span: 24 };
      } else if (recordProps.layout === "inline" && recordProps.labelLayout === "fixed") {
        labelCol = {};
        wrapperCol = { flex: 1 };
      } else if (recordProps.labelLayout === "fixed") {
        labelCol = {
          flex: `${typeof recordProps.labelWidth === "number" ? `${recordProps.labelWidth}px` : recordProps.labelWidth}`
        };
        wrapperCol = { flex: 1 };
      }
      return {
        ...recordProps,
        labelCol,
        wrapperCol
      };
    });
    function onFinish() {
    }
    const children = computed(() => {
      var _a2;
      return (_a2 = props.componentSchema.children) != null ? _a2 : [];
    });
    __expose({
      form,
      getData,
      resetData,
      setData,
      validate
    });
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(Form), mergeProps({ model: unref(formData) }, formProps.value, {
        onFinish,
        onVnodeMounted: mountedForm
      }), {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "edit-node", {}, () => [
            (openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList(children.value, (item) => {
                return renderSlot(_ctx.$slots, "node", { componentSchema: item });
              }),
              256
              /* UNKEYED_FRAGMENT */
            ))
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 16, ["model"]);
    };
  }
});

export { _sfc_main as default };
