// @epic-designer/.antd-y9OB2uUN/dist/chunks/input.mjs
import { defineComponent, inject, createBlock, openBlock, unref, withCtx, createElementBlock, createCommentVNode, renderSlot } from 'vue';
import { useEventBus } from 'epic-designer';
import { Input } from 'ant-design-vue';

const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "input",
  props: {
    componentSchema: { type: Object, required: false, default: () => ({
      field: "",
      type: "input"
    }) },
    dataSelector: { type: Boolean, required: false, default: false },
    options: { type: Array, required: false, default: () => [] }
  },
  setup(__props) {
    const props = __props;
    const formData = inject("formData", {});
    const eventBus = useEventBus();
    const handleDataSelector = () => {
      eventBus == null ? void 0 : eventBus.emit("openDataSelector", props.componentSchema, props.options);
    };
    eventBus == null ? void 0 : eventBus.on("dataSelector:selected", (data) => {
      var _a, _b;
      if (data.nodeId !== props.componentSchema.id) {
        return;
      }
      const dataFieldMap = (_b = (_a = props.componentSchema.props) == null ? void 0 : _a.dataFieldMap) != null ? _b : [];
      dataFieldMap.forEach((item) => {
        formData[item.formField] = data.record[item.dataField];
      });
    });
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(Input), null, {
        suffix: withCtx(() => [
          props.dataSelector ? (openBlock(), createElementBlock("span", {
            key: 0,
            onClick: handleDataSelector,
            class: "iconfont icon--epic icon--epic--search-rounded"
          })) : createCommentVNode("v-if", true),
          renderSlot(_ctx.$slots, "suffix")
        ]),
        _: 3
        /* FORWARDED */
      });
    };
  }
});

export { _sfc_main as default };
