// @epic-designer/.antd-y9OB2uUN/dist/chunks/tabs.cjs
'use strict';

const vue = require('vue');
const antDesignVue = require('ant-design-vue');

const tabs = vue.defineComponent({
  props: {
    componentSchema: {
      default: () => ({}),
      require: true,
      type: Object
    }
  },
  setup(props, { slots }) {
    var _a, _b;
    const modelValue = vue.ref();
    vue.watch(
      () => props.componentSchema.props.defaultValue,
      (newValue) => {
        var _a2;
        const tabPaneLabels = (_a2 = props.componentSchema.children) == null ? void 0 : _a2.map(
          (node) => node.label
        );
        modelValue.value = (tabPaneLabels == null ? void 0 : tabPaneLabels.includes(newValue)) ? newValue : tabPaneLabels == null ? void 0 : tabPaneLabels[0];
      },
      {
        immediate: true
      }
    );
    const componentSchema = vue.computed(() => {
      var _a2;
      return {
        ...(_a2 = props.componentSchema) == null ? void 0 : _a2.props,
        activeKey: modelValue.value,
        "onUpdate:activeKey": (e) => {
          modelValue.value = e;
        }
      };
    });
    const children = (_b = (_a = props.componentSchema) == null ? void 0 : _a.children) != null ? _b : [];
    return () => vue.h(antDesignVue.Tabs, componentSchema.value, {
      default: () => {
        var _a2, _b2;
        const defaultSlotList = (_a2 = slots["edit-node"]) == null ? void 0 : _a2.call(slots)[0].children;
        if (defaultSlotList == null ? void 0 : defaultSlotList.length) {
          const list = (_b2 = defaultSlotList[0].children) == null ? void 0 : _b2.map((item) => {
            item.props.key = item.props.schema.label;
            item.props.tab = item.props.schema.label;
            return item;
          });
          return list;
        }
        return children.map(
          (componentSchema2) => vue.renderSlot(slots, "node", {
            componentSchema: componentSchema2,
            key: componentSchema2.label,
            tab: componentSchema2.label
          })
        );
      }
    });
  }
});

exports.default = tabs;
