// @epic-designer/.antd-y9OB2uUN/dist/chunks/uploadFile.mjs
import { defineComponent, useAttrs, ref, watch, computed, createElementBlock, openBlock, createVNode, unref, normalizeProps, guardReactiveProps, withCtx, createBlock, createCommentVNode, createElementVNode, nextTick } from 'vue';
import { getUUID, pluginManager } from 'epic-designer';
import { Form, Upload, Button, message } from 'ant-design-vue';

const _hoisted_1 = { class: "epic-upload-file" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "uploadFile",
  props: {
    maxCount: { type: Number, required: false, default: 99 },
    modelValue: { type: String, required: false, default: "" }
  },
  emits: ["update:modelValue", "change"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const formItemContext = Form.useInjectFormItemContext();
    const attrs = useAttrs();
    const fileList = ref([]);
    let urlString = "";
    watch(
      () => fileList.value,
      (list) => {
        urlString = list.filter((file) => file.status === "done").map((file) => file.url).join(",");
        emits("update:modelValue", urlString);
        emits("change", urlString);
        formItemContext.onFieldChange();
      }
    );
    watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue && fileList.value) {
          fileList.value = modelValue.split(",").map((url) => ({
            name: url,
            status: "done",
            uid: getUUID(),
            url
          }));
        }
      },
      { deep: true, immediate: true }
    );
    function handleUpdate(e) {
      nextTick(() => {
        fileList.value = e;
      });
    }
    const handleChange = (info) => {
      var _a, _b, _c, _d;
      if (info.file.status === "uploading") {
        return;
      }
      if (info.file.status === "done") {
        const url = (_d = (_b = (_a = info.file.response) == null ? void 0 : _a.data) == null ? void 0 : _b.url) != null ? _d : (_c = info.file.response) == null ? void 0 : _c.data;
        if (!info.file.url && !url && typeof url !== "string") {
          info.file.status = "error";
          message.error("\u4E0A\u4F20\u5931\u8D25");
          return;
        }
        info.file.url = url;
        info.file.thumbUrl = url;
      }
      if (info.file.status === "error") {
        message.error("upload error");
      }
    };
    const beforeUpload = () => {
    };
    const getUploadProps = computed(() => {
      var _a, _b;
      return {
        ...attrs,
        action: (_a = pluginManager.global.uploadFile) != null ? _a : attrs.action,
        headers: (_b = pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
        "onBefore-upload": beforeUpload,
        onChange: handleChange,
        onPreview: handlePreview,
        "file-list": fileList.value,
        "onUpdate:file-list": handleUpdate
      };
    });
    function handlePreview(e) {
      if (!e.url) return;
      window.open(e.url, "_blank");
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        createVNode(
          unref(Upload),
          normalizeProps(guardReactiveProps(getUploadProps.value)),
          {
            default: withCtx(() => {
              var _a, _b;
              return [
                ((_b = (_a = fileList.value) == null ? void 0 : _a.length) != null ? _b : 0) < props.maxCount ? (openBlock(), createBlock(unref(Button), {
                  key: 0,
                  class: "flex items-center"
                }, {
                  default: withCtx(() => [..._cache[0] || (_cache[0] = [
                    createElementVNode(
                      "span",
                      { class: "icon--epic icon--epic--cloud-upload-outlined text-lg" },
                      null,
                      -1
                      /* CACHED */
                    ),
                    createElementVNode(
                      "div",
                      { class: "ant-upload-text" },
                      "\u4E0A\u4F20\u6587\u4EF6",
                      -1
                      /* CACHED */
                    )
                  ])]),
                  _: 1
                  /* STABLE */
                })) : createCommentVNode("v-if", true)
              ];
            }),
            _: 1
            /* STABLE */
          },
          16
          /* FULL_PROPS */
        )
      ]);
    };
  }
});

export { _sfc_main as default };
