// @epic-designer/.antd-y9OB2uUN/dist/chunks/uploadImage.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const antDesignVue = require('ant-design-vue');

const _hoisted_1 = { class: "epic-upload-image" };
const _hoisted_2 = { key: 0 };
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "uploadImage",
  props: {
    maxCount: { type: Number, required: false, default: 99 },
    modelValue: { type: String, required: false, default: "" }
  },
  emits: ["update:modelValue", "change"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const formItemContext = antDesignVue.Form.useInjectFormItemContext();
    const attrs = vue.useAttrs();
    const fileList = vue.ref([]);
    let urlString = "";
    const imgUrl = vue.ref("");
    const visible = vue.ref(false);
    const setVisible = (value) => {
      visible.value = value;
    };
    vue.watch(
      () => fileList.value,
      (list) => {
        urlString = list.filter((file) => file.status === "done").map((file) => file.url).join(",");
        emits("update:modelValue", urlString);
        emits("change", urlString);
        formItemContext.onFieldChange();
      }
    );
    vue.watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue && fileList.value) {
          fileList.value = modelValue.split(",").map((url) => ({
            name: url,
            status: "done",
            uid: epicDesigner.getUUID(),
            url
          }));
        }
      },
      { deep: true, immediate: true }
    );
    function handleUpdate(e) {
      vue.nextTick(() => {
        fileList.value = e;
      });
    }
    const handleChange = (info) => {
      var _a, _b, _c, _d;
      if (info.file.status === "uploading") {
        return;
      }
      if (info.file.status === "done") {
        const url = (_d = (_b = (_a = info.file.response) == null ? void 0 : _a.data) == null ? void 0 : _b.url) != null ? _d : (_c = info.file.response) == null ? void 0 : _c.data;
        if (!info.file.url && !url && typeof url !== "string") {
          info.file.status = "error";
          antDesignVue.message.error("\u4E0A\u4F20\u5931\u8D25");
          return;
        }
        info.file.url = url;
        info.file.thumbUrl = url;
      }
      if (info.file.status === "error") {
        antDesignVue.message.error("upload error");
      }
    };
    const beforeUpload = () => {
    };
    const getUploadProps = vue.computed(() => {
      var _a, _b;
      return {
        ...attrs,
        accept: "image/gif,image/jpeg,image/jpg,image/png,image/svg",
        action: (_a = epicDesigner.pluginManager.global.uploadImage) != null ? _a : attrs.action,
        headers: (_b = epicDesigner.pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
        "onBefore-upload": beforeUpload,
        onChange: handleChange,
        onPreview: handlePreview,
        "file-list": fileList.value,
        "list-type": "picture-card",
        "onUpdate:file-list": handleUpdate
      };
    });
    function handlePreview(e) {
      if (!e.url) return;
      imgUrl.value = e.url;
      setVisible(true);
    }
    function previewError() {
      if (!imgUrl.value) return;
      antDesignVue.message.error("\u56FE\u7247\u5730\u5740\u65E0\u6CD5\u8BBF\u95EE!");
    }
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", _hoisted_1, [
        vue.createVNode(
          vue.unref(antDesignVue.Upload),
          vue.normalizeProps(vue.guardReactiveProps(getUploadProps.value)),
          {
            default: vue.withCtx(() => {
              var _a, _b;
              return [
                ((_b = (_a = fileList.value) == null ? void 0 : _a.length) != null ? _b : 0) < props.maxCount ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_2, [..._cache[0] || (_cache[0] = [
                  vue.createElementVNode(
                    "span",
                    { class: "icon--epic icon--epic--cloud-upload-outlined mr-2px text-$ep-text-main text-lg" },
                    null,
                    -1
                    /* CACHED */
                  ),
                  vue.createElementVNode(
                    "div",
                    { class: "ant-upload-text" },
                    "\u70B9\u51FB\u4E0A\u4F20",
                    -1
                    /* CACHED */
                  )
                ])])) : vue.createCommentVNode("v-if", true)
              ];
            }),
            _: 1
            /* STABLE */
          },
          16
          /* FULL_PROPS */
        ),
        vue.createVNode(vue.unref(antDesignVue.Image), {
          style: { "display": "none" },
          src: imgUrl.value,
          preview: {
            visible: visible.value,
            onVisibleChange: setVisible
          },
          onError: previewError
        }, null, 8, ["src", "preview"])
      ]);
    };
  }
});

exports.default = _sfc_main;
