// @epic-designer/.antd-y9OB2uUN/dist/chunks/uploadFile.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const antDesignVue = require('ant-design-vue');

const _hoisted_1 = { class: "epic-upload-file" };
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "uploadFile",
  props: {
    maxCount: { type: Number, required: false, default: 99 },
    modelValue: { type: String, required: false, default: "" }
  },
  emits: ["update:modelValue", "change"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const formItemContext = antDesignVue.Form.useInjectFormItemContext();
    const attrs = vue.useAttrs();
    const fileList = vue.ref([]);
    let urlString = "";
    vue.watch(
      () => fileList.value,
      (list) => {
        urlString = list.filter((file) => file.status === "done").map((file) => file.url).join(",");
        emits("update:modelValue", urlString);
        emits("change", urlString);
        formItemContext.onFieldChange();
      }
    );
    vue.watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue && fileList.value) {
          fileList.value = modelValue.split(",").map((url) => ({
            name: url,
            status: "done",
            uid: epicDesigner.getUUID(),
            url
          }));
        }
      },
      { deep: true, immediate: true }
    );
    function handleUpdate(e) {
      vue.nextTick(() => {
        fileList.value = e;
      });
    }
    const handleChange = (info) => {
      var _a, _b, _c, _d;
      if (info.file.status === "uploading") {
        return;
      }
      if (info.file.status === "done") {
        const url = (_d = (_b = (_a = info.file.response) == null ? void 0 : _a.data) == null ? void 0 : _b.url) != null ? _d : (_c = info.file.response) == null ? void 0 : _c.data;
        if (!info.file.url && !url && typeof url !== "string") {
          info.file.status = "error";
          antDesignVue.message.error("\u4E0A\u4F20\u5931\u8D25");
          return;
        }
        info.file.url = url;
        info.file.thumbUrl = url;
      }
      if (info.file.status === "error") {
        antDesignVue.message.error("upload error");
      }
    };
    const beforeUpload = () => {
    };
    const getUploadProps = vue.computed(() => {
      var _a, _b;
      return {
        ...attrs,
        action: (_a = epicDesigner.pluginManager.global.uploadFile) != null ? _a : attrs.action,
        headers: (_b = epicDesigner.pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
        "onBefore-upload": beforeUpload,
        onChange: handleChange,
        onPreview: handlePreview,
        "file-list": fileList.value,
        "onUpdate:file-list": handleUpdate
      };
    });
    function handlePreview(e) {
      if (!e.url) return;
      window.open(e.url, "_blank");
    }
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", _hoisted_1, [
        vue.createVNode(
          vue.unref(antDesignVue.Upload),
          vue.normalizeProps(vue.guardReactiveProps(getUploadProps.value)),
          {
            default: vue.withCtx(() => {
              var _a, _b;
              return [
                ((_b = (_a = fileList.value) == null ? void 0 : _a.length) != null ? _b : 0) < props.maxCount ? (vue.openBlock(), vue.createBlock(vue.unref(antDesignVue.Button), {
                  key: 0,
                  class: "flex items-center"
                }, {
                  default: vue.withCtx(() => [..._cache[0] || (_cache[0] = [
                    vue.createElementVNode(
                      "span",
                      { class: "icon--epic icon--epic--cloud-upload-outlined text-lg" },
                      null,
                      -1
                      /* CACHED */
                    ),
                    vue.createElementVNode(
                      "div",
                      { class: "ant-upload-text" },
                      "\u4E0A\u4F20\u6587\u4EF6",
                      -1
                      /* CACHED */
                    )
                  ])]),
                  _: 1
                  /* STABLE */
                })) : vue.createCommentVNode("v-if", true)
              ];
            }),
            _: 1
            /* STABLE */
          },
          16
          /* FULL_PROPS */
        )
      ]);
    };
  }
});

exports.default = _sfc_main;
