// @epic-designer/antd/dist/index.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const antDesignVue = require('ant-design-vue');

const Button = {
  component: () => import('./chunks/button.cjs'),
  config: {
    attribute: [
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.type",
        label: "\u7C7B\u578B",
        props: {
          clearable: true,
          options: [
            {
              label: "primary",
              value: "primary"
            },
            {
              label: "ghost",
              value: "ghost"
            },
            {
              label: "dashed",
              value: "dashed"
            },
            {
              label: "link",
              value: "link"
            },
            {
              label: "text",
              value: "text"
            },
            {
              label: "default",
              value: "default"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.htmlType",
        label: "\u64CD\u4F5C\u7C7B\u578B",
        props: {
          clearable: true,
          options: [
            {
              label: "button",
              value: "button"
            },
            {
              label: "submit",
              value: "submit"
            },
            {
              label: "reset",
              value: "reset"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.href",
        label: "\u94FE\u63A5",
        props: {
          allowClear: true,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.target",
        label: "\u94FE\u63A5\u6253\u5F00\u65B9\u5F0F",
        props: {
          clearable: true,
          options: [
            {
              label: "blank",
              value: "_blank"
            },
            {
              label: "parent",
              value: "_parent"
            },
            {
              label: "self",
              value: "_self"
            },
            {
              label: "top",
              value: "_top"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        show: ({ values }) => {
          var _a;
          return (_a = values.props) == null ? void 0 : _a.href;
        },
        type: "select"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "default",
              value: "default"
            },
            {
              label: "small",
              value: "small"
            },
            {
              label: "large",
              value: "large"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.shape",
        label: "\u5F62\u72B6",
        props: {
          clearable: true,
          options: [
            {
              label: "default",
              value: "default"
            },
            {
              label: "circle",
              value: "circle"
            },
            {
              label: "round",
              value: "round"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.ghost",
        label: "\u5E7D\u7075\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.loading",
        label: "\u52A0\u8F7D\u72B6\u6001",
        type: "switch"
      },
      {
        field: "props.block",
        label: "\u5BBD\u5EA6\u81EA\u9002\u5E94",
        type: "switch"
      },
      {
        field: "props.danger",
        label: "\u5371\u9669\u6309\u94AE",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ],
    event: [
      {
        description: "\u70B9\u51FB\u6309\u94AE\u65F6",
        type: "click"
      },
      {
        description: "\u53CC\u51FB\u6309\u94AE\u65F6",
        type: "dblclick"
      }
    ]
  },
  defaultSchema: {
    field: "input",
    input: false,
    label: "\u6309\u94AE",
    type: "button"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--fit-screen-rounded",
  sort: 1200
};

const Card = {
  component: () => import('./chunks/card.cjs'),
  config: {
    attribute: [
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "default",
              value: "default"
            },
            {
              label: "small",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.hoverable",
        label: "\u9F20\u6807\u60AC\u505C\u9634\u5F71",
        type: "switch"
      },
      {
        field: "props.loading",
        label: "\u52A0\u8F7D\u72B6\u6001",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u5361\u7247\u5E03\u5C40",
    type: "card",
    children: []
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--wysiwyg-rounded",
  sort: 700
};

const Cascader = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).Cascader,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "cascader"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.placement",
        label: "\u5F39\u51FA\u6846\u4F4D\u7F6E",
        props: {
          allowClear: true,
          options: [
            {
              label: "bottomLeft",
              value: "bottomLeft"
            },
            {
              label: "bottomRight",
              value: "bottomRight"
            },
            {
              label: "topLeft",
              value: "topLeft"
            },
            {
              label: "topRight",
              value: "topRight"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.showSearch",
        label: "\u53EF\u641C\u7D22",
        type: "switch"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.maxTagTextLength",
        label: "\u6807\u7B7E\u6700\u5927\u957F\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => values.props.multiple,
        type: "number"
      },
      {
        field: "props.maxTagCount",
        label: "\u6807\u7B7E\u663E\u793A\u6570\u91CF",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => values.props.multiple,
        type: "number"
      },
      {
        field: "props.showCheckedStrategy",
        label: "\u56DE\u586B\u65B9\u5F0F",
        props: {
          clearable: true,
          options: [
            {
              label: "SHOW_PARENT",
              value: "SHOW_PARENT"
            },
            {
              label: "SHOW_CHILD",
              value: "SHOW_CHILD"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        show: ({ values }) => values.props.multiple,
        type: "select"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        props: {
          tree: true
        },
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "array"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "cascader",
    input: true,
    label: "\u7EA7\u8054\u9009\u62E9\u5668",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ],
      placeholder: "\u8BF7\u9009\u62E9"
    },
    type: "cascader"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--full-coverage-outline",
  sort: 900
};

const Checkbox = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).CheckboxGroup,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "checkbox"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "array"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "checkbox",
    input: true,
    label: "\u591A\u9009\u6846",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ]
    },
    type: "checkbox"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--dialogs-outline-rounded",
  sort: 860
};

const Col = {
  component: () => import('./chunks/col.cjs'),
  config: {
    attribute: [
      {
        field: "props.span",
        label: "\u5360\u4F4D\u683C\u6570",
        type: "number"
      }
    ]
  },
  defaultSchema: {
    label: "\u6805\u683C\u5E03\u5C40-\u5217",
    props: {
      span: 6
    },
    type: "col",
    children: []
  },
  editConstraints: {
    locked: true
  }
};

const colorPicker = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).Input,
  config: {
    action: [],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u4FEE\u6539\u65F6",
        type: "change"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9\u65F6",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9\u65F6",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "color-picker",
    input: true,
    label: "\u989C\u8272\u9009\u62E9\u5668",
    props: {
      style: {
        width: "80px"
      },
      type: "color"
    },
    type: "color-picker"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--palette-outline",
  sort: 950
};

const DatePicker = {
  bindModel: "modelValue",
  component: () => import('./chunks/datePicker.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "date"
      },
      {
        changeSync: true,
        field: "props.type",
        label: "\u65E5\u671F\u7C7B\u578B",
        onChange({ value, values }) {
          values.props.defaultValue = null;
          values.props.placeholder = [
            "daterange",
            "monthrange",
            "quarterrange",
            "weekrange",
            "yearrange"
          ].includes(value) ? ["\u8BF7\u8F93\u5165", "\u8BF7\u8F93\u5165"] : "\u8BF7\u8F93\u5165";
          if (["date", "daterange"].includes(value)) {
            values.props.format = "YYYY-MM-DD";
            values.props.valueFormat = "YYYY-MM-DD";
          } else if (["week", "weekrange"].includes(value)) {
            values.props.format = "ww [\u5468]";
            values.props.valueFormat = "YYYY-MM-DD";
          } else if (["month", "monthrange"].includes(value)) {
            values.props.format = "YYYY-MM";
            values.props.valueFormat = "YYYY-MM";
          } else if (["quarter", "quarterrange"].includes(value)) {
            values.props.format = "YYYY-Q\u5B63\u5EA6";
            values.props.valueFormat = "YYYY-Q\u5B63\u5EA6";
          } else if (['"yearrange"', "year"].includes(value)) {
            values.props.format = "YYYY";
            values.props.valueFormat = "YYYY";
          }
        },
        props: {
          options: [
            {
              label: "\u65E5\u671F",
              value: "date"
            },
            {
              label: "\u5468",
              value: "week"
            },
            {
              label: "\u6708\u4EFD",
              value: "month"
            },
            {
              label: "\u5B63\u5EA6",
              value: "quarter"
            },
            {
              label: "\u5E74\u4EFD",
              value: "year"
            },
            {
              label: "\u65E5\u671F\u8303\u56F4",
              value: "daterange"
            },
            {
              label: "\u5468\u8303\u56F4",
              value: "weekrange"
            },
            {
              label: "\u6708\u4EFD\u8303\u56F4",
              value: "monthrange"
            },
            {
              label: "\u5B63\u5EA6\u8303\u56F4",
              value: "quarterrange"
            },
            {
              label: "\u5E74\u4EFD\u8303\u56F4",
              value: "yearrange"
            }
          ]
        },
        type: "select"
      },
      {
        field: "props.showTime",
        label: "\u589E\u52A0\u65F6\u95F4\u9009\u62E9",
        onChange: ({ value, values }) => {
          if (value) {
            values.props.valueFormat = "YYYY-MM-DD HH:mm:ss";
            values.props.format = "YYYY-MM-DD HH:mm:ss";
          } else {
            values.props.valueFormat = "YYYY-MM-DD";
            values.props.format = "YYYY-MM-DD";
          }
        },
        show({ values }) {
          return ["date", "daterange"].includes(values.props.type);
        },
        type: "switch"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => ![
          "daterange",
          "monthrange",
          "quarterrange",
          "weekrange",
          "yearrange"
        ].includes(values.props.type),
        type: "input"
      },
      {
        field: "props.placeholder.0",
        label: "\u5F00\u59CB\u5360\u4F4D\u7B26",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => [
          "daterange",
          "monthrange",
          "quarterrange",
          "weekrange",
          "yearrange"
        ].includes(values.props.type),
        type: "input"
      },
      {
        field: "props.placeholder.1",
        label: "\u7ED3\u675F\u5360\u4F4D\u7B26",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => [
          "daterange",
          "monthrange",
          "quarterrange",
          "weekrange",
          "yearrange"
        ].includes(values.props.type),
        type: "input"
      },
      {
        field: "props.format",
        label: "\u663E\u793A\u683C\u5F0F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.valueFormat",
        label: "\u6570\u636E\u683C\u5F0F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.placement",
        label: "\u5F39\u51FA\u6846\u4F4D\u7F6E",
        props: {
          allowClear: true,
          options: [
            {
              label: "bottomLeft",
              value: "bottomLeft"
            },
            {
              label: "bottomRight",
              value: "bottomRight"
            },
            {
              label: "topLeft",
              value: "topLeft"
            },
            {
              label: "topRight",
              value: "topRight"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.inputReadOnly",
        label: "\u7981\u6B62\u952E\u76D8\u8F93\u5165",
        type: "switch"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      },
      {
        description: "\u70B9\u51FB\u786E\u5B9A\u6309\u94AE\u65F6",
        type: "ok"
      }
    ]
  },
  defaultSchema: {
    field: "date",
    input: true,
    label: "\u65E5\u671F\u9009\u62E9\u5668",
    props: {
      format: "YYYY-MM-DD",
      placeholder: "\u8BF7\u9009\u62E9",
      type: "date",
      valueFormat: "YYYY-MM-DD"
    },
    type: "date"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--calendar-month-outline-rounded",
  sort: 910
};

const Form = {
  component: () => import('./chunks/form.cjs'),
  config: {
    action: [
      {
        description: "\u83B7\u53D6\u8868\u5355\u6570\u636E",
        type: "getData"
      },
      {
        description: "\u8BBE\u7F6E\u8868\u5355\u6570\u636E",
        type: "setData"
      },
      {
        description: "\u6821\u9A8C\u8868\u5355",
        type: "validate"
      }
    ],
    attribute: [
      {
        field: "props.name",
        label: "Name",
        type: "input"
      },
      {
        field: "props.layout",
        label: "\u8868\u5355\u5E03\u5C40",
        props: {
          allowClear: true,
          "option-type": "button",
          options: [
            {
              label: "\u6C34\u5E73",
              value: "horizontal"
            },
            {
              label: "\u5782\u76F4",
              value: "vertical"
            },
            {
              label: "\u5185\u8054",
              value: "inline"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "radio"
      },
      {
        field: "props.labelLayout",
        label: "\u6807\u7B7E\u5E03\u5C40",
        props: {
          "option-type": "button",
          options: [
            {
              label: "\u56FA\u5B9A\u5BBD\u5EA6",
              value: "fixed"
            },
            {
              label: "\u81EA\u9002\u5E94\u5BBD\u5EA6",
              value: "flex"
            }
          ]
        },
        type: "radio"
      },
      {
        field: "props.labelWidth",
        label: "\u6807\u7B7E\u5BBD\u5EA6",
        show: ({ values }) => {
          var _a;
          return ((_a = values.props) == null ? void 0 : _a.labelLayout) === "fixed";
        },
        type: "EInputSize"
      },
      {
        field: "props.labelCol.span",
        label: "\u6807\u7B7E\u5360\u6BD4",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => {
          var _a;
          return ((_a = values.props) == null ? void 0 : _a.labelLayout) === "flex";
        },
        type: "input"
      },
      {
        field: "props.wrapperCol.span",
        label: "\u63A7\u4EF6\u5360\u6BD4",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        show: ({ values }) => {
          var _a;
          return ((_a = values.props) == null ? void 0 : _a.labelLayout) === "flex";
        },
        type: "input"
      },
      {
        field: "props.labelAlign",
        label: "\u6807\u7B7E\u5BF9\u9F50",
        props: {
          "option-type": "button",
          options: [
            {
              label: "\u5DE6",
              value: "left"
            },
            {
              label: "\u53F3",
              value: "right"
            }
          ]
        },
        type: "radio"
      },
      {
        defaultValue: "default",
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.colon",
        label: "\u9690\u85CF\u5192\u53F7",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.hideRequiredMark",
        label: "\u9690\u85CF\u5FC5\u987B\u6807\u5FD7",
        type: "switch"
      },
      {
        field: "props.labelWrap",
        label: "\u6807\u7B7E\u6587\u672C\u6362\u884C",
        type: "switch"
      },
      {
        field: "props.scrollToFirstError",
        label: "\u6EDA\u52A8\u6821\u9A8C\u9519\u8BEF\u5904",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u8868\u5355",
    props: {
      labelCol: {
        span: 5
      },
      labelLayout: "fixed",
      labelWidth: "100px",
      name: "default",
      wrapperCol: {
        span: 19
      }
    },
    type: "form",
    children: []
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--list-alt-outline-rounded",
  sort: 600
};

const FormItem = {
  component: () => import('./chunks/formItem.cjs'),
  config: {},
  defaultSchema: {
    label: "\u8868\u5355\u9879",
    type: "form-item"
  }
};

const Input = {
  bindModel: "value",
  component: () => import('./chunks/input.cjs'),
  config: {
    action: [
      {
        description: "\u4F7F\u8F93\u5165\u6846\u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F\u8F93\u5165\u6846\u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u9009\u4E2D\u8F93\u5165\u6846\u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.type",
        label: "\u8F93\u5165\u7C7B\u578B",
        props: {
          allowClear: true,
          options: [
            {
              label: "text",
              value: "text"
            },
            {
              label: "number",
              value: "number"
            },
            {
              label: "password",
              value: "password"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.maxlength",
        label: "\u6700\u5927\u8F93\u5165\u957F\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.showCount",
        label: "\u5C55\u793A\u5B57\u6570",
        type: "switch"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u8F93\u5165\u503C\u65F6",
        type: "input"
      },
      {
        description: "\u503C\u4FEE\u6539\u65F6",
        type: "change"
      },
      {
        description: "\u6309\u4E0B\u56DE\u8F66\u65F6",
        type: "pressEnter"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9\u65F6",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9\u65F6",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "input",
    input: true,
    label: "\u8F93\u5165\u6846",
    props: {
      placeholder: "\u8BF7\u8F93\u5165"
    },
    type: "input"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--border-color-outline-rounded",
  sort: 700
};

const InputNumber = {
  bindModel: "value",
  component: () => import('./chunks/input-number.cjs'),
  config: {
    action: [
      {
        description: "\u4F7F input \u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F input \u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u9009\u4E2D input \u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165",
          size: "middle"
        },
        type: "number"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.keyboard",
        label: "\u952E\u76D8\u5FEB\u6377\u884C\u4E3A",
        type: "switch"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.precision",
        label: "\u7CBE\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.step",
        label: "\u6B65\u957F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.stringMode",
        label: "\u5B57\u7B26\u6A21\u5F0F",
        type: "switch"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "number"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u8F93\u5165\u503C",
        type: "input"
      },
      {
        description: "\u503C\u4FEE\u6539",
        type: "change"
      },
      {
        description: "\u6309\u4E0B\u56DE\u8F66\u7684\u56DE\u8C03",
        type: "pressEnter"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "number",
    input: true,
    label: "\u6570\u5B57\u8F93\u5165\u6846",
    props: {
      placeholder: "\u8BF7\u8F93\u5165",
      style: { width: "100%" }
    },
    type: "number"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--123-rounded",
  sort: 710
};

const Modal = {
  component: () => import('./chunks/modal.cjs'),
  config: {
    attribute: [
      {
        field: "title",
        label: "\u6807\u9898",
        type: "input"
      }
    ]
  },
  defaultSchema: {
    label: "\u6A21\u6001\u6846",
    type: "modal",
    children: []
  },
  icon: "epic-icon-xiala"
};

const Radio = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).RadioGroup,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "radio"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.optionType",
        label: "\u6309\u94AE\u6837\u5F0F",
        props: {
          checkedValue: "button",
          unCheckedValue: "default"
        },
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "radio",
    input: true,
    label: "\u5355\u9009\u6846",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ]
    },
    type: "radio"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--radio-button-checked-outline",
  sort: 850
};

const Row = {
  component: () => import('./chunks/row.cjs'),
  config: {
    attribute: [
      {
        field: "props.align",
        label: "\u5782\u76F4\u5BF9\u9F50\u65B9\u5F0F",
        props: {
          options: [
            {
              label: "top",
              value: "top"
            },
            {
              label: "middle",
              value: "middle"
            },
            {
              label: "bottom",
              value: "bottom"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9",
          style: { width: "100%" }
        },
        type: "select"
      },
      {
        field: "props.justify",
        label: "\u6C34\u5E73\u6392\u5217\u65B9\u5F0F",
        props: {
          options: [
            {
              label: "start",
              value: "start"
            },
            {
              label: "end",
              value: "end"
            },
            {
              label: "center",
              value: "center"
            },
            {
              label: "space-around",
              value: "space-around"
            },
            {
              label: "space-between",
              value: "space-between"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9",
          style: { width: "100%" }
        },
        type: "select"
      },
      {
        field: "props.gutter",
        label: "\u6805\u683C\u95F4\u8DDD",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "children",
        label: "\u5217\u7F16\u8F91",
        type: "EColEditor"
      },
      {
        field: "props.wrap",
        label: "\u4E0D\u81EA\u52A8\u6362\u884C",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u6805\u683C\u5E03\u5C40",
    props: {
      gutter: 12
    },
    type: "row",
    children: [
      {
        type: "col",
        children: [],
        props: {
          span: 12
        }
      },
      {
        type: "col",
        children: [],
        props: {
          span: 12
        }
      }
    ]
  },
  editConstraints: {
    childImmovable: true
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--width-normal-outline",
  sort: 800
};

const Select = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).Select,
  config: {
    action: [
      {
        description: "\u4F7F input \u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F input \u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u9009\u4E2D input \u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "select"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.listHeight",
        label: "\u5F39\u7A97\u9AD8\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.mode",
        label: "\u6A21\u5F0F",
        props: {
          allowClear: true,
          options: [
            {
              label: "multiple",
              value: "multiple"
            },
            {
              label: "tags",
              value: "tags"
            },
            {
              label: "combobox",
              value: "combobox"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.showSearch",
        label: "\u53EF\u641C\u7D22",
        type: "switch"
      },
      {
        field: "props.autoClearSearchValue",
        label: "\u9009\u4E2D\u9009\u9879\u540E\u6E05\u7A7A\u641C\u7D22\u6846",
        show: ({ values }) => values.props.mode && values.props.showSearch,
        type: "switch"
      },
      {
        field: "props.maxTagTextLength",
        label: "\u6700\u5927tag\u6587\u672C\u957F\u5EA6",
        show: ({ values }) => values.props.mode,
        type: "number"
      },
      {
        field: "props.maxTagCount",
        label: "\u6700\u5927tag\u663E\u793A\u6570",
        show: ({ values }) => values.props.mode,
        type: "number"
      },
      {
        field: "props.placement",
        label: "\u5F39\u51FA\u6846\u4F4D\u7F6E",
        props: {
          allowClear: true,
          options: [
            {
              label: "bottomLeft",
              value: "bottomLeft"
            },
            {
              label: "bottomRight",
              value: "bottomRight"
            },
            {
              label: "topLeft",
              value: "topLeft"
            },
            {
              label: "topRight",
              value: "topRight"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.labelInValue",
        label: "labelInValue",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        field: "props.options",
        label: "\u9009\u9879\u7BA1\u7406",
        layout: "vertical",
        type: "EOptionsEditor"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u4FEE\u6539",
        type: "change"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "select",
    input: true,
    label: "\u9009\u62E9\u6846",
    props: {
      options: [
        {
          label: "\u9009\u98791",
          value: "\u9009\u98791"
        },
        {
          label: "\u9009\u98792",
          value: "\u9009\u98792"
        }
      ],
      placeholder: "\u8BF7\u9009\u62E9"
    },
    type: "select"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--select",
  sort: 900
};

const Slider = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).Slider,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        props: {
          style: {
            width: "100%"
          }
        },
        type: "slider"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        changeSync: true,
        field: "props.range",
        label: "\u8303\u56F4\u9009\u62E9",
        onChange: ({ value, values }) => {
          values.props.defaultValue = value ? [0, 100] : 0;
        },
        type: "switch"
      },
      {
        field: "props.reverse",
        label: "\u5012\u8F6C\u8F68\u9053",
        type: "switch"
      },
      {
        field: "props.step",
        label: "\u6B65\u957F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.min",
        label: "\u6700\u5C0F\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.max",
        label: "\u6700\u5927\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.vertical",
        label: "\u5782\u76F4\u6A21\u5F0F",
        type: "switch"
      },
      {
        field: "props.dots",
        label: "\u663E\u793A\u523B\u5EA6",
        type: "switch"
      },
      // {
      //   label: "持续显示提示",
      //   type: "switch",
      //   field: "props.tooltipOpen",
      // },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "number"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u4E0Emouseup\u89E6\u53D1\u65F6\u673A\u4E00\u81F4",
        type: "afterChange"
      },
      {
        description: "\u62D6\u62FD\u7ED3\u675F\u6539\u53D8\u503C\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "slider",
    input: true,
    label: "\u6ED1\u5757",
    type: "slider"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--switches",
  sort: 930
};

const Switch = {
  bindModel: "checked",
  component: async () => (await import('ant-design-vue')).Switch,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "switch"
      },
      {
        field: "props.checkedValue",
        label: "ON\u72B6\u6001\u503C",
        onChange(e) {
          setTimeout(() => setDefaultValue(e));
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.unCheckedValue",
        label: "OFF\u72B6\u6001\u503C",
        onChange(e) {
          setTimeout(() => setDefaultValue(e));
        },
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.checkedChildren",
        label: "ON\u72B6\u6001\u63CF\u8FF0",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.unCheckedChildren",
        label: "OFF\u72B6\u6001\u63CF\u8FF0",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "default",
              value: "default"
            },
            {
              label: "small",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "boolean"
        },
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "switch",
    input: true,
    label: "\u5F00\u5173",
    props: {
      defaultValue: false
    },
    type: "switch"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--toggle-off-outline",
  sort: 930
};
function setDefaultValue(e) {
  var _a;
  const defaultValue = ((_a = e.values.props) == null ? void 0 : _a.unCheckedValue) || false;
  if (e.values.props.unCheckedValue === "") {
    delete e.values.props.unCheckedValue;
  }
  if (e.values.props.checkedValue === "") {
    delete e.values.props.checkedValue;
  }
  if (e.values.props) {
    e.values.props.defaultValue = defaultValue;
  } else {
    e.values.props = {
      defaultValue
    };
  }
}

const TabsPane = {
  component: () => import('./chunks/tabPane.cjs'),
  config: {
    attribute: [
      {
        field: "props.tab",
        label: "\u6807\u7B7E\u540D\u79F0",
        type: "input"
      }
    ]
  },
  defaultSchema: {
    label: "\u6807\u7B7E",
    type: "tab-pane",
    children: []
  },
  editConstraints: {
    locked: true
  }
};

const Tabs = {
  component: () => import('./chunks/tabs.cjs'),
  config: {
    attribute: [
      {
        field: "props.tabPosition",
        label: "\u9875\u7B7E\u4F4D\u7F6E",
        props: {
          allowClear: true,
          options: [
            {
              label: "top",
              value: "top"
            },
            {
              label: "right",
              value: "right"
            },
            {
              label: "bottom",
              value: "bottom"
            },
            {
              label: "left",
              value: "left"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.animated",
        label: "\u5207\u6362\u52A8\u753B",
        type: "switch"
      },
      {
        field: "props.tabBarGutter",
        label: "\u6807\u7B7E\u95F4\u9699",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "children",
        label: "\u9875\u7B7E\u7BA1\u7406",
        layout: "vertical",
        type: "ETabPaneEditor"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "large",
              value: "large"
            },
            {
              label: "middle",
              value: "middle"
            },
            {
              label: "small",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      }
    ]
  },
  defaultSchema: {
    label: "\u6807\u7B7E\u9875",
    props: {
      defaultValue: "\u6807\u7B7E1"
    },
    type: "tabs",
    children: [
      {
        label: "\u6807\u7B7E1",
        type: "tab-pane",
        children: []
      },
      {
        label: "\u6807\u7B7E2",
        type: "tab-pane",
        children: []
      }
    ]
  },
  editConstraints: {
    childImmovable: true
  },
  groupName: "\u5E03\u5C40",
  icon: "icon--epic--layout-tabs"
};

const Textarea = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).Textarea,
  config: {
    action: [
      {
        description: "\u4F7F input \u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u4F7F input \u5931\u53BB\u7126\u70B9",
        type: "blur"
      },
      {
        description: "\u9009\u4E2D input \u4E2D\u7684\u6587\u5B57",
        type: "select"
      }
    ],
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "textarea"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.autoSize.minRows",
        label: "\u6700\u5C0F\u884C\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.autoSize.maxRows",
        label: "\u6700\u5927\u884C\u6570",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.maxLength",
        label: "\u6700\u5927\u8F93\u5165\u957F\u5EA6",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.showCount",
        label: "\u7EDF\u8BA1\u5B57\u6570",
        type: "switch"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u9664",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u8F93\u5165\u503C",
        type: "input"
      },
      {
        description: "\u503C\u4FEE\u6539",
        type: "change"
      },
      {
        description: "\u6309\u4E0B\u56DE\u8F66\u7684\u56DE\u8C03",
        type: "pressEnter"
      },
      {
        description: "\u83B7\u53D6\u7126\u70B9",
        type: "focus"
      },
      {
        description: "\u5931\u53BB\u7126\u70B9",
        type: "blur"
      }
    ]
  },
  defaultSchema: {
    field: "textarea",
    input: true,
    label: "\u6587\u672C\u57DF",
    props: {
      placeholder: "\u8BF7\u8F93\u5165"
    },
    type: "textarea"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--edit-square-outline-rounded",
  sort: 705
};

const TimePicker = {
  bindModel: "value",
  component: async () => (await import('ant-design-vue')).TimePicker,
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.defaultValue",
        label: "\u9ED8\u8BA4\u503C",
        type: "time"
      },
      {
        field: "props.placeholder",
        label: "\u5360\u4F4D\u5185\u5BB9",
        type: "input"
      },
      {
        field: "props.format",
        label: "\u663E\u793A\u683C\u5F0F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.valueFormat",
        label: "\u6570\u636E\u683C\u5F0F",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "input"
      },
      {
        field: "props.size",
        label: "\u5C3A\u5BF8",
        props: {
          allowClear: true,
          options: [
            {
              label: "\u5927\u53F7",
              value: "large"
            },
            {
              label: "\u4E2D\u7B49",
              value: "middle"
            },
            {
              label: "\u5C0F\u578B",
              value: "small"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.hourStep",
        label: "\u5C0F\u65F6\u9009\u9879\u95F4\u9694",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.minuteStep",
        label: "\u5206\u949F\u9009\u9879\u95F4\u9694",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.secondStep",
        label: "\u79D2\u9009\u9879\u95F4\u9694",
        props: {
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.placement",
        label: "\u5F39\u51FA\u6846\u4F4D\u7F6E",
        props: {
          allowClear: true,
          options: [
            {
              label: "bottomLeft",
              value: "bottomLeft"
            },
            {
              label: "bottomRight",
              value: "bottomRight"
            },
            {
              label: "topLeft",
              value: "topLeft"
            },
            {
              label: "topRight",
              value: "topRight"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        changeSync: true,
        field: "props.use12Hours",
        label: "12\u5C0F\u65F6\u5236",
        onChange({ value, values }) {
          values.props.defaultValue = null;
          if (value) {
            values.props.format = "h:mm:ss a";
            values.props.valueFormat = "h:mm:ss a";
          } else {
            values.props.format = "HH:mm:ss";
            values.props.valueFormat = "HH:mm:ss";
          }
        },
        type: "switch"
      },
      {
        field: "props.bordered",
        label: "\u65E0\u8FB9\u6846",
        props: {
          checkedValue: false,
          unCheckedValue: true
        },
        type: "switch"
      },
      {
        field: "props.inputReadOnly",
        label: "\u7981\u6B62\u952E\u76D8\u8F93\u5165",
        type: "switch"
      },
      {
        field: "props.allowClear",
        label: "\u53EF\u6E05\u7A7A",
        type: "switch"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        type: "ERuleEditor"
      }
    ],
    event: [
      {
        description: "\u503C\u53D8\u5316\u65F6",
        type: "change"
      }
    ]
  },
  defaultSchema: {
    field: "time",
    input: true,
    label: "\u65F6\u95F4\u9009\u62E9\u5668",
    props: {
      format: "HH:mm:ss",
      placeholder: "\u8BF7\u8F93\u5165",
      valueFormat: "HH:mm:ss"
    },
    type: "time"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--time-line",
  sort: 920
};

const UploadFile = {
  bindModel: "modelValue",
  component: () => import('./chunks/uploadFile.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.action",
        label: "\u8BF7\u6C42\u5730\u5740",
        type: "input"
      },
      {
        field: "props.listType",
        label: "\u5217\u8868\u6837\u5F0F",
        props: {
          allowClear: true,
          options: [
            {
              label: "text",
              value: "text"
            },
            {
              label: "picture",
              value: "picture"
            }
          ],
          placeholder: "\u8BF7\u9009\u62E9"
        },
        type: "select"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.maxCount",
        label: "\u5141\u8BB8\u4E0A\u4F20\u6700\u5927\u6570\u91CF",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "string"
        },
        type: "ERuleEditor"
      }
    ]
  },
  defaultSchema: {
    field: "uploadFile",
    input: true,
    label: "\u4E0A\u4F20\u6587\u4EF6",
    props: {
      action: "https://examples.epicjs.cn/epic-mock/common/upload"
    },
    type: "upload-file"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--upload-rounded",
  sort: 920
};

const UploadImage = {
  bindModel: "modelValue",
  component: () => import('./chunks/uploadImage.cjs'),
  config: {
    attribute: [
      {
        field: "field",
        label: "\u6570\u636E\u5B57\u6BB5",
        type: "EpField"
      },
      {
        field: "label",
        label: "\u6807\u9898",
        type: "input"
      },
      {
        field: "props.action",
        label: "\u8BF7\u6C42\u5730\u5740",
        type: "input"
      },
      {
        field: "props.multiple",
        label: "\u591A\u9009",
        type: "switch"
      },
      {
        field: "props.maxCount",
        label: "\u5141\u8BB8\u4E0A\u4F20\u6700\u5927\u6570\u91CF",
        props: {
          min: 0,
          placeholder: "\u8BF7\u8F93\u5165"
        },
        type: "number"
      },
      {
        field: "props.disabled",
        label: "\u7981\u7528",
        type: "switch"
      },
      {
        field: "props.hidden",
        label: "\u9690\u85CF",
        type: "switch"
      },
      {
        description: "\u6821\u9A8C\u89C4\u5219\u9700\u8981\u914D\u5408\u8868\u5355\u4F7F\u7528",
        field: "rules",
        label: "\u8868\u5355\u6821\u9A8C",
        layout: "vertical",
        props: {
          ruleType: "string"
        },
        type: "ERuleEditor"
      }
    ]
  },
  defaultSchema: {
    field: "uploadImage",
    input: true,
    label: "\u4E0A\u4F20\u56FE\u7247",
    props: {
      action: "https://examples.epicjs.cn/epic-mock/common/upload"
    },
    type: "upload-image"
  },
  groupName: "\u8868\u5355",
  icon: "icon--epic--imagesmode-outline-rounded",
  sort: 920
};

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z = ".epic-designer-main .ant-input-number {\n  width: 100%;\n}\n.epic-designer-main .ant-collapse-content > .ant-collapse-content-box {\n  padding: 8px;\n}\n.epic-designer-main .ant-select {\n  width: 100%;\n}\n.epic-designer-main .epic-upload-image .avatar-uploader > .ant-upload {\n  width: 128px;\n  height: 128px;\n}\n.ep-input-size.ant-input-affix-wrapper {\n  padding-top: 0;\n  padding-right: 0;\n  padding-bottom: 0;\n}\n.ep-input-size .ant-select-selector {\n  border: none !important;\n  box-shadow: none !important;\n  padding: 0 4px !important;\n}\n";
styleInject(css_248z);

function setupAntd(pluginManager = epicDesigner.pluginManager, config = {}) {
  var _a;
  const versionArray = antDesignVue.version.split(".");
  const firstNumber = Number.parseInt(versionArray[0]);
  const style = document.createElement("style");
  if (firstNumber > 3) {
    const css = `
.epic-modal-ant .epic-modal-main {
  padding: 4px 12px 12px;
}
.epic-modal-ant .ant-modal-title {
  padding: 16px 16px 0px;
}
.epic-modal-ant .ant-modal-content {
  padding: 0px;
}`;
    style.append(document.createTextNode(css));
    document.head.append(style);
  } else {
    Form.config.attribute = (_a = Form.config.attribute) == null ? void 0 : _a.filter(
      (item) => item.label !== "\u7981\u7528"
    );
  }
  pluginManager.component.add(
    "Collapse",
    async () => (await import('ant-design-vue')).Collapse
  );
  pluginManager.component.add(
    "CollapseItem",
    async () => (await import('ant-design-vue')).CollapsePanel
  );
  pluginManager.component.add(
    "epTable",
    async () => (await import('ant-design-vue')).Table
  );
  const componentArray = [
    Form,
    FormItem,
    Input,
    Textarea,
    InputNumber,
    Radio,
    Checkbox,
    Select,
    Slider,
    TimePicker,
    DatePicker,
    Cascader,
    Switch,
    UploadImage,
    UploadFile,
    colorPicker,
    Button,
    Card,
    Row,
    Col,
    Modal,
    Tabs,
    TabsPane
  ];
  const { uploadFile, uploadImage } = config;
  if (uploadImage) UploadImage.defaultSchema.props.action = uploadImage;
  if (uploadFile) UploadFile.defaultSchema.props.action = uploadFile;
  vue.watchEffect(() => {
    var _a2, _b;
    if (pluginManager.global.uploadImage) {
      UploadImage.config.attribute = (_a2 = UploadImage.config.attribute) == null ? void 0 : _a2.filter(
        (item) => item.field !== "props.action"
      );
    }
    if (pluginManager.global.uploadFile) {
      UploadFile.config.attribute = (_b = UploadFile.config.attribute) == null ? void 0 : _b.filter(
        (item) => item.field !== "props.action"
      );
    }
  });
  componentArray.forEach((item) => {
    pluginManager.component.register(item);
    pluginManager.component.addBaseComponentType(item.defaultSchema.type);
  });
  pluginManager.global.$message = {
    error: antDesignVue.message.error,
    info: antDesignVue.message.info,
    success: antDesignVue.message.success,
    warning: antDesignVue.message.warning
  };
  pluginManager.designer.setInitialized(true);
}

exports.setupAntd = setupAntd;
