// epic-designer/dist/ui-kit/panel-ui/src/activitybars/outline/outline.vue2.js
import { defineComponent as g, computed as h, createElementBlock as v, openBlock as a, createVNode as x, unref as e, withCtx as N, createElementVNode as s, withModifiers as l, normalizeClass as k, createBlock as C, createCommentVNode as S, createTextVNode as M, toDisplayString as m } from "vue";
/* empty css                                 */
import b from "../../../../base-ui/src/icon/icon.vue.js";
/* empty css                                             */
/* empty css                                                  */
import { pluginManager as w } from "../../../../../manager/src/pluginManager.js";
import "../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import "@vueuse/core";
import "../../../../../hooks/src/store/index.js";
import { useDesignerContext as B } from "../../../../../hooks/src/designer/useDesignerContext.js";
/* empty css                                           */
import V from "../../../../base-ui/src/tree/tree.vue.js";
const $ = { class: "epic-outline" }, D = ["onMouseenter"], E = { class: "max-w-full truncate" }, H = { class: "epic-node-type-text w-0 flex-1 truncate" }, Q = /* @__PURE__ */ g({
  __name: "outline",
  setup(T) {
    const t = B(), u = t.pageSchema, f = h(() => {
      var i;
      const n = (i = t.state.selectedNode) == null ? void 0 : i.id;
      return n ? [n] : [];
    });
    function _(n) {
      t.setSelectedNode(n.componentSchema);
    }
    return (n, i) => {
      var r;
      return a(), v("div", $, [
        x(e(V), {
          options: e(u).schemas,
          draggable: "",
          "selected-keys": f.value,
          "hover-key": ((r = e(t).state.hoverNode) == null ? void 0 : r.id) ?? "",
          onNodeClick: _
        }, {
          "tree-node": N(({ schema: o }) => {
            var d, p, c;
            return [
              s("div", {
                class: k(["epic-text-padding flex", { hidden: (d = o.props) == null ? void 0 : d.hidden }]),
                onMouseenter: l((y) => e(t).setHoverNode(o), ["stop"]),
                onMouseleave: i[0] || (i[0] = l((y) => e(t).setHoverNode(null), ["stop"]))
              }, [
                s("span", E, [
                  (p = o.props) != null && p.hidden ? (a(), C(e(b), {
                    key: 0,
                    name: "icon--epic--visibility-off-outline-rounded",
                    class: "translate-y-2px"
                  })) : S("", !0),
                  M(" " + m(o.label ?? ((c = e(w).component.getConfigByType(o.type)) == null ? void 0 : c.defaultSchema.label)), 1)
                ]),
                s("span", H, m(o.id), 1)
              ], 42, D)
            ];
          }),
          _: 1
        }, 8, ["options", "selected-keys", "hover-key"])
      ]);
    };
  }
});
export {
  Q as default
};
