// epic-designer/dist/ui-kit/panel-ui/src/components/ETabPaneEditor/index.vue2.js
import { defineComponent as _, computed as b, createElementBlock as r, openBlock as p, Fragment as x, createElementVNode as m, createVNode as n, unref as i, withCtx as y, createCommentVNode as k } from "vue";
/* empty css                                 */
import u from "../../../../base-ui/src/icon/icon.vue.js";
/* empty css                                             */
/* empty css                                                  */
import { pluginManager as U } from "../../../../../manager/src/pluginManager.js";
import { getUUID as C } from "../../../../../utils/src/common/string.js";
import "../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import "@vueuse/core";
import "../../../../../hooks/src/store/index.js";
/* empty css                                           */
import A from "vuedraggable";
const I = { class: "epic-propedit-item p-2" }, N = {
  key: 0,
  class: "epic-option-del-btn flex items-center"
}, H = /* @__PURE__ */ _({
  inheritAttrs: !1,
  __name: "index",
  props: {
    modelValue: {
      default: () => [],
      type: Array
    }
  },
  emits: ["update:modelValue"],
  setup(c, { emit: s }) {
    const g = c, v = s, f = U.component.get("input"), e = b({
      get() {
        return g.modelValue;
      },
      set(t) {
        v("update:modelValue", t);
      }
    });
    function h() {
      const t = e.value.length + 1, l = {
        id: C(),
        label: `标签${t}`,
        type: "tab-pane",
        children: []
      };
      e.value.push(l);
    }
    function V(t) {
      e.value.splice(t, 1);
    }
    return (t, l) => (p(), r(x, null, [
      m("div", I, [
        n(i(A), {
          modelValue: e.value,
          "onUpdate:modelValue": l[0] || (l[0] = (o) => e.value = o),
          "item-key": "id",
          "component-data": {
            type: "transition-group"
          },
          group: "option-list",
          handle: ".handle",
          animation: 200
        }, {
          item: y(({ element: o, index: d }) => [
            (p(), r("div", {
              key: d,
              class: "epic-tab-pane-editor-item my-2 grid grid-cols-[auto_auto_16px] items-center gap-2"
            }, [
              n(i(u), {
                class: "handle cursor-move text-lg",
                name: "icon--epic--drag"
              }),
              n(i(f), {
                value: o.label,
                "onUpdate:value": (a) => o.label = a,
                modelValue: o.label,
                "onUpdate:modelValue": (a) => o.label = a
              }, null, 8, ["value", "onUpdate:value", "modelValue", "onUpdate:modelValue"]),
              e.value.length > 1 ? (p(), r("div", N, [
                n(i(u), {
                  class: "hover:text-red cursor-pointer text-lg",
                  name: "icon--epic--delete-outline-rounded",
                  onClick: (a) => V(d)
                }, null, 8, ["onClick"])
              ])) : k("", !0)
            ]))
          ]),
          _: 1
        }, 8, ["modelValue"])
      ]),
      m("div", {
        class: "ep-button ghost primary",
        onClick: h
      }, "添加")
    ], 64));
  }
});
export {
  H as default
};
