// epic-designer/dist/hooks/src/plugin/useFormSchema.js
import "vue";
/* empty css                                     */
/* empty css                                                 */
/* empty css                                                      */
import "@vueuse/core";
import "../store/index.js";
import "../../../manager/src/pluginManager.js";
/* empty css                                               */
import "vuedraggable";
import { deepCompareAndModify as e } from "../../../utils/src/common/data.js";
import "../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
function s() {
  const o = [
    {
      id: "root",
      label: "表单",
      props: {
        colon: !0,
        labelAlign: "right",
        labelCol: {
          span: 5
        },
        labelLayout: "fixed",
        labelPlacement: "left",
        labelWidth: 100,
        layout: "horizontal",
        name: "default",
        style: {
          padding: "16px"
        },
        wrapperCol: {
          span: 19
        }
      },
      type: "form",
      children: []
    }
  ];
  function t(r) {
    e(o, r);
  }
  return {
    formSchema: o,
    setFormSchema: t
  };
}
export {
  s as useFormSchema
};
