// epic-designer/dist/hooks/src/plugin/usePageSchema.js
import { reactive as r } from "vue";
/* empty css                                     */
/* empty css                                                 */
/* empty css                                                      */
import "@vueuse/core";
import "../store/index.js";
import "../../../manager/src/pluginManager.js";
/* empty css                                               */
import "vuedraggable";
import { deepCompareAndModify as p, deepClone as a } from "../../../utils/src/common/data.js";
import "../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
const n = {
  mobile: "mobile",
  pad: "tablet",
  pc: "desktop"
};
function c(e) {
  var o;
  if ((o = e.canvas) != null && o.mode && e.canvas.mode in n) {
    const t = n[e.canvas.mode];
    return {
      ...e,
      canvas: {
        ...e.canvas,
        mode: t
      }
    };
  }
  return e;
}
const m = {
  script: `const { defineExpose, find } = epic;
  
  function test (){
      console.log('test')
  }
  
  // 通过defineExpose暴露的函数或者属性
  defineExpose({
   test
  })`
};
function C() {
  const e = r({
    schemas: [],
    script: m.script
  });
  function o(t) {
    const i = c(t);
    p(e, a(i));
  }
  return {
    pageSchema: e,
    setPageSchema: o
  };
}
export {
  C as usePageSchema
};
