// epic-designer/dist/core/src/components/designer/src/modules/header/index.vue2.js
import { defineComponent as h, createElementBlock as x, openBlock as _, createElementVNode as e, renderSlot as i, createTextVNode as n, toDisplayString as g, unref as o, createVNode as r, withCtx as p } from "vue";
import v from "../../../../../static/logo.png.js";
/* empty css                                                 */
import a from "../../../../../../../ui-kit/base-ui/src/icon/icon.vue.js";
/* empty css                                                             */
/* empty css                                                                  */
import { pluginManager as $ } from "../../../../../../../manager/src/pluginManager.js";
import "../../../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import "@vueuse/core";
import "../../../../../../../hooks/src/store/index.js";
import { useDesignerContext as k } from "../../../../../../../hooks/src/designer/useDesignerContext.js";
/* empty css                                                           */
import "vuedraggable";
const w = { class: "epic-header relative" }, C = { class: "epic-header-item flex flex-1 items-center" }, y = { class: "epic-header-item text-12px flex flex-1 justify-center" }, b = { class: "epic-header-item flex flex-1 items-center justify-end" }, z = { class: "flex! h-full items-center" }, B = { class: "ml-2" }, D = { class: "flex! h-full items-center" }, I = /* @__PURE__ */ h({
  __name: "index",
  emits: ["save", "reset", "preview"],
  setup(E, { emit: c }) {
    const l = c, m = $.component.get("button"), d = k().props;
    function f() {
      l("preview");
    }
    function u() {
      l("save");
    }
    return (s, t) => (_(), x("header", w, [
      e("div", C, [
        i(s.$slots, "prefix", {}, () => [
          t[0] || (t[0] = e("a", {
            class: "decoration-none flex items-center",
            href: "https://github.com/Kchengz/epic-designer",
            target: "_blank"
          }, [
            e("img", {
              src: v,
              class: "w-18px h-18px",
              alt: "",
              srcset: ""
            }),
            e("span", { class: "ml-3" }, "EpicDesigner")
          ], -1))
        ])
      ]),
      e("div", y, [
        i(s.$slots, "title", {}, () => [
          n(g(o(d).title), 1)
        ])
      ]),
      e("div", b, [
        i(s.$slots, "right-prefix"),
        i(s.$slots, "right-action", {}, () => [
          e("div", null, [
            r(o(m), {
              size: "small",
              onClick: f
            }, {
              default: p(() => [
                e("span", z, [
                  r(o(a), {
                    name: "icon--epic--eye",
                    class: "mr-4px"
                  }),
                  t[1] || (t[1] = n(" 预览 ", -1))
                ])
              ]),
              _: 1
            })
          ]),
          e("div", B, [
            r(o(m), {
              size: "small",
              onClick: u
            }, {
              default: p(() => [
                e("span", D, [
                  r(o(a), {
                    name: "icon--epic--save-outline-rounded",
                    class: "mr-4px"
                  }),
                  t[2] || (t[2] = n(" 保存 ", -1))
                ])
              ]),
              _: 1
            })
          ])
        ]),
        i(s.$slots, "right-suffix")
      ])
    ]));
  }
});
export {
  I as default
};
