// epic-designer/dist/core/src/components/designer/src/designer.vue.js
import { defineComponent as X, ref as _, watchEffect as Y, provide as j, onUnmounted as q, createElementBlock as x, createBlock as $, unref as o, openBlock as l, createVNode as r, Suspense as F, withCtx as t, createElementVNode as g, createCommentVNode as M, renderSlot as i, normalizeClass as Q, resolveDynamicComponent as ee, nextTick as oe } from "vue";
/* empty css                                           */
import R from "../../../../../ui-kit/base-ui/src/asyncLoader/designerLoader.vue.js";
/* empty css                                                       */
/* empty css                                                            */
import { loadAsyncComponent as p } from "../../../../../utils/src/common/component.js";
import { migrateComponentProps as te, deepClone as re } from "../../../../../utils/src/common/data.js";
import "../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import { createEventBus as ie } from "../../../../../hooks/src/common/createEventBus.js";
import "@vueuse/core";
import { useStore as ae } from "../../../../../hooks/src/store/index.js";
import { DESIGNER_CONTEXT_KEY as ne } from "../../../../../hooks/src/designer/useDesignerContext.js";
import { providePageManager as se } from "../../../../../hooks/src/designer/usePageManager.js";
import { pluginManager as y } from "../../../../../manager/src/pluginManager.js";
/* empty css                                                     */
import "vuedraggable";
import { setupPanel as de } from "../../../../../ui-kit/panel-ui/src/index.js";
import { useDesigner as le } from "../hooks/useDesigner.js";
import { useHotkeys as pe } from "../hooks/useHotkeys.js";
import me from "./modules/preview/index.vue.js";
const fe = {
  key: 0,
  class: "epic-loading-box"
}, ce = {
  key: 0,
  class: "epic-header-container"
}, ue = { class: "epic-loading-box" }, Ie = /* @__PURE__ */ X({
  __name: "designer",
  props: {
    canvasMode: { default: "desktop" },
    defaultSchema: {},
    disabledZoom: { type: Boolean, default: !1 },
    draggable: { type: Boolean, default: !0 },
    formMode: { type: Boolean, default: !1 },
    hiddenHeader: { type: Boolean, default: !1 },
    hidePreviewConfirm: { type: Boolean },
    lockDefaultSchemaEdit: { type: Boolean, default: !1 },
    showHiddenItems: { type: Boolean, default: !0 },
    sourceCodeReadOnly: { type: Boolean },
    tableJson: {},
    title: { default: "EpicDesigner默认项目" }
  },
  emits: [
    "ready",
    "save",
    "reset",
    "imported",
    "toggleDeviceMode"
  ],
  setup(E, { expose: P, emit: N }) {
    const s = E, a = N;
    de(y);
    const T = p(() => import("./modules/header/index.vue.js")), I = p(
      () => import("./modules/activityBar/index.vue.js")
    ), O = p(
      () => import("./modules/editContainer/index.vue.js")
    ), V = p(
      () => import("./modules/rightSidebar/index.vue.js")
    ), k = y.component.get("epBuilderSlot"), B = _(null), {
      handleDelete: w,
      handleDuplicate: Z,
      pageManager: m,
      pageSchema: f,
      ready: z,
      reset: D,
      revoke: n,
      setHoverNode: S,
      setSelectedNode: c,
      state: u
    } = le(s, a), { setTarget: A } = pe({
      emit: a,
      handleDelete: w,
      pageSchema: f,
      revoke: n,
      setSelectedNode: c,
      state: u
    }), { disabledZoom: K } = ae();
    Y(() => {
      K.value = s.disabledZoom;
    }), ie(), j(ne, {
      handleDelete: w,
      handleDuplicate: Z,
      handleImported: W,
      handleToggleDeviceMode: U,
      pageSchema: f,
      preview: h,
      props: s,
      reset: D,
      revoke: n,
      save: v,
      setDisabledHover: G,
      setHoverNode: S,
      setSelectedNode: c,
      state: u
    }), se(m);
    const d = _(null);
    function L() {
      oe(() => {
        var e;
        z.value = !0, d.value && A(d.value), (e = d.value) == null || e.addEventListener("wheel", H, {
          passive: !1
        }), a("ready", { pageManager: m });
      });
    }
    async function G(e = !1) {
      u.disabledHover = e;
    }
    function J(e) {
      te(e), m.setPageSchema(e), c(), n.push("加载数据");
    }
    function C() {
      return re(f);
    }
    function v() {
      a("save", C());
    }
    function U(e) {
      a("toggleDeviceMode", e);
    }
    function W(e) {
      a("imported", e);
    }
    function h() {
      B.value.handleOpen();
    }
    function H(e) {
      e.ctrlKey && e.preventDefault();
    }
    return q(() => {
      var e;
      (e = d.value) == null || e.removeEventListener("wheel", H);
    }), P({
      exportHistory: n.exportHistory,
      getData: C,
      importHistory: n.importHistory,
      preview: h,
      reset: D,
      revoke: n,
      save: v,
      setData: J
    }), (e, b) => o(y).designer.initialized.value ? (l(), $(F, {
      key: 1,
      onResolve: L
    }, {
      default: t(() => [
        g("div", {
          class: "epic-designer-main ep-scoped",
          onMouseover: b[0] || (b[0] = (ve) => o(S)()),
          ref_key: "designerRef",
          ref: d,
          tabindex: "0"
        }, [
          s.hiddenHeader ? M("", !0) : (l(), x("div", ce, [
            i(e.$slots, "header", {}, () => [
              r(o(T), {
                onPreview: h,
                onSave: v
              }, {
                header: t(() => [
                  i(e.$slots, "header-prefix")
                ]),
                prefix: t(() => [
                  i(e.$slots, "header-prefix")
                ]),
                title: t(() => [
                  i(e.$slots, "header-title")
                ]),
                "right-prefix": t(() => [
                  i(e.$slots, "header-right-prefix")
                ]),
                "right-action": t(() => [
                  i(e.$slots, "header-right-action")
                ]),
                "right-suffix": t(() => [
                  i(e.$slots, "header-right-suffix")
                ]),
                _: 3
              })
            ])
          ])),
          g("div", {
            class: Q(["epic-split-view-container", { "hidden-header": E.hiddenHeader }])
          }, [
            r(o(I)),
            r(o(O)),
            r(o(V))
          ], 2),
          r(me, {
            ref_key: "previewRef",
            ref: B,
            "hide-confirm": s.hidePreviewConfirm
          }, null, 8, ["hide-confirm"]),
          o(k) ? (l(), $(ee(o(k)), { key: 1 })) : M("", !0)
        ], 544)
      ]),
      fallback: t(() => [
        g("div", ue, [
          r(o(R))
        ])
      ]),
      _: 3
    })) : (l(), x("div", fe, [
      r(o(R))
    ]));
  }
});
export {
  Ie as default
};
