// epic-designer/dist/ui-kit/panel-ui/src/components/EActionEditor/src/EActionEditorItem.vue.js
import { defineComponent as E, createElementBlock as d, openBlock as c, Fragment as V, renderList as C, createElementVNode as i, toDisplayString as v, createVNode as l, unref as r, withCtx as $, createCommentVNode as N, createTextVNode as w } from "vue";
/* empty css                                    */
import a from "../../../../../base-ui/src/icon/icon.vue.js";
/* empty css                                                */
/* empty css                                                     */
import { findSchemaById as A } from "../../../../../../utils/src/common/data.js";
import "../../../../../../git/epic/epic-designer/node_modules/.pnpm/jsep@1.4.0/node_modules/jsep/dist/jsep.js";
import "@vueuse/core";
import "../../../../../../hooks/src/store/index.js";
import { useDesignerContext as B } from "../../../../../../hooks/src/designer/useDesignerContext.js";
import "../../../../../../manager/src/pluginManager.js";
/* empty css                                              */
import D from "vuedraggable";
const I = { class: "epic-event-info" }, O = ["title"], S = { class: "epic-event-btn text-$ep-text-secondary flex items-center text-lg" }, j = { class: "epic-action-editor-main" }, L = { class: "epic-editor-item rounded" }, U = { class: "w-36px flex items-center text-lg" }, F = { class: "flex-1" }, T = { key: 0 }, q = { key: 1 }, z = { key: 2 }, G = { class: "epic-action-box text-$ep-text-helper text-lg" }, H = ["onClick"], J = ["onClick"], ie = /* @__PURE__ */ E({
  __name: "EActionEditorItem",
  props: {
    allEvents: {
      default: () => [],
      type: Array
    },
    events: {
      default: () => ({}),
      type: Object
    },
    itemEvents: {
      default: () => [],
      type: Array
    },
    modelValue: {
      default: () => ({}),
      type: Object
    }
  },
  emits: ["add", "edit", "update:modelValue"],
  setup(_, { emit: f }) {
    const s = _, p = f, { pageSchema: h } = B();
    function y(o) {
      p("add", o);
    }
    function x(o) {
      const t = A(h.schemas, o);
      return t == null ? void 0 : t.label;
    }
    function g(o, t) {
      var n;
      const e = k(t);
      e[t] = s.events[t].filter(
        (m, u) => o !== u
      ), (n = e[t]) != null && n.length || delete e[t], p("update:modelValue", e);
    }
    function b(o, t, e) {
      p("edit", o, t, e);
    }
    function k(o) {
      const t = {};
      return s.allEvents.forEach((e) => {
        if (s.events[e.type].length === 0 || e.type === o)
          return !1;
        t[e.type] = s.events[e.type];
      }), t;
    }
    return (o, t) => (c(!0), d(V, null, C(_.itemEvents, (e) => (c(), d("div", {
      key: e.type,
      class: "epic-event-item"
    }, [
      i("div", I, [
        i("div", {
          class: "epic-event-label",
          title: e.describe ?? e.description
        }, v(e.describe ?? e.description), 9, O),
        i("div", S, [
          l(r(a), {
            name: "icon--epic--add-rounded",
            onClick: (n) => y(e.type)
          }, null, 8, ["onClick"])
        ])
      ]),
      i("div", j, [
        l(r(D), {
          modelValue: s.events[e.type],
          "onUpdate:modelValue": (n) => s.events[e.type] = n,
          "item-key": "id",
          "component-data": {
            type: "transition-group"
          },
          group: "option-list",
          handle: ".handle",
          animation: 200
        }, {
          item: $(({ element: n, index: m }) => [
            i("div", L, [
              i("div", U, [
                l(r(a), {
                  class: "handle text-$ep-text-helper mr-2 cursor-move text-lg",
                  name: "icon--epic--drag"
                })
              ]),
              i("div", F, [
                n.type === "component" ? (c(), d("div", T, v(x(n.componentId)), 1)) : n.type === "custom" ? (c(), d("div", q, "自定义函数")) : n.type === "public" ? (c(), d("div", z, "公共函数")) : N("", !0),
                w(" " + v(n.methodName), 1)
              ]),
              i("div", G, [
                i("div", {
                  class: "epic-edit-btn",
                  onClick: (u) => b(m, e.type, n)
                }, [
                  l(r(a), { name: "icon--epic--page-info-outline-rounded" })
                ], 8, H),
                i("div", {
                  class: "epic-del-btn",
                  onClick: (u) => g(m, e.type)
                }, [
                  l(r(a), { name: "icon--epic--delete-outline-rounded" })
                ], 8, J)
              ])
            ])
          ]),
          _: 2
        }, 1032, ["modelValue", "onUpdate:modelValue"])
      ])
    ]))), 128));
  }
});
export {
  ie as default
};
