// @epic-designer/.naive-ui-di7cei2S/dist/chunks/uploadFile.cjs
'use strict';

const vue = require('vue');
const epicDesigner = require('epic-designer');
const naiveUi = require('naive-ui');

const uploadFile = vue.defineComponent({
  emits: ["update:modelValue", "change"],
  props: {
    modelValue: {
      default: "",
      type: String
    }
  },
  setup(props, { attrs, emit }) {
    const fileList = vue.ref([]);
    let urlString = "";
    vue.watch(
      () => fileList.value,
      (list) => {
        urlString = list.filter((file) => file.status === "finished").map((file) => file.url).join(",");
        emit("update:modelValue", urlString);
        emit("change", urlString);
      }
    );
    vue.watch(
      () => props.modelValue,
      (modelValue) => {
        if (urlString === modelValue) return;
        if (modelValue === "") {
          fileList.value = [];
          return;
        }
        if (modelValue !== null && fileList.value !== null) {
          fileList.value = modelValue.split(",").map((url) => ({
            id: epicDesigner.getUUID(),
            name: epicDesigner.getFileNameByUrl(url),
            status: "finished",
            url
          }));
        }
      },
      { immediate: true }
    );
    function handleUpdate(e) {
      vue.nextTick(() => {
        fileList.value = e;
      });
    }
    const handleError = () => {
    };
    const handleSuccess = ({ event, file }) => {
      var _a, _b, _c;
      const resInfo = event == null ? void 0 : event.target;
      const resData = JSON.parse((_a = resInfo == null ? void 0 : resInfo.response) != null ? _a : "{}");
      file.url = (_c = (_b = resData.data) == null ? void 0 : _b.url) != null ? _c : resData.data;
    };
    return () => {
      return vue.h("div", null, {
        default: () => {
          var _a, _b;
          return [
            vue.h(
              naiveUi.NUpload,
              {
                ...attrs,
                action: (_a = epicDesigner.pluginManager.global.uploadFile) != null ? _a : attrs.action,
                "default-upload": true,
                headers: (_b = epicDesigner.pluginManager.global.axiosConfig) == null ? void 0 : _b.headers,
                onError: handleError,
                onFinish: handleSuccess,
                "file-list": fileList.value,
                "onUpdate:file-list": handleUpdate
              },
              {
                default: () => [
                  vue.h(
                    naiveUi.NButton,
                    {},
                    {
                      default: () => [
                        vue.h("span", {
                          class: "icon--epic icon--epic--cloud-upload-outlined text-lg",
                          style: { "margin-right": "2px" }
                        }),
                        vue.h("span", null, { default: () => "\u4E0A\u4F20\u6587\u4EF6" })
                      ]
                    }
                  )
                ]
              }
            )
          ];
        }
      });
    };
  }
});

exports.default = uploadFile;
